# TOOL mothur-classifyseqs.R: "Classify 16S or 18S sequences to taxonomic units using Silva" (Classifies 16S and 18S rRNA sequences to taxonomic units using the Wang method with Silva reference set and taxonomy. As inputs you need a fasta file of aligned sequences, and a count_table file generated by Mothur. If you would like to use your own reference and taxonomy file, please provide them as input too. This tool is based on the Mothur tool classify.seqs.)
# INPUT a.fasta: "Aligned reads in FASTA file" TYPE FASTA
# INPUT OPTIONAL a.count_table: "Count table" TYPE MOTHUR_COUNT
# INPUT OPTIONAL own_reference.fasta: "Reference FASTA" TYPE FASTA
# INPUT OPTIONAL own_reference.tax: "Reference taxonomy file" TYPE GENERIC
# OUTPUT OPTIONAL sequences-taxonomy-assignment.txt
# OUTPUT OPTIONAL classification-summary.tsv
# PARAMETER OPTIONAL reference: "Reference" TYPE [silva: "silva.nr_v138.1", own:"own reference"] DEFAULT silva (Reference to use.)
# PARAMETER OPTIONAL iters: "Number of iterations" TYPE INTEGER FROM 10 TO 1000 DEFAULT 100 (How many iterations to do when calculating the bootstrap confidence score for your taxonomy.)

# EK 18.06.2013
# JTT 28.8.2013 count table and phenodata added
# ML 21.12.2016 update (new Silva version)
# ML 14.3.2017  reference option (bacterial vs whole)
# ML 23.3.2017  detach the last steps to another tool (mothur-classify-counttable.R), add iters-parameter and remove.lineage option
# EK 22.8.2018  updated Silva to v 132, removed bacterial option, added processors-parameter, and made link in the working directory to the reference files
# EK 11.5.2020  zip output fasta
# EK 25.3.2021  removed parameters for removing lineages as this is now handled with Phyloseq tools "Remove selected taxa" and "Filter by taxonomic group"
# EK 25.3.2021  add option to use own reference
# EK 19.4.2021  updated to Silva v138.1

# to be removed later:
# OUTPUT OPTIONAL log.txt
# PARAMETER OPTIONAL reference: "Reference" TYPE [bacterial: "bacterial subset of Silva db", full: "whole Silva db"] DEFAULT bacterial (Silva reference set to use.)
# PARAMETER OPTIONAL remove.chloroplast: "Remove taxon Chloroplast" TYPE [yes, no] DEFAULT yes (Remove taxon Chloroplast.)
# PARAMETER OPTIONAL remove.mitochondria: "Remove taxon Mitochondria" TYPE [yes, no] DEFAULT yes (Remove taxon Mitochondria.)
# PARAMETER OPTIONAL remove.archaea: "Remove taxon Archaea" TYPE [yes, no] DEFAULT yes (Remove taxon Archaea.)
# PARAMETER OPTIONAL remove.eukaryota: "Remove taxon Eukaryota" TYPE [yes, no] DEFAULT yes (Remove taxon Eukaryota.)
# PARAMETER OPTIONAL remove.unknown: "Remove taxon unknown" TYPE [yes, no] DEFAULT yes (Remove taxon unknown.)
# PARAMETER OPTIONAL remove.other: "Remove other lineages" TYPE STRING DEFAULT empty (List of other lineages to remove. You must use dots \(\".\"\) instead of semicolons \(\";\"\). Use dash \(\"-\"\) to separate taxons. For example: Bacteria.Firmicutes.-Bacteria.Bacteroidetes. )
# OUTPUT OPTIONAL picked.fasta.gz 
# OUTPUT OPTIONAL picked.count_table
# OUTPUT OPTIONAL picked-summary.tsv

source(file.path(chipster.common.path,"tool-utils.R"))
source(file.path(chipster.common.path,"zip-utils.R"))

# check out if the file is compressed and if so unzip it
unzipIfGZipFile("a.fasta")

# binary
binary <- c(file.path(chipster.tools.path,"mothur","mothur"))
version <- system(paste(binary,"--version"),intern = TRUE)
documentVersion("Mothur",version)

if (reference == "own") {
  if (fileNotOk("own_reference.fasta") || fileNotOk("own_reference.tax")){
    stop('CHIPSTER-NOTE: Provide your own reference and taxonomy files or select the Silva reference.')
  }
  reference.path <- "own_reference.fasta"
  taxonomy.path <- "own_reference.tax"
} else if (reference == "silva") {
  # Silva v132. Check data path when new tools-bin ready
  data.path <- c(file.path(chipster.tools.path,"mothur-silva-reference","silva"))
  # template.path <- c(file.path(data.path,"silva.nr_v132.align"))
  # taxonomy.path <- c(file.path(data.path,"silva.nr_v132.tax"))
  template.path <- c(file.path(data.path,"silva.nr_v138_1.align"))
  taxonomy.path <- c(file.path(data.path,"silva.nr_v138_1.tax"))
  # copy to working dir because mothur generates more files to the same directory
  # system(paste("ln -s ",template.path," silva.nr_v132.align"))
  # system(paste("ln -s ",taxonomy.path," silva.nr_v132.tax"))
  system(paste("ln -s ",template.path," silva.nr_v138_1.align"))
  system(paste("ln -s ",taxonomy.path," silva.nr_v138_1.tax"))
  # template.path <- "silva.nr_v132.align"
  # taxonomy.path <- "silva.nr_v132.tax"
  template.path <- "silva.nr_v138_1.align"
  taxonomy.path <- "silva.nr_v138_1.tax"
}

# batch file
# write(paste("classify.seqs(fasta=a.fasta, iters=1000, template=", template.path, ", taxonomy=", taxonomy.path, ")", sep=""), "batch.mth", append=F)
classifyseqs.options <- paste("classify.seqs(fasta=a.fasta")
if (file.exists("a.count_table")) {
  classifyseqs.options <- paste(classifyseqs.options,", count=a.count_table")
}
classifyseqs.options <- paste(classifyseqs.options,", iters=",iters,", template=",template.path,", taxonomy=",taxonomy.path,", processors=",chipster.threads.max,")",sep = "")
documentCommand(classifyseqs.options)
write(classifyseqs.options,"batch.mth",append = FALSE)
# command
command <- paste(binary,"batch.mth","> log.txt 2>&1")

# run
system(command)

# Output File Names: 
# a.silva.wang.taxonomy
# a.silva.wang.tax.summary


## test
write("get.current()","batch2.mth",append = FALSE)
system(paste(binary,"batch2.mth",">> log.txt 2>&1"))

system("ls -l >> log.txt")
# Postprocess output
# Postprocess output
system("find . -type f -name a.*.taxonomy -exec mv {} sequences-taxonomy-assignment.txt \\;")
system("find . -type f -name a.*.summary -exec mv {} classification-summary.tsv \\;")

#if (reference == "silva") {
#  system("mv a.nr_v132.wang.taxonomy sequences-taxonomy-assignment.txt")
#  system("mv a.nr_v132.wang.tax.summary classification-summary.tsv")
#}

# batch file 2: remove lineage, if the taxons to remove were listed
# toremove <- ""

#if (remove.chloroplast == "yes") {
#  toremove <- paste(toremove,"Chloroplast",sep = "-")
#}
#if (remove.mitochondria == "yes") {
#  toremove <- paste(toremove,"Mitochondria",sep = "-")
#}
#if (remove.archaea == "yes") {
#  toremove <- paste(toremove,"Archaea",sep = "-")
#}
#if (remove.eukaryota == "yes") {
#  toremove <- paste(toremove,"Eukaryota",sep = "-")
#}
#if (remove.unknown == "yes") {
#  toremove <- paste(toremove,"unknown",sep = "-")
#}
#if (remove.other != "empty") {
  # Change periods to semicolons. Semicolons are not accepted in STRING input.
#  remove.other <- gsub(".",";",remove.other,fixed = TRUE)
#  toremove <- paste(toremove,remove.other,sep = "-")
#}

#if (toremove != "") {
  # Remove leading dash
#  toremove <- substring(toremove,2)
#  removelineage.options <- paste("remove.lineage(fasta=a.fasta")
#  if (file.exists("a.count_table")) {
#    removelineage.options <- paste(removelineage.options,", count=a.count_table")
#  }
#  removelineage.options <- paste(removelineage.options,", taxonomy=sequences-taxonomy-assignment.txt, taxon=",toremove,")",sep = "")
#  documentCommand(removelineage.options)
#  write(removelineage.options,"batch.mth",append = FALSE)

  # command
#  command <- paste(binary,"batch.mth",">> log.txt 2>&1")

  # run
#  system(command)

  

  # Rename output files
  #system("mv a.pick.fasta picked.fasta")
  #if (file.exists("a.pick.count_table")) {
  #  system("mv a.pick.count_table picked.count_table")
  #}

  # batch file 3: classify.seqs again

  # write(paste("classify.seqs(fasta=a.fasta, iters=1000, template=", template.path, ", taxonomy=", taxonomy.path, ")", sep=""), "batch.mth", append=F)
  # classifyseqs.options <- paste("classify.seqs(fasta=picked.fasta")
  #if (file.exists("picked.count_table")) {
  #  classifyseqs.options <- paste(classifyseqs.options,", count=picked.count_table")
  #}
  #classifyseqs.options <- paste(classifyseqs.options,", processors=",chipster.threads.max,", iters=",iters,", template=",template.path,", taxonomy=",taxonomy.path,")",sep = "")
  #documentCommand(classifyseqs.options)
  #write(classifyseqs.options,"batch.mth",append = FALSE)
  # command
  #command <- paste(binary,"batch.mth",">> log.txt 2>&1")
  # run
  #system(command)

  ## test
  #write("get.current()","batch2.mth",append = FALSE)
  #system(paste(binary,"batch2.mth",">> log.txt 2>&1"))


  # Postprocess output
  #if (reference == "silva") {
  #  system("mv picked.nr_v132.wang.taxonomy sequences-taxonomy-assignment.txt")
  #  system("mv picked.nr_v132.wang.tax.summary classification-summary.tsv")
  #}

  # batch file 3: summary
  #summaryseqs.options <- paste("summary.seqs(fasta=picked.fasta, count=picked.count_table)")
  #documentCommand(summaryseqs.options)
  #write(summaryseqs.options,"summary.mth",append = FALSE)

  # command 3
  #command3 <- paste(binary,"summary.mth","> log_raw.txt")

  # run
  #system(command3)

  # zip output fasta
  #system("gzip picked.fasta")

  # Post process output
  #system("grep -A 10 Start log_raw.txt > picked-summary2.tsv")
  # Remove one tab to get the column naming look nice:
  #system("sed 's/^		/	/' picked-summary2.tsv > picked-summary.tsv")
#}
