# TOOL mothur-classifyseqs-its.R: "Classify ITS sequences to taxonomic units using UNITE" (Classifies unaligned ITS sequences to taxonomic units using the Wang method with UNITE reference set and taxonomy. As inputs you need a fasta file of unaligned sequences, and a count_table file generated by Mothur. If you would like to use your own reference and taxonomy file, please provide them as input too. This tool is based on the Mothur tools classify.seqs and summary.seqs.)
# INPUT a.fasta: "Unaligned contigs in FASTA file" TYPE FASTA
# INPUT a.count_table: "Count table" TYPE MOTHUR_COUNT
# INPUT OPTIONAL own_reference.fasta: "Reference FASTA" TYPE FASTA
# INPUT OPTIONAL own_reference.tax: "Reference taxonomy file" TYPE GENERIC
# OUTPUT sequences-taxonomy-assignment.txt
# OUTPUT classification-summary.tsv
# PARAMETER reference: "Reference" TYPE [own: "own reference", "FILES mothur-unite-reference .tax"] DEFAULT own (The reference options marked with _s are from the UNITE download2 set, so they comprise of all quality filtered unclustered UNITE and INSD sequences for Fungi \(887 399 for UNITE v8.2\). The options without _s are from the UNITE download1 set, which contains sequences included in the UNITE species hypotheses \(565 915 for UNITE v8.2\). The dynamic options are based on varying clustering thresholds for different species hypothesis. These choices were made manually by experts of those particular lineages of fungi.)
# PARAMETER OPTIONAL iters: "Number of iterations" TYPE INTEGER FROM 10 TO 1000 DEFAULT 100 (How many iterations to do when calculating the bootstrap confidence score for your taxonomy.)
# RUNTIME R-4.1.1

# AMS 18.12.2020
# EK 10.2.2021 removed unnecessary remove.lineages parameters
# EK 8.4.2021 removed parameters for removing unknown and user-specified lineages
# EK 16.1.2023 moved to Mothur v1.48

# PARAMETER OPTIONAL remove.chloroplast: "Remove taxon Chloroplast" TYPE [yes, no] DEFAULT yes (Remove taxon Chloroplast.)
# PARAMETER OPTIONAL remove.mitochondria: "Remove taxon Mitochondria" TYPE [yes, no] DEFAULT yes (Remove taxon Mitochondria.)
# PARAMETER OPTIONAL remove.archaea: "Remove taxon Archaea" TYPE [yes, no] DEFAULT yes (Remove taxon Archaea.)
# PARAMETER OPTIONAL remove.bacteria: "Remove taxon Bacteria" TYPE [yes, no] DEFAULT yes (Remove taxon Bacteria.)
# PARAMETER OPTIONAL remove.unknown: "Remove taxon unknown" TYPE [yes, no] DEFAULT yes (Remove taxon unknown.)
# PARAMETER OPTIONAL remove.other: "Remove lineages" TYPE STRING DEFAULT empty (List of other lineages to remove. You must use dots \(\".\"\) instead of semicolons \(\";\"\). Use dash \(\"-\"\) to separate taxons. For example: Bacteria.Firmicutes.-Bacteria.Bacteroidetes.)
# OUTPUT OPTIONAL picked.fasta.gz
# OUTPUT OPTIONAL picked.count_table
# OUTPUT OPTIONAL picked-summary.tsv


source(file.path(chipster.common.path, "tool-utils.R"))
source(file.path(chipster.common.path, "zip-utils.R"))

# check out if the file is compressed and if so unzip it
unzipIfGZipFile("a.fasta")

# binary
# binary <- c(file.path(chipster.tools.path,"mothur-1.44.3","mothur"))
binary <- c(file.path(chipster.tools.path, "mothur", "mothur"))
version <- system(paste(binary, "--version"), intern = TRUE)
documentVersion("Mothur", version)

if (reference == "own") {
  if (fileNotOk("own_reference.fasta") || fileNotOk("own_reference.tax")) {
    stop("CHIPSTER-NOTE: Provide your own reference and taxonomy files or select one of the provided ones.")
  }
  reference.path <- "own_reference.fasta"
  taxonomy.path <- "own_reference.tax"
} else {
  # Check data path when new tools-bin ready
  data.path <- c(file.path(chipster.tools.path, "mothur-unite-reference"))
  reference.file <- paste(reference, ".fasta", sep = "")
  taxonomy.file <- paste(reference, ".tax", sep = "")
  reference.path <- c(file.path(data.path, reference.file))
  taxonomy.path <- c(file.path(data.path, taxonomy.file))
  # copy to working dir because mothur generates more files to the same directory
  system(paste("ln -s ", reference.path, reference.file))
  system(paste("ln -s ", taxonomy.path, taxonomy.file))
  reference.path <- reference.file
  taxonomy.path <- taxonomy.file
}

# batch file
classifyseqs.options <- paste("classify.seqs(fasta=a.fasta, count=a.count_table, iters=", iters, ", reference=", reference.path, ", taxonomy=", taxonomy.path, ", processors=", chipster.threads.max, ")", sep = "")
documentCommand(classifyseqs.options)
write(classifyseqs.options, "batch.mth", append = FALSE)
# command
command <- paste(binary, "batch.mth", "> log.txt 2>&1")

# run
system(command)

## test
write("get.current()", "batch2.mth", append = FALSE)
system(paste(binary, "batch2.mth", ">> log.txt 2>&1"))

# Postprocess output
system("find . -type f -name a.*.taxonomy -exec mv {} sequences-taxonomy-assignment.txt \\;")
system("find . -type f -name a.*.summary -exec mv {} classification-summary.tsv \\;")

# batch file 2: remove lineage, if the taxons to remove were listed
# toremove <- ""

# if (remove.chloroplast == "yes") {
#  toremove <- paste(toremove,"Chloroplast",sep = "-")
# }
# if (remove.mitochondria == "yes") {
#  toremove <- paste(toremove,"Mitochondria",sep = "-")
# }
# if (remove.archaea == "yes") {
#  toremove <- paste(toremove,"Archaea",sep = "-")
# }
# if (remove.bacteria == "yes") {
#  toremove <- paste(toremove,"Bacteria",sep = "-")
# }
# if (remove.unknown == "yes") {
#   toremove <- paste(toremove,"unknown",sep = "-")
# }
# if (remove.other != "empty") {
# Change periods to semicolons. Semicolons are not accepted in STRING input.
#   remove.other <- gsub(".",";",remove.other,fixed = TRUE)
#   toremove <- paste(toremove,remove.other,sep = "-")
# }

# if (toremove != "") {
# Remove leading dash
#   toremove <- substring(toremove,2)
#   removelineage.options <- paste("remove.lineage(fasta=a.fasta")
#   if (file.exists("a.count_table")) {
#     removelineage.options <- paste(removelineage.options,", count=a.count_table")
#   }
#   removelineage.options <- paste(removelineage.options,", taxonomy=sequences-taxonomy-assignment.txt, taxon=",toremove,")",sep = "")
#   documentCommand(removelineage.options)
#   write(removelineage.options,"batch.mth",append = FALSE)

# command
#   command <- paste(binary,"batch.mth",">> log.txt 2>&1")

# run
#   system(command)

# Rename output files
#   system("mv a.pick.fasta picked.fasta")
#   if (file.exists("a.pick.count_table")) {
#     system("mv a.pick.count_table picked.count_table")
#   }

# batch file 3: classify.seqs again

#   classifyseqs.options <- paste("classify.seqs(fasta=picked.fasta")
#   if (file.exists("picked.count_table")) {
#     classifyseqs.options <- paste(classifyseqs.options,", count=picked.count_table")
#   }
#   classifyseqs.options <- paste(classifyseqs.options,", processors=",chipster.threads.max,", iters=",iters,", reference=",reference.path,", taxonomy=",taxonomy.path,")",sep = "")
#   documentCommand(classifyseqs.options)
#   write(classifyseqs.options,"batch.mth",append = FALSE)
# command
#   command <- paste(binary,"batch.mth",">> log.txt 2>&1")
# run
#   system(command)

## test
#   write("get.current()","batch2.mth",append = FALSE)
#   system(paste(binary,"batch2.mth",">> log.txt 2>&1"))

# Postprocess output
#   system("find . -type f -name picked.*.taxonomy -exec mv {} sequences-taxonomy-assignment.txt \\;")
#   system("find . -type f -name pecked.*.summary -exec mv {} classification-summary.tsv \\;")

# batch file 3: summary
#   summaryseqs.options <- paste("summary.seqs(fasta=picked.fasta, count=picked.count_table)")
#   documentCommand(summaryseqs.options)
#   write(summaryseqs.options,"summary.mth",append = FALSE)

# command 3
#   command3 <- paste(binary,"summary.mth","> log_raw.txt")

# run
#   system(command3)

# zip output fasta
#   system("gzip picked.fasta")

# Post process output
#   system("grep -A 10 Start log_raw.txt > picked-summary2.tsv")
# Remove one tab to get the column naming look nice:
#   system("sed 's/^		/	/' picked-summary2.tsv > picked-summary.tsv")
# }
