# TOOL single-cell-extract-information-v5.R: "Seurat v5 -Extract information from Seurat object" (This tool extracts data from Seurat object.)
# INPUT OPTIONAL seurat_obj.Robj: "Input" TYPE GENERIC (A Seurat object or an RDS file.)
# OUTPUT OPTIONAL slots.txt
# OUTPUT OPTIONAL gene_list.tsv
# OUTPUT OPTIONAL var_genes.tsv
# OUTPUT OPTIONAL commands.tsv
# OUTPUT OPTIONAL meta_data.tsv
# OUTPUT OPTIONAL spatial_image.pdf
# PARAMETER OPTIONAL gene_list: "Print list of the genes" TYPE [yes, no] DEFAULT no
# PARAMETER OPTIONAL var_genes_list: "Print highly variable genes" TYPE [yes, no] DEFAULT no
# PARAMETER OPTIONAL commands: "Print commands table" TYPE [yes, no] DEFAULT no
# PARAMETER OPTIONAL meta_data: "Print metadata stored in the object" TYPE [yes, no] DEFAULT no
# PARAMETER OPTIONAL spatial_image: "Print the tissue image stored in the object" TYPE [yes, no] DEFAULT no
# RUNTIME R-4.3.2-single-cell
# TOOLS_BIN ""

library(dplyr, quietly = TRUE)
library(Seurat, quietly = TRUE)
library(gplots)
library(ggplot2)
options(Seurat.object.assay.version = "v5")
source(file.path(chipster.common.lib.path, "tool-utils.R"))

# read input names
inputnames <- read_input_definitions()

# load Seurat object
if (grepl(".rds", inputnames$seurat_obj.Robj)) {
    seurat_obj <- readRDS("seurat_obj.Robj")
} else {
    load("seurat_obj.Robj")
}
# save integrated seurat object also as seurat_obj
if (exists("data.combined")) {
    seurat_obj <- data.combined
}

# Create a text file containing the different data slots in the Seurat object
sink("slots.txt")
print("Assays in the seurat object: ")
print(seurat_obj@assays)
print("Active assay in the object: ")
print(seurat_obj@active.assay)
print("Active cluster identity in the cluster: ")
print(head(seurat_obj@active.ident))
print("List of graph objects in the seurat object:")
print(seurat_obj@graphs)
print("List of neighbor objects in the seurat object:")
print(seurat_obj@neighbors)
print("List of dimensional reductions for this object:")
print(seurat_obj@reductions)
print("List of spatial image objects in this object:")
print(seurat_obj@images)
print("Name of the project:")
print(seurat_obj@project.name)
print("A list of miscellaneous information in the Seurat object:")
print(seurat_obj@misc)
print("Version of Seurat this object was built under:")
print(seurat_obj@version)
print("A list of miscellaneous data generated by other tools:")
print(seurat_obj@tools)
sink()

# Create a list of the genes and the highly variable features in the object
if (gene_list == "yes" || var_genes_list == "yes") {
    obj <- seurat_obj@assays
    # Object consists of multiple samples that have been integrated
    if (seurat_obj@active.assay == "integrated") {
        genes <- obj$integrated@data@Dimnames[1]
        var_genes <- data.frame(obj$integrated@var.features)
        # Object contains spatially transcriptomic data
    } else if (seurat_obj@active.assay == "Spatial") {
        genes <- obj$Spatial@counts@Dimnames[1]
        var_genes <- data.frame(obj$Spatial@var.features)
        # Object has been normalised with SCTransform
    } else if (seurat_obj@active.assay == "SCT") {
        genes <- obj$SCT@counts@Dimnames[1]
        var_genes <- data.frame(obj$SCT@var.features)
        # object hasn't been normalised or it has been normalised with the global scaling normalisation
    } else if (seurat_obj@active.assay == "RNA") {
        genes <- obj$RNA$counts@Dimnames[1]
        #var_genes <- data.frame(obj$RNA@var.features)
    } else {
        stop(paste("CHIPSTER-NOTE: ", "Object doesn't contain single-cell RNA-seq or spatially resolved data."))
    }
    # Write out the gene list
    if (gene_list == "yes") {
        names(genes)[1] <- "Genes"
        write.table(genes, file = "gene_list.tsv", sep = "\t", row.names = T, col.names = T, quote = F)
    }
    # Write out the highly variable genes
    if (var_genes_list == "yes") {
        names(var_genes)[1] <- "Highly variable genes"
        write.table(var_genes, file = "var_genes.tsv", sep = "\t", row.names = T, col.names = T, quote = F)
    }
}

# Write out commands table
if (commands == "yes") {
    sink("commands.tsv")
    print(seurat_obj@commands)
    sink()
}

# Write out metadata data frame
if (meta_data == "yes") {
    write.table(as.data.frame(as.matrix(seurat_obj@meta.data)), file = "meta_data.tsv", sep = "\t", row.names = T, col.names = T, quote = F)
}

# Print the tissue image
if (spatial_image == "yes") {
    if (length(seurat_obj@images) != 0) {
        pdf(file = "spatial_image.pdf", width = 9, height = 9)
        # print needed because otherwise would only print the last line inside the braces
        print(SpatialDimPlot(seurat_obj, pt.size.factor = 0))
        dev.off() # close the pdf
    } else {
        stop(paste("CHIPSTER-NOTE: ", "Object doesn't contain any spatial image information."))
    }
}





# EOF
