<html>

<body>

    <h1>New features for single cell RNA-seq tools </h1>

    <p>
        First of all, we would like to kindly thank our active users and course participants for the
        valuable feedback we got from them! These new features are based on those ideas and wishes.

        Many improvements to our manuals, plots and user instructions were made based on the feedback
        -we wonâ€™t go through all those here, but again would like to thank our active user base.
    </p>

    <p>
        3 minute <a href="https://youtu.be/S6v65bl7uN8">video explaining the new features</a> </b>:
        <br><br>

        <iframe width="560" height="315" src="https://www.youtube.com/embed/S6v65bl7uN8" title="YouTube video player"
            frameborder="0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowfullscreen></iframe>
    </p>

    <p>
        Smaller improvements include:
    <ul>
        <li> <b>The possibility to export PCA loadings from the
                <a href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-pca.html">PCA tool</a> </b>.
            If this parameter is turned on, an .txt file listing the top N genes associatied with the loadings is
            given as an output. Default for the N is 5, but user can tune this. </li>

        <li> <b>Separation of the
                <a href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-find-markers.html">expression
                    analysis</a> from the
                <a href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-clustering.html">clustering tool</a>.
            </b>
            Parameters remain the same, but separating the two tools allows faster and easier re-runs of
            expression analysis with different parameters.
        </li>

        <li><b>Plot cells in the the order of expression.</b> Some of the Seurat tools give as output FeaturePlots,
            where
            dots representing the cells
            are colored according to some feature of the data, like for example particular genes expression.
            In Seurat tools, by default, the cells are NOT ordered according to the expression levels,
            but there is a parameter that does that. We now included this parameter in the tools including this
            kind of plots, but it is "off" by default, according the original tool.
            It can be useful to turn this "on" if cells expressing given feature are getting buried.
        </li>

        <li> <b><a href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-modify-clusters.html">Tool for removing
                    clusters</a> from the data.</b>
            User can define either the clusters to remove, or list the clusters to keep. </li>

        <li> <b> <a href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-rename-clusters.html">Renaming of the
                    clusters</a>. </b>
            By default, the cluster names are just numbers,
            but by importing a table with cluster new cluster names it is now possible to rename them. </li>
        <li> <b>HDF5 input file format option available for <a
                    href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-setup.html">the Setup tool</a>.</b>
            The analysis can now begin from the HDF5 (.h5) files. See the new COVID data example session, and make sure
            you assign the files correctly in the parameters view. </li>


    </ul>

    <img src="-text-images/scRNAseq_named_clusters.png" width="400" height="400" />

    </p>

    <p>
        Bigger improvements include:
    <ul>
        <li> <b> <a href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-singler.html">SingleR cluster
                    annotation tool</a>. </b>
            With SingleR tool and CellDex annotation datasets, users can label their cell types based on the
            similarity to the reference datasets.
            As output, users get both higher level annotations as well as more detailed labels.
            Some plots for estimating the quality of these annotations are also provided.</li>

        <li> It is now also possible to use <b>data normalised with the
                <a href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-sctransform.html">SCTransform tool</a>
                in the multiple sample analysis</b>.
            Note that you need to let the tool know that this normalisation method was used!
            When using this option, the tool acknowledges the used parameters in scaling and integration steps.</li>

        <li> Last but not least, it is now possible to <b>
                <a href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-combine-multiple-samples.html">integrate
                    multiple samples</a> in the analysis</b>,
            instead of just the previous two.
            As before, it is important that the samples are named in the Setup tool. The tool
            "Seurat -Find conserved cluster markers and DE genes" recognises if there are
            more than two samples in the object, in which case it will compare each sample to all other samples, and
            print out separate DEG list file for each sample. There's also a new tool, <b>
                <a href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-diffexp-chosen-samples.html">Find
                    DE genes between chosen samples</a></b>, which allows comparing any samples or group of samples
            to other sample/samples. See the new COVID data example session to get an overview of analysis workflow of
            several samples.
        </li>

    </ul>

    <img src="-text-images/scRNAseq_3_samples.png" width="400" height="400" />

    </p>

    <p>
        We further encourage our users to communicate their ideas, wishes and notes on the function of
        these and other tools in Chipster -help us to help you!
    </p>



</body>

</html>