<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Bowtie for paired-end reads</h2>

<h3>Description</h3>

<p>
    This tool uses Bowtie to align paired-end reads to a publicly available reference genome. You need to
    supply the paired-end reads in FASTQ format.
</p>
<p> 
    Note that this tool uses publicly available reference genomes. If you would like us to add new reference 
    genomes to Chipster, please contact us. If you would like to align reads against your own datasets, 
    please use the tool <i>Bowtie for paired-end reads and own genome</i>.
</p>
<p>
    If you have multiple samples of paired-end reads, you can use the "Run for Each Sample" option to run multiple 
    alignment jobs at the same time. To define the files with paired-end reads for each sample, please use the
    "Define Samples" option. This option allows to define paired-end reads for multiple samples and enables the
    "Run for Each Sample" option such that the alignment tool can be run for multiple samples at the same time. 
    Please see the tutorial video <a href="https://www.youtube.com/watch?v=TIDS6aBHqqo">Define Samples to Enable 
    “Run for Each Sample” Option</a> for more information.
</p>

<h3>Parameters</h3>

<ul>
<li>Genome or transcriptome (human genome hg19, mouse genome mm9, rat genome rn4) [mouse genome mm9]</li>
<li>Number of mismatches allowed (0, 1, 2, 3) [2]</li>
<li>Consider mismatches only in the seed region (yes, no) [no]</li>
<li>Length of the seed region (5-50) [28]</li>
<li>Allowed total of mismatch qualities (10-100) [70]</li>
<li>Quality value format used (Sanger, Illumina GA v1.3 or later) [Sanger]</li>
<li>How many places is a read allowed to align to (1, 2, no limit) [no limit]</li>
<li>Minimum insert size (0-1000) [0]</li>
<li>Maximum insert size (50-1500) [250]</li>
<li>Upstream-downstream mate orientation (mate1 upstream of reverse complement of mate2 or vice versa, upstream mate1 reverse-complemented and mate2 forward-oriented) [mate1 upstream of reverse complement of mate2 or vice versa]</li>
<li>How many valid alignments are reported per read (1, 2, 3) [1]</li>
<li>Put multireads to a separate file (yes, no) [no]</li>
<li>Put unaligned reads to a separate file (yes, no) [no]</li>
</ul>

<h3>Details</h3>


<p><a href="http://bowtie-bio.sourceforge.net/index.shtml">Bowtie</a> aligns reads to a reference sequence such as genome or transcriptome. There are two modes: mismatches are considered either throughout the read (this is the so-called v-mode when running
Bowtie on command line), or only in the user-defined seed region in the high-quality left end of the read (n-mode in command-line Bowtie). In the latter case also base quality values at <i>all</i> mismatch positions are taken into account.</p> 

<p>Bowtie needs you to specify the minimum and maximum insert size for valid paired-end alignments. If e.g. 60 is specified for minimum insert size and a paired-end alignment consists of two 20-bp alignments in the appropriate orientation with a 20-bp gap between them, that alignment is considered valid.
A 19-bp gap would not be valid in that case. If 100 is specified for maximum insert size and a paired-end alignment consists of two 20-bp alignments in the proper orientation with a 60-bp gap between them, that alignment is considered valid. A 61-bp gap would not be valid in that case.</p> 

<p>
You also need to specify the upstream/downstream mate orientations for a valid paired-end alignment against the forward reference strand. 
The default setting "mate1 upstream of reverse complement of mate2 or vice versa" is appropriate for most Illumina datasets. 
With this setting candidate paired-end alignments are valid if mate1 appears upstream of the reverse complement of mate2 and the insert length constraints are met. 
Also, alignments are valid if mate2 appears upstream of the reverse complement of mate1 and all other constraints are met.
</p>
<p>
A paired-end alignment is reported as a pair of mate alignments, both on a separate line. The alignment for the mate that occurs closest to the beginning of the reference sequence (the "upstream" mate) 
is always printed before the alignment for the downstream mate.
</p>
<p>If a read has a higher number of reportable alignments than what was allowed, the user can request these multireads to be put in a separate fastq file for further inspection.
Similarly, the user can request that reads which don't align are put in a separate fastq file.</p>

<p>After running Bowtie, Chipster converts the alignment file to BAM format, and sorts and indexes it using the <a href="http://samtools.sourceforge.net/">SAMtools</a> package. </p>


<h3>Output</h3>

<p>This tool returns the alignment in BAM format and an index file for it (.bai). It also produces a log file,
which allows you to see what percentage of the reads align with the selected parameter settings. Optionally also fastq files are produced for the unaligned reads, or reads that align to multiple locations.
 </p>
 

<h3>Reference</h3>
Langmead et al. <a href="http://genomebiology.com/2009/10/3/R25">Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</a>  (2009) Genome Biology 10:R25 


</body>