<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Bowtie2 for paired-end reads and own genome</h2>

<h3>Description</h3>

<p>This tool uses <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a> software to align paired-end reads 
to a reference genome or sequence set. You need to supply the reads in two or more files containing the reads in the same 
order and a FASTA formatted reference sequence. Reads can be in either FASTA or FASTQ format, but all reads files need to be in 
the same format.

</p>
<p>
<p>After running Bowtie2, Chipster converts the alignment file to BAM format, and sorts and indexes it using the <a href="http://samtools.sourceforge.net/">SAMtools</a> package.</p>

<h3>Details</h3>
<p><b>If you have more than two reads files</b>, you will need to provide a list of filenames of the reads files for each direction (one file for read1 files, and another one for the read2 files) as a text 
file (<i>e.g.</i>R1files.txt and R2files.txt). These lists can be generated with the tool  <strong><a href="/manual/make_namelist.html" >Utilities / Make a list of file names</a> </strong>. The read 
pairs must be ordered identically in both lists. </p>
<p>
Select BOTH list files (R1files.txt and R2files.txt),  ALL reads files (8 of them in the example below) and the reference FASTA, and 
assign the list files correctly. When assigning the list files, they are automatically inactivated in the "reads" file list.
<img src="tophat-assign-list-files.jpg"> <br>
<br>
<img src="tophat-list-files-demo.png"> <br>
</p>

<h3>Mapping strategy</h3>
Bowtie2 can map the reads to the reference either by aligning the reads for they full length (end-to-end read alignment) or by using local alignments. 
Other possibility is to use local read alignment based mapping strategies. In this mode, Bowtie2 might "trim" or "clip" some read characters from one or both ends of the alignment if doing so maximizes the alignment score.
</p><p>
Bowtie2 uses heuristics for mapping the reads to the reference genome. This means that the reported mapping is not guaranteed to be the best mapping.
Several Bowtie2 parameters affect to the mapping accuracy. However higher accuracy (sensitivity) also requires more computing time.
In Chipster you can choose, for both end-to-end and local alignment strategy, the sensitivity level from a set of pre-defined parameter combinations that allow
you to tune the balance between the computing time and mapping sensitivity:
</p>
<table border=1>
<thead border=2><b><tr><th>Strategy</th><th>Corresponding command line options</th></tr></b></thead>
 <tr><td> Very fast</td><td> -D 5 -R 1 -N 0 -L 22 -i S,0,2.50</td></tr>
 <tr><td> Fast </td><td> -D 10 -R 2 -N 0 -L 22 -i S,0,2.50</td></tr> 
 <tr><td> Sensitive (default)</td><td> -D 15 -R 2 -N 0 -L 22 -i S,1,1.15</td></tr>
 <tr><td> Very sensitive </td><td> -D 20 -R 3 -N 0 -L 20 -i S,1,0.50</td></tr>
 <tr><td> Very fast local</td><td> --local -D 5 -R 1 -N 0 -L 25 -i S,1,2.00</td></tr>
 <tr><td> Fast local </td><td> --local -D 10 -R 2 -N 0 -L 22 -i S,1,1.75</td></tr> 
 <tr><td> Sensitive local </td><td> --local -D 15 -R 2 -N 0 -L 20 -i S,1,0.75</td></tr>
 <tr><td> Very sensitive local </td><td> --local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50</td></tr>
</table>
<p>
By default Chipster does the alignment using the end-to-end alignment method with "Sensitive" parameters.
</p>
<h3>Other parameters</h3>
<p>In addition to the mapping strategy, the Bowtie2 tool in Chipster allows users to modify many other mapping and output parameters too.
Please check the <a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">Bowtie2 manual</a> for more detailed description of these parameters.
</p>
<table border=1>
<thead border=2><b><tr><th>Parameter</th><th>Corresponding command line option</th><th>Description of the parameter</th></tr></b></thead>
<tr><td>Match bonus</td><td>--ma</td><td>Match bonus for a match in local alignment. Default value 2</td></tr>
<tr><td>Maximum penalty for mismatch </td><td>--mp</td><td>Maximum penalty for mismatch; lower quality = lower penalty. Default value 6</td></tr>
<tr><td>Penalty for non-ACGTs</td><td>--np</td><td>Sets penalty for positions where the read, reference, or both, contain an ambiguous character such as N. Default: 1.</td></tr> 
<tr><td>Gap opening penalty for the read</td><td>--rdg</td><td>Gap opening penalty for the reads. Default value: 5.</td></tr>
<tr><td>Gap extension penalty for the reads</td><td>--rdg</td><td>Gap extension penalty for the reads. Default value: 3.</td></tr>
<tr><td>Gap opening penalty for the reads</td><td>--rfg</td><td>Gap opening penalty for the reference. Default value: 5.</td></tr>
<tr><td>Gap extension penalty for the reads</td><td>--rfg</td><td>Gap extension penalty for the reference. Default value: 3.</td></tr>
<tr><td>Minimum insert length</td><td>--minins</td><td>Minimum insert length between the mate pairs. Default value: 0</td></tr>
<tr><td>Maximum insert length</td><td>--maxins</td><td>Maximum insert length between the mate pairs. Default value: 500</td></tr>
<tr><td>Order of mates to align</td><td>--fr, --rf or --ff</td><td>The orientation of the mate pairs. Default: forward/revrse</td></tr>
<tr><td>Suppress unpaired alignments</td><td>--no-mixed</td><td>By default, when bowtie2 cannot find a concordant or discordant alignment for a pair, it then tries to find alignments for the individual mates. This option disables that behavior.</td></tr>
<tr><td>Suppress discordant alignments</td><td>--no-discordant</td><td>By default, bowtie2 looks for discordant alignments if it cannot find any concordant alignments. A discordant alignment is an alignment where both mates align uniquely, but that does not satisfy the paired-end constraints. This option disables that behavior</td></tr>
<tr><td>Not concordant when mates extend past each other</td><td>--no-dovetail</td><td>If the mates "dovetail", that is if one mate alignment extends past the beginning of the other such that the wrong mate begins upstream, consider that to be concordant. Default: mates cannot dovetail in a concordant alignment. </td></tr> 
<tr><td>Not concordant when one mate alignment contains other</td><td>--no-contain</td><td>If one mate alignment contains the other, consider that to be non-concordant. Default: a mate can contain the other in a concordant alignment.</td></tr>
<tr><td>Not concordant when mates overlap at all</td><td>--no.overlap</td><td>If one mate alignment overlaps the other at all, consider that to be non-concordant. Default: mates can overlap in a concordant alignment.</td></tr>
<tr><td>Quality value format used</td><td>--phred33, --phred64 or --ignore-quals</td><td>Quality scale used in the fastq-file.</td></tr>
<tr><td>How many valid alignments are reported per read</td><td>none, -k or --all</td><td>By default, Bowtie2 reports only the best alignment of the read (based on the mapping quality\). Optionally, if there are several, equally good alignments, you can choose how many of them should be reported?</td></tr>
<tr><td>Put reads that did not align concordantly to a separate file</td><td>--un-conc</td><td>Write paired-end reads that fail to align concordantly to a fastq file.</td></tr>
<tr><td>Put unaligned reads to a separate file</td><td>--un</td><td>Would you like to store unaligned reads to a new fastq file? Note that also multireads will be added to this file, unless you asked them to be put to a separate file.</td></tr>
</table>


<h3>Output</h3>

<p>This tool returns the alignment in BAM format and an index file for it (.bai). It also produces a log file,
which allows you to see what percentage of the reads align with the selected parameter settings. Optionally also fastq files are produced for the unaligned reads, or reads that align discordantly.
 </p>
 

<h3>Reference</h3>
Langmead B, Salzberg S. <a href="http://www.nature.com/nmeth/journal/v9/n4/full/nmeth.1923.html">Fast gapped-read alignment with Bowtie 2. Nature Methods. 2012, 9:357-359.</a> 

</body>