<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Bowtie2 for single-end reads</h2>

    <h3>Description</h3>

    <p>
        This tool uses <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a> to align
        single-end reads to a publicly available reference genome. Single-end reads can either be in FASTA or 
        FASTQ format. 
    </p>
    <p> Note that this tool uses publicly available reference genomes. If you would like us to add new reference 
        genomes to Chipster, please contact us. If you would like to align reads against your own datasets, please
        use the tool Bowtie2 for single-end reads and own genome.
    </p>
    <p>
        If you have multiple samples of single-end reads, you can use the "Run for Each Sample" option to run multiple 
        single-end alignment jobs at the same time. To define the files with single-end reads for each sample, please 
        use the "Define Samples" option. This option allows to define the multiple different samples of single-end reads
        and enables the "Run for Each Sample" option such that the alignment tool can be run for multiple samples at the 
        same time. This option is similar to the option for paired-end alignment as explained in the tutorial video
         <a href="https://www.youtube.com/watch?v=TIDS6aBHqqo">Define Samples to Enable “Run for Each Sample” Option</a>, 
        but you need to select the "Single end" option in "Define Samples".
    </p>
    <p> After running Bowtie2, Chipster converts the alignment file to BAM format, and sorts and indexes it using the 
        <a href="http://samtools.sourceforge.net/">SAMtools</a> package.</p>

    <h3>Mapping strategy</h3>
    Bowtie2 can map the reads to the reference either by aligning the reads for they full length (end-to-end read
    alignment) or by using local alignments.
    Other possibility is to use local read alignment based mapping strategies. In this mode, Bowtie2 might "trim" or
    "clip" some read characters from one or both ends of the alignment if doing so maximizes the alignment score.
    </p>
    <p>
        Bowtie2 uses heuristics for mapping the reads to the reference genome. This means that the reported mapping is
        not guaranteed to be the best mapping.
        Several Bowtie2 parameters affect to the mapping accuracy. However higher accuracy (sensitivity) also requires
        more computing time.
        In Chipster you can choose, for both end-to-end and local alignment strategy, the sensitivity level from a set
        of pre-defined parameter combinations that allow
        you to tune the balance between the computing time and mapping sensitivity:


    <table border=1>
        <thead border=2><b>
                <tr>
                    <th>Strategy</th>
                    <th>Corresponding command line options</th>
                </tr>
            </b></thead>
        <tr>
            <td> Very fast</td>
            <td> -D 5 -R 1 -N 0 -L 22 -i S,0,2.50</td>
        </tr>
        <tr>
            <td> Fast </td>
            <td> -D 10 -R 2 -N 0 -L 22 -i S,0,2.50</td>
        </tr>
        <tr>
            <td> Sensitive (default)</td>
            <td> -D 15 -R 2 -N 0 -L 22 -i S,1,1.15</td>
        </tr>
        <tr>
            <td> Very sensitive </td>
            <td> -D 20 -R 3 -N 0 -L 20 -i S,1,0.50</td>
        </tr>
        <tr>
            <td> Very fast local</td>
            <td> --local -D 5 -R 1 -N 0 -L 25 -i S,1,2.00</td>
        </tr>
        <tr>
            <td> Fast local </td>
            <td> --local -D 10 -R 2 -N 0 -L 22 -i S,1,1.75</td>
        </tr>
        <tr>
            <td> Sensitive local </td>
            <td> --local -D 15 -R 2 -N 0 -L 20 -i S,1,0.75</td>
        </tr>
        <tr>
            <td> Very sensitive local </td>
            <td> --local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50</td>
        </tr>
    </table>



    By default Chipster does the alignment using the end-to-end alignment method with "Sensitive" parameters.
    </p>
    <h3>Other parameters</h3>
    <p>In addition to the mapping strategy, the Bowtie2 tool in Chipster allows users to modify many other mapping and
        output parameters too.
        Please check the <a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">Bowtie2 manual</a> for more
        detailed description of these parameters.
    </p>
    <table border=1>
        <thead border=2><b>
                <tr>
                    <th>Parameter</th>
                    <th>Corresponding command line option</th>
                    <th>Description of the parameter</th>
                </tr>
            </b></thead>
        <tr>
            <td>Create index file for BAM</td>
            <td> </td>
            <td>Create an index file for the alignment</td>
        </tr>
        <tr>
            <td>Quality value format used</td>
            <td>--phred33, --phred64 or --ignore-quals</td>
            <td>Quality scale used in the fastq-file.</td>
        </tr>
        <tr>
            <td>How many valid alignments are reported per read</td>
            <td>none, -k or --all</td>
            <td>By default, Bowtie2 reports only the best aligmnmet of the read (based on the mapping quality\).
                Optionally, if there are several, equally good alignments, you can choose how many of them should be
                reported?</td>
        </tr>
        <tr>
            <td>Put unaligned reads to a separate file</td>
            <td>--un</td>
            <td>Store unaligned reads to a new fastq file.</td>
        </tr>
        <tr>
            <td>Match bonus</td>
            <td>--ma</td>
            <td>Match bonus for a match in local alignment. Default value 2</td>
        </tr>
        <tr>
            <td>Maximum penalty for mismatch </td>
            <td>--mp</td>
            <td>Maximum penalty for mismatch; lower quality = lower penalty. Default value 6</td>
        </tr>
        <tr>
            <td>Penalty for non-ACGTs</td>
            <td>--np</td>
            <td>Sets penalty for positions where the read, reference, or both, contain an ambiguous character such as N.
                Default: 1.</td>
        </tr>
        <tr>
            <td>Gap opening penalty for the read</td>
            <td>--rdg</td>
            <td>Gap opening penalty for the reads. Default value: 5.</td>
        </tr>
        <tr>
            <td>Gap extension penalty for the reads</td>
            <td>--rdg</td>
            <td>Gap extension penalty for the reads. Default value: 3.</td>
        </tr>
        <tr>
            <td>Gap opening penalty for the reference</td>
            <td>--rfg</td>
            <td>Gap opening penalty for the reference. Default value: 5.</td>
        </tr>
        <tr>
            <td>Gap extension penalty for the reference</td>
            <td>--rfg</td>
            <td>Gap extension penalty for the reference. Default value: 3.</td>
        </tr>



    </table>


    <h3>Output</h3>

    <p>This tool returns the alignment in BAM format and an index file for it (.bai). It also produces a log file,
        which allows you to see what percentage of the reads align with the selected parameter settings. Optionally also
        a fastq file is produced for the unaligned reads.
    </p>

    <h3>Reference</h3>
    Langmead B, Salzberg S. <a href="http://www.nature.com/nmeth/journal/v9/n4/full/nmeth.1923.html">Fast gapped-read
        alignment with Bowtie 2. Nature Methods. 2012, 9:357-359.</a>



</body>