<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>BWA MEM for single or paired end reads and own genome</h2>

<h3>Description</h3>

<p>
    This tool uses BWA MEM to align single-end or paired-end reads to a provided reference genome. You need to supply 
    the single end or paired-end reads in FASTQ format. If you provide one reads file as input, single-end alignment
    is performed. If you provide paired-end reads files as input, paired-end alignment is performed. You also need
    to provide the reference genome as input either in FASTA format or as a tar package with a BWA MEM index. If 
    you would like to align reads against publicly available reference genomes supplied by Chipster, please use 
    the tool <i>BWA MEM for single or paired-end reads</i>.
</p>
<p>
    If you use a reference genome in FASTA format as input, the tool will output a .tar file with the BWA MEM index.
    If you would like to run the tool again for other samples of single-end or paired-end reads using the same 
    reference genome, you should use the .tar file as the reference input instead of the reference in FASTA format
    as this saves the time needed to generate the index.
</p>
<p>
    If you have multiple samples of paired-end reads, you can use the "Run for Each Sample" option to run multiple 
    alignment jobs at the same time. To define the files with paired-end reads for each sample, please use the
    "Define Samples" option. This option allows to define paired-end reads for multiple samples and enables the
    "Run for Each Sample" option such that the alignment tool can be run for multiple samples at the same time. 
    Please see the tutorial video <a href="https://www.youtube.com/watch?v=TIDS6aBHqqo">Define Samples to Enable 
    “Run for Each Sample” Option</a> for more information.
</p>

<h3>Parameters</h3>
<ul>
<li><b>Create an index file</b> Create index file for BAM.</li>
<li><b>Minimum seed length</b> Matches shorter than this will be missed 
when looking for maximal exact matches or MEMs in the first alignment phase.(BWA MEM option -k)</li>
<li><b>Maximum gap length</b> Gaps longer than this will not be found. Note also that scoring matrix and hit 
length affect the maximum gap length, in addition to this band width parameter.(BWA MEM option -w)</li>
<li><b>Match score</b> Score for a matching base.(BWA MEM option -A)</li>
<li><b>Mismatch penalty</b> Penalty for a mismatching base (BWA MEM option -B).</li>
<li><b>Gap opening penalty</b> Gap opening penalty (BWA MEM option -O).</li>
<li><b>Gap extension penalty</b> Gap extension penalty (BWA MEM option -E).</li>
<li><b>Penalty for end clipping</b> Penalty for 5'- and 3'-end clipping. When performing the Smith-Waterman 
extension of the seed alignments, BWA-MEM keeps track of the best score reaching the end of the read. 
If this score is larger than the best SW score minus the clipping penalty, clipping will not be applied (BWA MEM option -L).</li>
<li><b>Read group identifier</b> If you want to add the read group line into the BAM file, 
 you have to define read group identifier (DI:<i>value</i>).</li>
<li><b>Sample name for read group</b> The name of the sample sequenced in this read group (SM:<i>value</i>).</li>
<li><b>Platform for read group</b> With this setting you can platform or technology used to produce 
the read. Options: ILLUMINA, SOLID, LS454, HELICOS, PACBIO.(PT:<i>value</i>)</li>
<li><b>Library identifier for read group</b> DNA preparation library identifier. The Mark Duplicates 
tool uses this field to determine which read groups might contain molecular duplicates, in case the 
same DNA library was sequenced on multiple lanes.(LB:<i>value</i>).</li>
</ul>
<p>
<li>More information: <a href="http://bio-bwa.sourceforge.net/bwa.shtml">BWA manual</a></li>
</p>

<h3>Details</h3>
<p>
It is possible to give the tool more than one FASTQ file/file pair. The tool will run the alignment for each
file/file pair separately, and finally merge the resulting BAM files.
</p>
<p>
If you provide two FASTQ files, the tool will by default perform a paired-end alignment with them. It will try assign 
R1 and R2 reads correctly by file name. 
<p>

<p><b>If you have more than two FASTQ files</b> (or wish to perform single-end alignment for two files), you will 
need to provide a list of filenames of the FASTQ files; one list for single-end alignment, and two for 
paired-end alignment (one file for R1 files, and another one for the R2 files) as a text file 
(<i>e.g.</i>R1files.txt and R2files.txt). These lists can be generated with the tool  
<strong><a href="/manual/make_namelist.html" >Utilities / Make a list of file names</a> </strong>. 
The read pairs must be ordered identically in both lists. 
</p>
<p>
To run, Select the list file/files (R1files.txt and R2files.txt) and ALL FASTQ files, and assign the list files correctly. 
When assigning the list files, they are automatically inactivated in the "reads" file list.
</p>

<h3>Output</h3>

<p>As a result the tool returns a sorted and indexed BAM-formatted alignment.</p>
</body>