<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title>Chipster manual</title>
</head>

<body>

	<h2>Data analysis training materials</h2>


	<p>We run several data analysis courses every year in Finland and abroad. The courses at CSC
		are open for everybody, and you can also contact us to discuss options for hosting a course at your institute.
	</p>

	<p>You are welcome to use our course materials, please find them below.
		The materials include slides and exercises, and many courses have also lecture videos.
		The exercise data are available as example sessions on the Chipster server, and
		we also provide ready-made analysis sessions which you can use as a reference when doing exercises on your own.
	</p>

	<p>
		In order to do the exercises users from Finnish universities and research institutes can use their HAKA and
		VIRTU logins to access
		Chipster, and <b>training accounts</b> are available for others. <b>Please inform us about your upcoming
			Chipster courses</b>
		(number of participants, type of analysis jobs) so that we can add computing resources to the server if needed.
	</p>
	</p>
	<p> We run currently the following Chipster courses:
	</p>

	<ol>
		<li> <a href="#visium">Visium data analysis (spatially resolved transcriptomics)</a> </li>
		<li> <a href="#single-cell">Single-cell RNA-seq data analysis</a> </li>
		<li> <a href="#rna">Bulk RNA-seq data analysis</a> </li>
		<li> <a href="#quantseq">3' RNA-seq data analysis (QuantSeq)</a> </li>
		<li> <a href="#microbiome">Microbial community analysis </a> </li>
		<li> <a href="#virusdetect">Virus detection using small RNA-seq</a> </li>
		<li> <a href="#variant">Detection and annotation of genomic variants</a> </li>
		<li> <a href="#chip">ChIP-seq data analysis</a> </li>
		<li> <a href="#microarray">Microarray data analysis</a> </li>
	</ol>
	<hr>

	<h3><a name="visium">Visium data analysis (spatially resolved transcriptomics)</a></h3>

	<br>
	This course covers the analysis of 10X Visium data using Seurat v4 (note that the materials will be updated to v5
	during the summer 2024). We usually run it as a three-hour module after the single-cell RNA-seq course,
	because part of the theory is the same. You will learn how to

	<ul>
		<li>create a Seurat v4 object</li>
		<li>perform QC and filter out low quality spots (damaged tissue)</li>
		<li>normalize gene expression values and detect highly variable genes with SCTransform</li>
		<li>reduce dimensions with PCA using the highly variable genes</li>
		<li>use the PCs to cluster spots with graph based clustering</li>
		<li>visualize clusters with UMAP and overlay with the tissue image</li>
		<li>detect spatially variable genes</li>
		<li>visualise gene expression on the tissue image</li>
		<li>integrate samples using the mutual nearest neighbor approach (anchors)</li>
		<li>subset anatomical regions</li>
		<li>predict cell type composition in spots: Integrate with scRNA-seq data</li>
	</ul>
	Course material:
	<ul>
		<li><a href="https://chipster.github.io/training/visium/Visium_slides.pdf">slides</a> (2022)</li>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elBSUi3uGOemwT36u5sscXBS">lecture videos</a> (2022)
		<li><a href="https://chipster.github.io/training/visium/Visium_exercises.pdf">exercises</a> (2022).
			The data are available on Chipster server in the example sessions listed in the exercise sheet.</li>
	</ul>
	<br>
	<hr>

	<h3><a name="single-cell">Single-cell RNA-seq data analysis</a></h3>

	<br>
	<u>Course 1 (January 2024)</u>
	<br>This course covers the analysis of 10X data using Seurat v4 and SingleR, from digital gene expression matrix
	(DGE) to clustering cells and finding marker genes for those clusters.
	It also covers how to compare two samples and detect conserved cluster markers and differentially expressed genes in
	them. The course takes one day. You will learn how to
	<ul>
		<li>create Seurat v4 object</li>
		<li>perform QC and filter out low quality cells</li>
		<li>normalize expression values</li>
		<li>detect highly variable genes using VST</li>
		<li>scale data and regress out unwanted variability</li>
		<li>perform principle component analysis (PCA) and select PCs to be used for clustering</li>
		<li>cluster cells</li>
		<li>visualize clusters with UMAP and tSNE</li>
		<li>find marker genes for the clusters</li>
		<li>visualize marker genes with UMAP, violin plot and ridge plot</li>
		<li>annotate clusters with cell type information using SingleR</li>
		<li>run canonical correlation analysis (CCA) to identify common sources of variation between two datasets</li>
		<li>integrate samples using the mutual nearest neighbor approach (anchors)</li>
		<li>find conserved cluster marker genes for two samples</li>
		<li>find differentially expressed genes in a cluster between two samples</li>
		<li>visualize genes with cell type specific responses in two samples</li>
	</ul>
	Course material:
	<ul>
		<li><a
				href="https://chipster.github.io/training/single-cell/Single-cell RNA-seq data analysis_2024.pdf">slides</a>
			(2024)</li>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elALRmEKHrEL0OeRnJvpPW5q">lecture videos</a>
		<li><a href="https://chipster.github.io/training/single-cell/scRNA-seq_exercises_202401.pdf">exercises:
				Clustering
				10X Genomics data and Integrated analysis of two samples</a> (2024).
			The data are available on Chipster server in the example sessions listed in the exercise sheet.</li>
		<li>Note: excellent <a href="https://www.youtube.com/playlist?list=PLjiXAZO27elC_xnk7gVNM85I2IQl5BEJN">lecture
				videos covering scRNA-seq theory</a> from another course (May 2019)</li>
	</ul>
	<br>
	<u>Course 2 (March 2019)</u>
	<br>This course covers the preprocessing of DropSeq data from raw reads to a DGE. The course takes half a day. You
	will learn how to
	<ul>
		<li>check the quality of reads with FastQC</li>
		<li>tag reads with cell and molecular barcodes</li>
		<li>trim and filter reads</li>
		<li>align reads to the reference genome with HISAT2 and STAR</li>
		<li>tag reads with gene names</li>
		<li>estimate the number of usable cells by checking the inflection point</li>
		<li>detect bead synthesis errors</li>
		<li>create and filter DGE</li>
	</ul>

	Course material:
	<ul>
		<li><a href="https://chipster.github.io/training/single-cell/single_cell_RNAseq2019.pdf">slides</a> (March 2019)
		</li>
		<!--  <li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elD-tnVMX_oYTCF1mWOmB8Xo">lecture and exercise videos</a> on one sample analysis</li>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elD3R5bcssgM5JTcgyytKB9M">tutorial videos</a> on integrated analysis of two samples</li>-->
		<li><a href="https://chipster.github.io/training/single-cell/Single-cell-exercises-DropSeq-2018.pdf">exercises:
				Preprocessing DropSeq data</a>. The data is available on Chipster server in the example session listed
			in the exercise sheet.</li>
	</ul>

	<hr>

	<h3><a name="rna">Bulk RNA-seq data analysis</a></h3>

	This course introduces RNA-seq data analysis methods, tools and file formats. It covers all the steps from quality
	control and alignment to quantification and differential expression analysis, and also experimental design is
	discussed.
	The course takes two days. You will learn how to
	<ul>
		<li>check the quality of reads with FastQC and MultiQC</li>
		<li>remove bad quality data with Trimmomatic</li>
		<li>infer strandedness with RseQC</li>
		<li>align reads to the reference genome with HISAT2 and STAR</li>
		<li>perform alignment level quality control using RseQC and SAMtools</li>
		<li>quantify expression by counting reads per genes using HTSeq</li>
		<li>check the experiment level quality with PCA plots and heatmaps</li>
		<li>analyze differential expression with DESeq2 and edgeR</li>
		<li>take multiple factors (including batch effects) into account in differential expression analysis</li>
		<li>plot heatmaps</li>
		<li>annotate Ensembl identifiers</li>
		<li>perform pathway analysis with ConsensusPathDB</li>
	</ul>

	Course material (2023):
	<ul>
		<li><a href="https://chipster.github.io/training/rnaseq/Chipster.pdf">introductory slides about Chipster and the
				course</a></li>
		<li><a href="https://chipster.github.io/training/rnaseq/RNAseqDataAnalysis2023.pdf">slides on RNA-seq data
				analysis</a></li>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elBj3KYi7ACscgOxlNkNOxPc">lecture videos</a>
			(2019-2022)
		</li>
		<li>exercises (data is available on Chipster server in the example sessions listed in the exercise sheets)</li>
		<ol>
			<li><a
					href="https://chipster.github.io/training/rnaseq/exercises1_RNAseq_lung-lymphnode_comparison_2samples.pdf">Analysis
					starting from FASTQ files: QC, alignment, counting, combining, DE analysis. Human lung and lymph
					node sample (2 samples)</a>. These exercises cover the whole workflow.</li>

			<li><a
					href="https://chipster.github.io/training/rnaseq/exercises2_RNAseq_lung-lymphnode_comparison_10samples.pdf">Differential
					expression (DE) analysis starting from a count table</a>. Human lung and lymph node samples (10
				samples)</li>

			<li><a href="https://chipster.github.io/training/rnaseq/exercises3_RNAseq_withChipster_drosophila.pdf">DE
					analysis when there is a batch effect</a>. These exercises focus on differential expression analysis
				and how to take
				Drosophila pasilla gene knock-down experiment (7 samples).</li>

			<li><a href="https://chipster.github.io/training/rnaseq/exercises4_RNAseq_parathyroid.pdf">DE analysis when
					there are big differences between individuals</a>.
				Human adenocarcinoma samples before and after treatment (12 samples).</li>

			<li><a
					href="https://chipster.github.io/training/rnaseq/exercises7_Convert_counttable_to_ChipsterFormat.pdf">How
					to bring in a read count table and convert it to Chipster format</a>.</li>

		</ol>
		<li>additional material: We have collected <a
				href="https://chipster.2.rahtiapp.fi/manual/rna-seq-summary.html">instructions for RNA-seq</a> and
			written a <a
				href="https://www.crcpress.com/RNA-seq-Data-Analysis-A-Practical-Approach/Korpelainen-Tuimala-Somervuo-Huss-Wong/p/book/9781466595002">book</a>
			which might be useful for you when getting started.</li>
	</ul>

	<hr>

	<h3><a name="quantseq">3' RNA-seq data analysis (QuantSeq)</a></h3>

	This course module covers the preprocessing steps specific for QuantSeq FWD UMI 3â€™ mRNA-seq data. For the rest of
	the analysis,
	the typical RNA-seq data analysis steps described above in the RNA-seq course material can be used.

	QuantSeq reads contain 6 nt unique molecular identifiers (UMIs), which help to reduce amplification bias. This is
	particularly useful for low-input
	and FFPE samples. In data analysis the UMIs need to be removed and stored in the read name before alignment to
	genome. Reads which map to the same
	genomic location and have the same UMI are grouped together, and only one representative read is kept. In addition
	to dealing with UMIs,
	polyA read-through to adapters needs to be trimmed.

	The course takes half a day. You will learn how to
	<ul>
		<li>extract UMIs from reads using the extract tool of the UMI-Tools package</li>
		<li>trim reads using the BBDuk tool of the BBTools package</li>
		<li>check the quality of reads after preprocessing with MultiQC</li>
		<li>align reads to the reference genome with HISAT2 and STAR</li>
		<li>deduplicate the aligned QuantSeq reads using the dedup tool of the UMI-Tools package</li>
	</ul>

	Course material (2022):
	<ul>
		<li><a href="https://chipster.github.io/training/rnaseq/QuantSeqDataAnalysis2022.pdf">Slides</a></li>
		<li><a
				href="https://chipster.github.io/training/rnaseq/exercises5_QuantSeq_KawasakiDiseaseMouseModel_AnakinraTreatment_2samples.pdf">
				From raw QuantSeq reads to counts per genes</a>. Mouse model of Kawasaki Disease, treatment with IL-1
			receptor antagonist, Anakinra (2 samples).</li>
		<li><a
				href="https://chipster.github.io/training/rnaseq/exercises6_QuantSeq_KawasakiDiseaseMouseModel_AnakinraTreatment_6samples.pdf">
				From count table to differentially expressed genes</a>. Mouse model of Kawasaki Disease, treatment with
			IL-1 receptor antagonist, Anakinra (6 samples).</li>
		<li><a href="https://www.youtube.com/watch?v=NT96sQ_3X_0">lecture video</a>
		</li>
	</ul>
	<hr>

	<h3><a name="microbiome">Microbial community analysis</a></h3>

	We offer two courses for microbial community analysis: Course 1 covers amplicon sequence variant (ASV) analysis
	using DADA2, and Course 2
	covers OTU-based analysis using Mothur. Both courses contain a Phyloseq-based part where you learn to how to compare
	the structure of microbial communities
	using ordinations and multivariate statistics. The courses introduce the whole workflow from quality control and
	filtering to quantification and statistical analysis.
	There are separate slides and exercise sets for MiSeq and Ion Torrent 16S data. Also the analysis of ITS data is
	explained in the slides. The courses take two days.

	<br>
	<br>
	<u>Course 1: ASV-based analysis with DADA2 and Phyloseq (2023)</u>
	<br>
	<br>
	Topics covered:
	<ul>
		<li>Checking the quality of reads with MultiQC</li>
		<li>Preprocessing </li>
		<ul>
			<li>Remove primers/adapters with Cutadapt</li>
			<li>Filter and trim reads</li>
		</ul>
		<li>Sample/ASV inference</li>
		<li>Combine paired reads to contigs</li>
		<li>Make an ASV table and remove chimeras</li>
		<li>Assign taxonomy</li>
		<li>Make a phyloseq object</li>
		<li>Merge phenodata to the phyloseq object</li>
		<li>Data inspection and tidying using phyloseq</li>
		<ul>
			<li>Removing unwanted taxa</li>
			<li>Filtering singletons and doubletons</li>
			<li>Prevalence filtering</li>
		</ul>
		<li>Data transformation, ordination and conversion</li>
		<ul>
			<li>CLR, Hellinger and % relative abundance</li>
			<li>nMDS and db-RDA</li>
			<li>DESeq2 format conversion and variance-stabilising transformation</li>
		</ul>
		<li>Relative abundance plots</li>
		<li>Community comparisons using PERMANOVA, PERMDISP and DESeq2</li>
	</ul>

	Course material:
	<ul>
		<li><a href="https://youtube.com/playlist?list=PLjiXAZO27elApE_xX8idUFRbRXNj0m6GU&si=tGvdY8D0YS5Utl1e">lecture
				videos</a></li>
		<li><a href="https://chipster.github.io/training/microbiome-asv/microbiomeCommunityAnalysisASV.pdf">DADA2 part
				slides</a></li>
		<li><a href="https://csc-training.github.io/chipster-microbial/MiSeq/slides/d2.pdf">Phyloseq part slides</a>
		</li>

		<li>Example analysis session in Chipster: 08_microbial_community_analysis_16S_ASV_preprocessing_DADA2</li>
	</ul>
	<br>
	<u>Course 2: OTU-based analysis of Miseq and Ion Torrent data with Mothur, VSEARCH and Phyloseq (2023)</u>
	<br>
	<br>
	Topics covered:
	<ul>
		<li>Checking the quality of reads with MultiQC</li>
		<li>Preprocessing </li>
		<ul>
			<li>Ion Torrent data: Trimming primers with Cutadapt</li>
			<li>Ion Torrent data: Trimming reads based on quality with Trimmomatic</li>
			<li>MiSeq data: Combining paired reads to contigs with VSEARCH</li>
			<li>Combine FASTQ files and create a Mothur count file</li>
			<li>Screening sequences for length and ambiguous bases</li>
			<li>Removing identical sequences</li>
		</ul>
		<li>Alignment</li>
		<ul>
			<li>Aligning sequences to the Silva reference alignment</li>
			<li>Screening aligned sequences for alignment position and homopolymers</li>
			<li>Filtering alignment for empty columns and overhangs</li>
			<li>Removing new identical sequences</li>
		</ul>
		<li>Preclustering, removing chimeric sequences</li>
		<li>Taxonomic classification</li>
		<li>Data tidying and analysis</li>
		<li>Creating and importing phyloseq input files</li>
		<ul>
			<li>Including taxonomic assignments and OTU clustering</li>
		</ul>
		<li>Data inspection and tidying using phyloseq</li>
		<ul>
			<li>Removing unwanted taxa</li>
			<li>Filtering singletons and doubletons</li>
			<li>Prevalence filtering</li>
		</ul>
		<li>Data transformation, ordination and conversion</li>
		<ul>
			<li>CLR, Hellinger and % relative abundance</li>
			<li>nMDS and db-RDA</li>
			<li>DESeq2 format conversion and variance-stabilising transformation</li>
		</ul>
		<li>Relative abundance plots</li>
		<li>Community comparisons using PERMANOVA, PERMDISP and DESeq2</li>
	</ul>

	Course material:
	<ul>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elBjPaknlze6BkxebpEuj9KL">lecture videos</a></li>
		<li>Slides</li>
		<ul>
			<li>Part 1, preprocessing:
				<ul>
					<!--<li><a href="https://drive.google.com/file/d/1X5HHc_rWX12lFI7dJRmBCCWktEl-B7gi/view?usp=sharing">MiSeq</a></li>-->
					<li><a
							href="https://chipster.github.io/training/microbiome-otu-miseq/MicrobialCommunityAnalysis_day1.pdf">MiSeq</a>
					</li>
					<li><a
							href="https://chipster.github.io/training/microbiome-otu-iontorrent/microbiome-iontorrent-OTU_day1.pdf">Ion
							Torrent</a></li>
				</ul>
			</li>
			<li>Part 2, analysis:
				<ul>
					<li><a href="https://chipster.github.io/training/microbiome-otu-miseq/miseq-OTU_d2.pdf">MiSeq</a>
					</li>
					<li><a
							href="https://chipster.github.io/training/microbiome-otu-iontorrent/microbiome-iontorrent-OTU_day2.pdf">Ion
							Torrent</a></li>
				</ul>
			</li>
		</ul>
		<li>Exercises (the data is available on Chipster server in the example session listed in the part 1 exercise
			sheet)</li>
		<ul>
			<li>Miseq:
				<ul>
					<li><a href="https://csc-training.github.io/chipster-microbial/MiSeq/Exercises_day1.html">part 1
							exercises</a></li>
					<li><a href="https://csc-training.github.io/chipster-microbial/MiSeq/Exercises_day2.html">part 2
							exercises</a></li>
				</ul>
			</li>
			<li>Ion Torrent:
				<ul>
					<li><a href="https://csc-training.github.io/chipster-microbial/IonTorrent/Exercises_day1.html">part
							1 exercises</a></li>
					<li><a href="https://csc-training.github.io/chipster-microbial/IonTorrent/Exercises_day2.html">part
							2 exercises</a></li>
				</ul>
			</li>
		</ul>
	</ul>

	<hr>

	<h3><a name="virusdetect">Virus detection using small RNA-seq</a></h3>

	This course introduces the VirusDetect pipeline covering all the analysis steps and file formats. VirusDetect allows
	you to detect known viruses and identify new ones by sequencing small RNAs (siRNA) in host samples. siRNA sequences
	are assembled to contigs and compared to known virus sequences.
	The course takes about 5 hours. You will learn how to
	<ul>
		<li>run VirusDetect and interpret the result files</li>
		<li>subtract reads originating from the host genome</li>
		<li>set parameters for filtering the BLAST matches</li>
	</ul>

	Course material (2018):
	<ul>
		<li><a href="https://chipster.github.io/training/virusdetect/VirusDetect2017.pdf">slides</a></li>
		<li><a href="https://www.youtube.com/watch?v=Z-zxzeQxRkQ&index=2&list=PLjiXAZO27elCYwsme8hPqdw0zCTd8mXs6&t=0s">lecture
				video from a webinar</a></li>
		<li>exercises (data is available on Chipster server in the example sessions listed in the exercise sheets)</li>
		<ol>
			<li><a href="https://chipster.github.io/training/virusdetect/exercises1_VirusDetect_potato.pdf">Potato
					dataset</a> with different amounts of reads.</li>
			<li><a href="https://chipster.github.io/training/virusdetect/exercises2_VirusDetect_raspberry.pdf">Rasberry
					dataset</a> demonstrates how to provide the host genome.</li>
		</ol>
	</ul>

	<hr>

	<h3><a name="variant">Detection and annotation of genomic variants</a></h3>
	This course covers variant analysis from raw sequence reads to variant annotation, introducing the theory, analysis
	tools and file formats involved.
	The course takes one day. You will learn how to

	<ul>
		<li>check the quality control with FastQC and PRINSEQ</li>
		<li>remove bad quality data with Trimmomatic</li>
		<li>align reads to the reference genome with BWA</li>
		<li>perform alignment level quality control with SAMtools</li>
		<li>mark duplicates with Picard</li>
		<li>call and filter variants with Samtools, BCFtools and VCFtools. Note that the GATK variant calling platform
			was integrated in Chipster in 2019.</li>
		<li>annotate variants with Variant Effect Predictor (VEP)</li>
	</ul>

	Course material (2016):
	<ul>
		<li>slides</li>
		<ol>
			<li><a href="https://chipster.github.io/training/variants/IntroductionToVariantAnalysis2016.pdf">Variant
					analysis with Chipster</a></li>
			<li><a href="https://chipster.github.io/training/variants/Ensembl_Presentation.pdf">Ensembl variation data
					resources</a> (Variant Effect Predictor VEP described from slide 52 onwards) by Ben Moore</li>
			<li><a href="https://chipster.github.io/training/variants/HOPE_lectures.pdf">Effects of variants on protein
					structure</a> by Hanka Venselaar</li>
		</ol>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elBwER5r-HxKgFs05OUuAk05">lecture videos</a></li>
		<li><a href="https://chipster.github.io/training/variants/exercises_Variant_analysis_2016.pdf">exercises</a>.
			The
			data is available on Chipster server in the example session listed in the exercise sheet.</li>
	</ul>
	<hr>

	<h3><a name="chip">ChIP-seq data analysis</a></h3>
	This course covers ChIP-seq analysis from quality control and alignment to peak calling, motif detection, and
	pathway analysis. It introduces the theory, analysis tools and file formats involved.
	The course takes one day. You will learn how to

	<ul>
		<li>check the quality control with FastQC</li>
		<li>align reads to the reference genome with Bowtie</li>
		<li>call peaks with MACS2</li>
		<li>filter peaks</li>
		<li>retrieve nearby genes</li>
		<li>pathway analysis</li>
		<li>motif detection</li>
	</ul>

	Course material (2016):
	<ul>
		<li><a href="https://chipster.github.io/training/chipseq/ChIP-seq_Chipster2016.pdf">slides</a></li>
		<li><a
				href="https://chipster.github.io/training/chipseq/exercises_ChIPseq_withChipster_STAT1_2016.pdf">exercises</a>.
			The data is available on Chipster server in the example session listed in the exercise sheet.</li>
	</ul>
	<hr>

	<h3><a name="microarray">Microarray data analysis</a></h3>
	This course covers microarray data analysis from quality control and normalization to differential expression
	analysis, clustering and pathway analysis. It introduces the theory, analysis tools and file formats involved.
	The course takes one and half day. You will learn how to

	<ul>
		<li>perform quality control</li>
		<li>normalize data</li>
		<li>find differentially expressed genes and take batch effects into acount</li>
		<li>perform clustering</li>
		<li>visualize results in different ways</li>
		<li>pathway analysis</li>
	</ul>

	Course material (2019):
	<ul>
		<li><a href="https://chipster.github.io/training/microarray/microarray_2019.pdf">slides (including
				exercises)</a>.
			The data is available on Chipster server in the example sessions listed in the exercises.</li>
	</ul>
	<hr>



</body>

</html>