<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title>Chipster manual</title>
</head>

<body>

	<h2><a href="https://chipster.csc.fi/">Chipster</a> training courses</h2>


	<p>We run several bioinformatics courses on different topics every year in Finland and abroad. The courses at CSC
		are open for everybody,
		but you can also contact us to discuss options for hosting a course on your site.</p>

	<p>We make our course materials publicly available so that anyone can download them for their own use.
		The materials include slides and exercises, and many courses have also lecture videos.
		The exercise data are available as example sessions on the Chipster server, and
		we also provide ready-made analysis sessions which you can use as a reference when doing exercises on your own.
	</p>

	<p>
		We also provide <b>training accounts</b> to our Chipster server in Finland.
		Note, that users from Finnish universities and research institutes can use their HAKA and VIRTU logins to access
		Chipster, so no training account is needed.
		However, <b>we kindly ask you to inform us about upcoming Chipster courses</b>
		(number of participants, type of analysis jobs) so that we can add computing resources to the server if needed.
	</p>
	</p>

	<p>We run currently the following Chipster courses (follow the links for more info). CSC runs also <a
			href="https://research.csc.fi/bioscience-learning-materials">other bioinformatics courses</a>,
		you can find the upcoming courses in our <a href="https://www.csc.fi/en/web/training/events">course
			calendar</a>.</p>

	<ol>
		<li> <a href="#rna">RNA-seq data analysis</a> </li>
		<li> <a href="#single-cell">Single cell RNA-seq data analysis</a> </li>
		<li> <a href="#virusdetect">Virus detection using small RNA-seq</a> </li>
		<li> <a href="#16S">Microbial community analysis of amplicon sequencing data (16S)</a> </li>
		<li> <a href="#variant">Detection and annotation of genomic variants</a> </li>
		<li> <a href="#chip">ChIP-seq data analysis</a> </li>
		<li> <a href="#microarray">Microarray data analysis</a> </li>
	</ol>
	<hr>

	<h3><a name="rna">RNA-seq data analysis</a></h3>

	This course introduces RNA-seq data analysis methods, tools and file formats. It covers all the steps from quality
	control and alignment to quantification and differential expression analysis, and also experimental design is
	discussed.
	The course takes two days (or one long day if you omit exercise sheets 3 and 4). You will learn how to
	<ul>
		<li>check the quality of reads with FastQC and MultiQC</li>
		<li>remove bad quality data with Trimmomatic</li>
		<li>infer strandedness with RseQC</li>
		<li>align reads to the reference genome with HISAT2 and STAR</li>
		<li>perform alignment level quality control using RseQC and SAMtools</li>
		<li>quantify expression by counting reads per genes using HTSeq</li>
		<li>check the experiment level quality with PCA plots and heatmaps</li>
		<li>analyze differential expression with DESeq2 and edgeR</li>
		<li>take multiple factors (including batch effects) into account in differential expression analysis</li>
	</ul>

	Course material (2020):
	<ul>
		<li><a href="https://chipster.csc.fi/material/rna/RNAseqDataAnalysis2020.pdf">slides</a></li>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elBj3KYi7ACscgOxlNkNOxPc">lecture videos</a> (2019)
		</li>
		<li>exercises (data is available on Chipster server in the example sessions listed in the exercise sheets)</li>
		<ol>
			<li><a href="https://chipster.csc.fi/material/rna/exercises1_RNAseq_withChipster_ENCODE.pdf">ENCODE data
					with two samples</a>. These exercises cover the whole workflow.</li>
			<li><a href="https://chipster.csc.fi/material/rna/exercises2_RNAseq_withChipster_drosophila.pdf">drosophila
					pasilla dataset</a>. These exercises focus on differential expression analysis and how to take
				confounding factors into account.</li>
			<li><a href="https://chipster.csc.fi/material/rna/exercises3_parathyroid.pdf">parathyroid dataset</a>. As
				before, but even harder.</li>
			<li><a href="https://chipster.csc.fi/material/rna/exercises4_lung_lymphnode_comparison.pdf">lung and
					lymphnode comparison</a>. Test your new skills with minimal instructions!</li>
		</ol>
		<li>additional material: We have collected <a
				href="https://chipster.csc.fi/manual/rna-seq-summary.html">instructions for RNA-seq</a> and written a <a
				href="https://www.crcpress.com/RNA-seq-Data-Analysis-A-Practical-Approach/Korpelainen-Tuimala-Somervuo-Huss-Wong/p/book/9781466595002">book</a>
			which might be useful for you when getting started.</li>
	</ul>
	<hr>

	<h3><a name="single-cell">Single cell RNA-seq data analysis</a></h3>

	There are two courses for single cell RNA-seq data analysis. The more recent one focuses on the analysis of 10X data
	starting from the digital gene expression matrix (DGE),
	while the older one also covers the preprocessing of DropSeq data from raw reads to a DGE. Both courses show how to
	find sub-populations of cells using clustering with the Seurat tools, but the older course uses Seurat v2 instead of
	v3.
	You will also learn how to compare two samples and detect conserved cluster markers and differentially expressed
	genes in them. The course takes one day.
	<br>
	<br>
	<u>Course 1 (September 2019)</u>
	<br>You will learn how to
	<ul>
		<li>create Seurat v3 object</li>
		<li>perform QC and filter out low quality cells</li>
		<li>normalize expression values</li>
		<li>scale data and regress out unwanted variability</li>
		<li>detect highly variable genes using VST</li>
		<li>perform principle component analysis (PCA) and select PCs to be used for clustering</li>
		<li>cluster cells and find marker genes for a cluster</li>
		<li>visualize clusters with UMAP and tSNE</li>
		<li>run canonical correlation analysis (CCA) to identify common sources of variation between two datasets</li>
		<li>integrate samples using the mutual nearest neighbor approach (anchors)</li>
		<li>find conserved cluster marker genes for two samples</li>
		<li>find differentially expressed genes in a cluster between two samples</li>
		<li>visualize genes with cell type specific responses in two samples</li>
	</ul>
	Course material:
	<ul>
		<li><a href="https://chipster.csc.fi/material/single-cell/scrnaseq20190924.pdf">slides</a> (September 2019)</li>
		<li><a href="https://chipster.csc.fi/material/single-cell/scrnaseq_exercises20190924.pdf">exercises: Clustering
				10X Genomics data and Integrated analysis of two samples</a>.
			The data are available on Chipster server in the example sessions listed in the exercise sheet.</li>
		<li>Note: excellent <a href="https://www.youtube.com/playlist?list=PLjiXAZO27elC_xnk7gVNM85I2IQl5BEJN">lecture
				videos covering scRNA-seq theory</a> from another course (May 2019)</li>
	</ul>
	<br>
	<u>Course 2 (March 2019)</u>
	<br>You will learn how to
	<ul>
		<li>check the quality of reads with FastQC</li>
		<li>tag reads with cell and molecular barcodes</li>
		<li>trim and filter reads</li>
		<li>align reads to the reference genome with HISAT2 and STAR</li>
		<li>tag reads with gene names</li>
		<li>visualize aligned reads in genomic context using the Chipster genome browser</li>
		<li>estimate the number of usable cells by checking the inflection point</li>
		<li>detect bead synthesis errors</li>
		<li>create and filter DGE</li>
		<li>create Seurat v2 object</li>
		<li>regress out unwanted variability</li>
		<li>detect variable genes and perform principle component analysis</li>
		<li>cluster cells and find marker genes for a cluster</li>
		<li>run canonical correlation analysis (CCA) to identify common sources of variation between two datasets</li>
		<li>align two samples for integrated analysis</li>
		<li>find conserved cluster markers for two samples</li>
		<li>find differentially expressed genes in a cluster between two samples</li>
		<li>visualize genes with cell type specific responses in two samples</li>
	</ul>

	Course material:
	<ul>
		<li><a href="https://chipster.csc.fi/material/single-cell/single_cell_RNAseq2019.pdf">slides</a> (March 2019)
		</li>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elD-tnVMX_oYTCF1mWOmB8Xo">lecture and exercise
				videos</a> on one sample analysis</li>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elD3R5bcssgM5JTcgyytKB9M">tutorial videos</a> on
			integrated analysis of two samples</li>
		<li><a href="https://chipster.csc.fi/material/single-cell/Single-cell-exercises-DropSeq-2018.pdf">exercises 1:
				Preprocessing DropSeq data</a>. The data is available on Chipster server in the example session listed
			in the exercise sheet.</li>
		<li><a href="https://chipster.csc.fi/material/single-cell/Single-cell-exercises-Seurat-one-sample-2018.pdf">exercises
				2: Clustering 10X Genomics data</a>. The data is available on Chipster server in the example session
			listed in the exercise sheet.</li>
		<li><a href="https://chipster.csc.fi/material/single-cell/Single-cell-exercises-Seurat-two-samples-2018.pdf">exercises
				3: Integrated analysis of two samples using 10X Genomics data</a>. The data is available on Chipster
			server in the example session listed in the exercise sheet.</li>
	</ul>
	<hr>

	<h3><a name="virusdetect">Virus detection using small RNA-seq</a></h3>

	This course introduces the VirusDetect pipeline covering all the analysis steps and file formats. VirusDetect allows
	you to detect known viruses and identify news ones by sequencing small RNAs (siRNA) in host samples. siRNA sequences
	are assembled to contigs and compared to known virus sequences.
	The course takes about 5 hours. You will learn how to
	<ul>
		<li>run VirusDetect and interpret the result files</li>
		<li>subtract reads originating from the host genome</li>
		<li>set parameters for filtering the BLAST matches</li>
	</ul>

	Course material (2018):
	<ul>
		<li><a href="https://chipster.csc.fi/material/virusdetect/VirusDetect2017.pdf">slides</a></li>
		<li><a href="https://www.youtube.com/watch?v=Z-zxzeQxRkQ&index=2&list=PLjiXAZO27elCYwsme8hPqdw0zCTd8mXs6&t=0s">lecture
				video from a webinar</a></li>
		<li>exercises (data is available on Chipster server in the example sessions listed in the exercise sheets)</li>
		<ol>
			<li><a href="https://chipster.csc.fi/material/virusdetect/exercises1_VirusDetect_potato.pdf">Potato
					dataset</a> with different amounts of reads.</li>
			<li><a href="https://chipster.csc.fi/material/virusdetect/exercises2_VirusDetect_raspberry.pdf">Rasberry
					dataset</a> demonstrates how to provide the host genome.</li>
		</ol>
	</ul>
	<hr>

	<h3><a name="16S">Microbial community analysis of 16S amplicon sequencing data</a></h3>

	This course introduces microbial community analysis of (16S rRNA) amplicon sequencing data. It covers preprocessing,
	alignment to reference, taxonomic classification, and statistical analysis.
	The course takes one day. You will learn how to
	<ul>
		<li>check the quality of reads with FastQC and MultiQC</li>
		<li>remove bad quality data with Trimmomatic and Mothur</li>
		<li>remove redudancy, perform alignment and preclustering with Mothur</li>
		<li>do taxonomic classification with Mothur</li>
		<li>perform statistical analysis (based on R packages vegan, rich, biodiversityR, pegas, labdsv and DESeq2)</li>
	</ul>

	Course material (2020):
	<ul>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elA5SthFXdP7s2FuIgpx9kWF">lecture videos</a></li>
		<li>slides</li>
		<ul>
			<li><a href="https://chipster.csc.fi/material/16S/microbialCommunityAnalysis_slides_20200420.pdf">Community
					analysis with Chipster</a></li>
			<li><a href="https://chipster.csc.fi/material/16S/IntroductionToCommunityAnalysis_AnuMikkonen.pdf">Introduction
					to community analysis (2017)</a> by Anu Mikkonen</li>
			<!--  <li><a href="https://chipster.csc.fi/material/16S/StatisticalAnalysisOfMarkerGeneData_JarnoTuimala2017.pdf">Statistical analysis of marker gene data (2017)</a> by Jarno Tuimala</li>-->
		</ul>
		<li><a
				href="https://chipster.csc.fi/material/16S/microbialCommunityAnalysis_exercises_20200420.pdf">exercises</a>.
			The data is available on Chipster server in the example session listed in the exercise sheet.</li>
	</ul>
	<hr>

	<h3><a name="variant">Detection and annotation of genomic variants</a></h3>
	This course covers variant analysis from raw sequence reads to variant annotation, introducing the theory, analysis
	tools and file formats involved.
	The course takes one day. You will learn how to

	<ul>
		<li>check the quality control with FastQC and PRINSEQ</li>
		<li>remove bad quality data with Trimmomatic</li>
		<li>align reads to the reference genome with BWA</li>
		<li>perform alignment level quality control with SAMtools</li>
		<li>mark duplicates with Picard</li>
		<li>call and filter variants with Samtools, BCFtools and VCFtools. Note that the GATK variant calling platform
			will be integrated in Chipster in 2019.</li>
		<li>annotate variants with Variant Effect Predictor (VEP)</li>
		<li>visualize variants in genomic context with the Chipster genome browser</li>
	</ul>

	Course material (2016):
	<ul>
		<li>slides</li>
		<ol>
			<li><a href="https://chipster.csc.fi/material/variant/IntroductionToVariantAnalysis2016.pdf">Variant
					analysis with Chipster</a></li>
			<li><a href="https://chipster.csc.fi/material/variant/Ensembl_Presentation.pdf">Ensembl variation data
					resources</a> (Variant Effect Predictor VEP described from slide 52 onwards) by Ben Moore</li>
			<li><a href="https://chipster.csc.fi/material/variant/HOPE_lectures.pdf">Effects of variants on protein
					structure</a> by Hanka Venselaar</li>
		</ol>
		<li><a href="https://www.youtube.com/playlist?list=PLjiXAZO27elBwER5r-HxKgFs05OUuAk05">lecture videos</a></li>
		<li><a href="https://chipster.csc.fi/material/variant/exercises_Variant_analysis_2016.pdf">exercises</a>. The
			data is available on Chipster server in the example session listed in the exercise sheet.</li>
	</ul>
	<hr>

	<h3><a name="chip">ChIP-seq data analysis</a></h3>
	This course covers ChIP-seq analysis from quality control and alignment to peak calling, motif detection, and
	pathway analysis. It introduces the theory, analysis tools and file formats involved.
	The course takes one day. You will learn how to

	<ul>
		<li>check the quality control with FastQC</li>
		<li>align reads to the reference genome with Bowtie</li>
		<li>call peaks with MACS2</li>
		<li>filter peaks</li>
		<li>visualize reads and peaks in genomic context with the Chipster genome browser</li>
		<li>retrieve nearby genes</li>
		<li>pathway analysis</li>
		<li>motif detection</li>
	</ul>

	Course material (2016):
	<ul>
		<li><a href="https://chipster.csc.fi/material/chip/ChIP-seq_Chipster2016.pdf">slides</a></li>
		<li><a href="https://chipster.csc.fi/material/chip/exercises_ChIPseq_withChipster_STAT1_2016.pdf">exercises</a>.
			The data is available on Chipster server in the example session listed in the exercise sheet.</li>
	</ul>
	<hr>

	<h3><a name="microarray">Microarray data analysis</a></h3>
	This course covers microarray data analysis from quality control and normalization to differential expression
	analysis, clustering and pathway analysis. It introduces the theory, analysis tools and file formats involved.
	The course takes one and half day. You will learn how to

	<ul>
		<li>perform quality control</li>
		<li>normalize data</li>
		<li>find differentially expressed genes and take batch effects into acount</li>
		<li>perform clustering</li>
		<li>visualize results in different ways</li>
		<li>pathway analysis</li>
	</ul>

	Course material (2018):
	<ul>
		<li><a href="https://chipster.csc.fi/material/microarray/microarray_2018.pdf">slides (including exercises)</a>.
			The data is available on Chipster server in the example sessions listed in the exercises.</li>
	</ul>
	<hr>



</body>

</html>