<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Assign taxonomy</h2>

<h3>Description</h3>

<p>
Given a sequence table saved as a .Rda object, this tool assigns taxonomy to the sequence variants (ASVs) and outputs a taxonomy table.
This tool uses DADA2-formatted SILVA v.138.1 reference fastas for assignment if own reference files are not provided. If the species parameter is selected "yes",
this tool will use the addSpecies() function for species level assignments. 
Note that the DADA2 formatted v.138 Silva reference fastas are optimized for classification of bacteria and archae and aren't 
suitable for classifying Eukaryotes.
<h3>Parameters</h3>
<ul>
  <li>The minimum bootstrap confidence for assigning a taxonomic level. (0 - 100 ) [50]</li>
  <li>Do you want to assign the sequences to the species level? (yes, no) [yes]</li>
  <li>Combine the taxonomy and the sequence table. (yes, no) [yes]</li>
</ul>

<h3>Details</h3>
<p>
This tool uses the assignTaxonomy() and assignSpecies() functions to assign taxonomy to the sequence variants.
AssignTaxonomy() uses naive Bayesian classifier method of <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1950982/"> Wang et.al 2007</a> 
to assign taxonomy across multiple ranks (Kingdom to Genus). 
<br>
You can specify the minimum bootstrap confidence for assigning a taxonomic level during the assignTaxonomy() function. 
By default it's set to 50 which is the recommended threshold for sequences of 250nts. Otherwise, the recommended threshold is
80. 
<br>
The minimum bootsrap confidence of 50 means that at least 50 of the 100 bootstraps needs to return the same genus
for those sequences. 
<br>
<br>
If you want to use your own reference files, please provide the reference fasta as input.
<br>
Otherwise the DADA2 formatted Silva reference file v.138.1 (silva_nr99_v138.1_train_set.fa) will be used <a href="https://zenodo.org/record/4587955#.YuJ1TuxBwUE">Silva reference set</a> 
<br> 
For other DADA2 supported reference fastas check this link: <a href="https://benjjneb.github.io/dada2/training.html">Taxonomic reference data</a>.
<br>If you want to use those reference files for example the UNITE files for ITS taxonomic assignment,
 you can download the reference fastas from the page and give as input for this tool. 
<br>
<br>
As of the version 138 release, the SILVA databases and all the files provided for
download are freely available for academic and commercial use as long as SILVA
is credited as original author and a link to the full license is provided.
<a href="https://www.arb-silva.de/silva-license-information">Silva license</a>: 
<br>For more information on the v.138.1 release, check this: <a href="https://www.arb-silva.de/documentation/release-138.1/">Silva 138.1 release information</a>
<br>
<br>
If the assign species parameter is set to "yes", the addSpecies() command is used for the species level assignment. 
AddSpecies() function uses exact (100% identity) string matching against a reference database to assign genus species binomials. 
<br> So the query sequence is compared against all the reference sequences that had binomial genus-species nomenclature assigned, 
and the genus-species of all exact matches are recorded and returned if it is unambiguous
<br>
The 100% identity/exact matching seems to be the best way for species level assignment to 16S amplicon data.
<a href="https://academic.oup.com/bioinformatics/article/34/14/2371/4913809">Species level assignment</a>
<br>
The assignTaxonomy() function can also be used for species level assignment, if the reference file contains the Species level information.
<br>
Again, if you want to use your own reference file for the species level assignment, please provide it as input. 
<br> 
Otherwise the DADA2 formatted Silva v.138.1 reference file (silva_species_assignment_v138.1.fa) will be used. 
<br> 
<br> 
For more information about the taxonomy assignment check: <a href="https://benjjneb.github.io/dada2/assign.html">Taxonomic Assignment</a>
<br> 
For formatting custom databases/reference fastas check the end of this page: <a href="https://benjjneb.github.io/dada2/training.html">Taxonomic reference data</a>
<br> 
<br> 
To make the visualisation easier, the names of the ASV sequences are changed in the .tsv
tables to asv0, asv1 ... <br>.Rda objects have still the original row and columnames.

</p>
<h3>Output</h3>
<p>
The analysis output consists of the following:
<ul>
    <li>taxonomy-assignment-matrix.Rda: Taxonomy sequence table saved as a .RDA object. </li>
    <li>taxonomy_assignment.tsv: Taxonomy table where rownames are changed to asv0, asv1... and columnames are the taxonomies</li>
    <li>taxa_seqtab_combined.tsv: Taxonomy and sequence (ASV) table combined in a .tsv table where rownames are changed to asv0, asv1... and columnames are the taxonomies and the sample names.</li>
</ul>
</p>

<h3>References</h3>
<p>
    check assignTaxonomy() and addSpecies() commands from the dada2 manual: <br>
<a href=https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf> DADA2 manual </a> 
<br>
<br>
<a href="https://benjjneb.github.io/dada2/tutorial.html">DADA2 tutorial</a>
<br>
<a href="https://benjjneb.github.io/dada2/assign.html">Taxonomic Assignment</a>
<br>
By Robert c Edgar, 2018:
<a href="https://academic.oup.com/bioinformatics/article/34/14/2371/4913809">Updating the 97% identity threshold for 16S ribosomal RNA OTUs</a>
<br>
<a href="https://zenodo.org/record/4587955#.YuJ1TuxBwUE">Silva reference set</a> 
<br>
<a href="https://www.arb-silva.de/documentation/release-138.1/">Silva 138.1 release documentation</a>
<br>
by Callahan BJ, McMurdie PJ, Rosen MJ, Han AW, Johnson AJ, Holmes SP (2016). 13(7), 581-3. <br>
<a href="https://www.researchgate.net/publication/345890170_DADA2_High_resolution_sample_inference_from_amplicon_data"> DADA2: High resolution sample inference from Illumina amplicon data. Nature Methods </a> 
</p>
</body>