<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Combine paired reads to contigs with Dada2</h2>

<h3>Description</h3>

<p>
This tool combines the paired FASTQ files to sequence contigs with DADA2. The tool named "Sample inference" should be run before. 
</p>

<h3>Parameters</h3>
<ul>
  <li>The minimum length of the overlap required for merging the forward and reverse reads (0 - ) [12]</li>
  <li>The maximum number of mismatches allowed in the overlap region (0 - ) [0]</li>
</ul>

<h3>Details</h3>
<p>
Given a tar package of filtered FASTQ files and 2 lists of dada-class objects saved as .Rda objects (The output of the tool 
"Sample inderence"), this tool combines the paired FASTQ files to contigs.
<br>
The input tar package, containing the FASTQ files, should be the same as the input tar package
for the "Sample inference" tool.
<br>
At first, this tool merges the FASTQ files to contigs with the function mergePairs(). This function attemps to merge each denoised 
pair of forward and reverse reads, and rejects any pairs which don't sufficiently overlap or which contain too many
 mismatches in the overlap region (0 by default). You can change those parameters and make the merging less strict if many reads are being discarded.
<br>
By default, mergePairs() uses alignment parameters that heavily penalizes mismatches and gaps when
aligning the forward and reverse sequences.
<br>
MergePairs() function is based on the Needleman-Wunsch alignment. For more information check the references.
<br>
<br>
The output of this function is a data.frame or a list of data.frames(), and it can be used as the input for the tool
"Make a sequence table and remove chimeras".
<br>
<br>
If a majority of the reads failed to merge, you may need to revisit the truncLen parameter used in the filtering step
and make sure that the truncated reads span your amplicon.
<br>
</p>
<h3>Output</h3>
<p>
The analysis output consists of the following:
<ul>
    <li>contigs.Rda: An merger object which is the output of the mergePairs() function </li>
    <li>contigs_summary.tsv: Summary table to see how many reads failed to merge.</li>
</ul>
</p>

<h3>References</h3>
<p>
    check mergePairs() comamnd from the dada2 manual: <br>
<a href=https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf> DADA2 manual </a> 
<br>
<a href="https://benjjneb.github.io/dada2/tutorial.html">DADA2 tutorial</a>
<br>
by Callahan BJ, McMurdie PJ, Rosen MJ, Han AW, Johnson AJ, Holmes SP (2016). 13(7), 581-3. <br>
<a href="https://www.researchgate.net/publication/345890170_DADA2_High_resolution_sample_inference_from_amplicon_data"> DADA2: High resolution sample inference from Illumina amplicon data. Nature Methods </a> 
</p>
</body>