<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Filter sequences with DADA2</h2>

<h3>Description</h3>

<p>
Given a tar package of FASTQ files, this tool filters the input sequences which don't fullfill the user defined criteria.
Note that this tool is used for paired reads which are not merged to contigs yet. The filtered FASTQ files are put in one tar package
</p>

<h3>Parameters</h3>
<ul>
    <li>Truncate forward reads after this amount of bases ( 0- ) [0] </li>
    <li>Truncate reverse reads after this amount of bases (0 - ) [0]</li>
    <li>Discard input sequences with more than specified number of Ns (0 - ) [0]</li>
    <li>Discard forward sequences with more than the specified number of expected errors (0 - ) [empty]</li>
    <li>Discard forward sequences with more than the specified number of expected errors (0 - ) [empty]</li>
    <li>Truncuate reads at the first instance of a quality score less than or equal to the specified number (0 - ) [2]</li>
</ul>

<h3>Details</h3>
<p>
This tool takes a Tar package of FASTQ files (you can create one using the tool Utilities / Make a Tar package) and filters
them based on user specified parameters. The fastq files in the tar package can be compressed.
<br>
For one sample, one reverse and one forward fastq file should be provided in which case filtering is performed on the forward and 
reverse reads independently, and both reads must pass for the read pair no to be filtered.
<br>
<br>
Truncate reads after specified amount of bases is set to 0 by default which means that no truncation will happen.
<br>
Setting parameter "Truncuate reads at the first instance of a quality score less than or equal" to 0, will turn off this behaviour.
<br>
For example, if parameter TRUNCQ=2 is set, it means that it will truncate the read at the first nucleotide with a quality score of 2 (if there is one). 
The read is only then discarded if it is now shorter than truncLen.
<br>
Max Ns should be kept 0 if you want to continue with the asv/dada2 pipeline, because dada() doesn't allow any Ns.
<br>
By default max expected error is not used which will mean that the parameter is set to 'Inf'. After truncation, reads with more 
than the specified number of errors will be discarded. 
<br>Expected errors are calculated from the nominal definition of the quality score: sum(10^(-Q/10).
<br>
DADA2 tutorial suggest that max expected errors should be set to 2. If you want to speed up downstream computation try to tighten this parameter.
<br>
Running the tool "Read quality with MultiQC for many FASTQ files" can help to determine the values for the parameters.
<br>
If too few reads are passing the filter, you should try to change the parameters. Especially the maxee parameter on the reverse reads.
<br>
Remember when choosing the truncLen parameter you must maintain the overlap region for paired reads after truncation to merge them to contigs later.

<br>
Note that this tool is used for paired FASTQ files which are not merged to contigs yet. 
The FASTQ files/pair from one sample is recognized by the name of the files. Therefore they should be named so that the files from 
one sample/pair should have the same name but a different tag. And the forward files should be before the reverse files 
after the sort() function.<br>
The filtered FASTQ files will be named by splitting the filename with '_' and adding a 'F' or 'R' tag to the name.
<br>
For example the FASTQ files should be named like:
<br>
<br>
<pre> 
sample_xx_R1.fastq      sample_xx_R2.fastq  
</pre>      
<br>
<br>
This tool is based on the filterAndTrim() tool in the DADA2 package.
<p>

<h3>Output</h3>
</p>
The analysis output consists of the following:
<ul>
    <li>filtered.tar: A Tar package containing all the filtered FASTQ files. </li>
    <li>summary.txt: Summary statistics on how many reads where filtered from each sample.</li>
</ul>
</p>

<h3>References</h3>
<p>
    check filterAndTrim() from the dada2 manual: <br>
<a href=https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf> DADA2 manual </a> 
<br>
<a href="https://benjjneb.github.io/dada2/tutorial.html">DADA2 tutorial</a>
<br>
by Callahan BJ, McMurdie PJ, Rosen MJ, Han AW, Johnson AJ, Holmes SP (2016). 13(7), 581-3. <br>
<a href="https://www.researchgate.net/publication/345890170_DADA2_High_resolution_sample_inference_from_amplicon_data"> DADA2: High resolution sample inference from Illumina amplicon data. Nature Methods </a> 
</p>
</body>