<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Plot normalized counts for a gene</h2>

    <h3>Description</h3>
    Given a table of read counts for an experiment, this tool makes a plot of the normalized counts using the DESeq2
    Bioconductor package.

    <br>
    <br>
    <h3>Parameters</h3>

    <ul>
        <li> Gene name</li>
        <li> Show names in plot (yes, no) [no]</li>
        <li> How many of the top genes from the input file are plotted</li>
    </ul>

    <h3>Details</h3>


    <p>This tool takes as input a table of raw counts. The count table has to be associated with a phenodata file
        describing the experimental groups. These files are best created by the tool "Utilities / Define NGS
        experiment",
        which combines count files for different samples to one table, and creates a phenodata file for it.
    </p>
    <p>The tool normalizes counts by sequencing depth and adds a pseudocount of 1/2 to allow for log scale plotting.
        The counts are grouped by "group", and labeled with the sample name from the "description" column.
    </p>
    <p>You can give the input gene names either by:
    <ul>
        <li> typing them in the parameters field, (use comma (,) as a separator between the gene names), or</li>
        <li> as a .tsv file, where the gene names are the row names. Note,
            that by default maximum of 5 genes from the top of the list are plotted.
            You can tune this with the corresponding parameter, but please keep in mind that this makes the output pdf
            grow. </li>
    </ul>
    </p>


    <h3>Output</h3>

    <p>The plot of normalized counts.
    </p>

    <h3><br>References</h3>

    <p>This tool uses the DESeq2 package. Please read the following article for more detailed information:</p>

    <p> M Love, W Huber and S Anders: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4302049/">Moderated
            estimation of fold change and
            dispersion for RNA-Seq data with DESeq2.</a> Genome Biol. 2014 15:550</p>

</body>