<html lang="en">

<head>
	<meta http-equiv="Content-Type" content="text/html; charset=us-ascii">
</head>

<body>
	<h2>RNA-seq / Differential expression using edgeR for multivariate experiments</h2>
	<h3>Description</h3>
	Differential expression analysis for multifactor experiments using the generalized linear models (glm) -based
	statistical methods of the edgeR Bioconductor package.

	<br>
	<br>
	<h3>Parameters</h3>
	<ul>
		<li>Main effect 1 </li>
		<li>Main effect 2 </li>
		<li>Main effect 3 </li>
		<li>Treat main effect 1 as factor (yes, no) [yes] </li>
		<li>Treat main effect 2 as factor (yes, no) [yes] </li>
		<li>Treat main effect 3 as factor (yes, no) [yes] </li>
		<li>Include interactions in the model (yes, no, nested) [no] </li>
		<li>Apply TMM normalization (yes, no) [yes]</li>
		<li>Filter out genes which don't have counts in at least this many samples (1-10000) [1]</li>
		<li>Plot width (200-3200 [600]</li>
		<li>Plot height (200-3200) [600]</li>
		<li>Give as an output the design matrix (yes, no) [no]</li>
	</ul>
	<br>

	<h3>Details</h3>

	<p>This tool takes as input a table of raw counts from the different samples. The count file has to be associated
		with a phenodata file describing the experimental factors.
		You can generate these files using the
		<a href="https://chipster.2.rahtiapp.fi/manual/ngs-create-experiment.html"> Define NGS experiment</a>
		tool, which combines count files for different samples to one table and creates a phenodata file for it.
		We recommend marking the <i>control</i> group with 0 and the other groups with 1,2,3... in the phenodata
		columns.
		The group with the smaller number (or, if using letters, the alphabetically first letter) is used as the
		baseline.
		This means that in the results table, positive fold changes are upregulated and negative downregulated
		genes compared to the control group.
	</p>
	<p>
		You should set the filtering parameter to the number of samples in your smallest experimental group.
		Filtering will cause those genes which are not expressed or are expressed in very low levels (less than 5
		counts) to be ignored in statistical testing.
		These genes have little chance of showing significant evidence for differential expression, and removing them
		reduces the severity of multiple testing correction of p-values.
	</p>
	<p>
		Trimmed mean of M-values (TMM) normalization is used to calculate normalization factors in order to reduce RNA
		composition effect,
		which can arise for example when a small number of genes are very highly expressed in one experiment condition
		but not in the other.
	</p>
	<p>
		Dispersion is estimated using Cox-Reid profile-adjusted likelyhood (CR) method. Trended dispersions are
		estimated prior to estimating tagwise dispersions.
		After dispersion estimation, negative binomial generalized linear models are fitted to the data,
		after which the differentially expressed genes are determined using quasi-likelihood (QL) F-test.
		You can read more about these methods in the edgeR user guide linked below.
	</p>
	<p>Statistical analysis to identify differentially expressed genomic features (genes, miRNAs,...) is performed
		using a multivariate regression model. A maximum of three different variables, and their interactions
		can be specified for the model. It is highly recommended to always have at least two biological replicates for
		each experiment condition.
	</p>
	<p> Note, that unlike with basic edgeR or DESeq2, no filtering is done to the results table.
		Please notice that in the results table you will have all the comparisons between the different effects defined
		in the parameters -section.
		You can filter the table for example based on the adjusted p-values (FDR-columns) using the tool
		<a href="https://chipster.2.rahtiapp.fi/manual/ngs-filter-results-column.html">Utilities /
			Filter table by column value</a> .
	</p>

	<h4>Choosing the effects and understanding the results</h4>

	<p>
		What are then the questions you want to ask, what are the comparisons you want to make?
		We recommend reading
		<a href="http://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">
			Chapter 3.2.4 "Questions and Contrasts" in edgeR user guide </a> for a nice overview.
	</p>
	<p>
		EdgeR's glm mode used in this tool uses a <i>design matrix</i> to determine the sample groups and
		creates the <i>contrasts</i> based on those.
		In Chipster, this design matrix is generated from the <i>phenodata</i> file based on the parameters you choose
		(Main effects 1-3 and how to treat the interaction terms).
		The contrasts (comparisons) are always done and reported in the result table in the same way in Chipster.
		The same statistics (listed below in the Output section) columns are reported for all the comparisons.
		If you wish to see what the design matrix looks like, you can ask the tool to give it as an output.
	</p>
	<p>
		The first columns in the table, marked "Intercept", are measuring the baseline in the first treatment
		condition, and the comparisons after that are relative to this baseline.
		This makes sense, when the first condition ("group") is the control group.
		This assumption means, that Chipster doesn't automatically allow you to create all the different comparisons:
		for example, if you have 3 groups, A, B and C, Chipster can only give you the comparisons A vs B and A vs C
		(A is considered as the "control" here).
		To get the third comparison (B vs C), you need to run the tool again and change the order of the groups,
		so that for example B = control (= marked with the lowest number in the phenodata file). As a result
		from this comparison you would get: B vs A (which you already had, just the other way around) and B vs C.
		In the result table, you can see which comparison is in question from the column name:
		for example, Pvalue(group)2 is the (unadjusted) p-value for the comparison between the first (=control,
		baseline)
		and the second groups in the "group" column.
	</p>
	<p>
		Which effects should I include in the model? You can estimate this from looking at the PCA plot
		(try the tool <a href="https://chipster.2.rahtiapp.fi/manual/deseq2-pca-heatmap.html">PCA and heatmap of samples
			with
			DESeq2</a>).
		If you can see some of the effects there, it is advisable to add them to the model.
		Adding the batch effects to the model corrects for them, and your comparison of the actual groups of
		interest will be clearer. This effect is demonstrated in the example session
		<i>course_RNAseq_parathyroid_done</i>,
		where you are not able to find any differentially expressed genes before you add all the relevant effects to the
		model.
	</p>

	<h4>Including interactions</h4>

	<p> You can choose whether you want to include only the main effects ("Include interactions in the model" = "no"),
		the main effects and all the interaction terms ("yes"), or
		the main effect 1 and its interactions with main effects 2 and 3 ("nested", see more information about this
		option below).
		As a rule of thumb, you can add the interactions when you think they are relevant.
	</p>
	<p>
		Paired samples: for paired samples, do not include the interaction terms. See
		<a href="http://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">
			Chapter 3.4.1 "Paired samples" in edgeR user guide </a> for more information.
	</p>


	<h4>Example of the use of the "nested" option:</h4>

	<p>
		The "nested" option is needed when you want to do comparisons both between and within subjects
		(you might want to see the chapter 3.5. in <a
			href="http://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">
			EdgeR userguide</a>.
		If you are using "nested" option, note that you need to choose the effects accordingly: <br />
		Main effect 1 = the group in which the comparison is done between the subjects ("disease" in the example) <br />
		Main effect 2 = the group in which the comparison is done within the subjects ("treatment" in the example)
		<br />
		Main effect 3 = the subjects / column explaining the pairing of the samples ("patient" in the example)
	</p>
	<p>
		We have an experiment with 3 cancer patients and 3 controls. There are 2 samples from each patient, before and
		after the treatment.
		The normal-cancer comparison is thus done between individuals, while the untreated-treated comparison is done
		within individuals.
		In the phenodata-file we mark create corresponding groups: <br />
		disease = normal/cancer <br />
		treatment = untreated/treated <br />
		patient = the individuals <br />
		Note that you want to mark the "control"-situation (normal patient, untreated) with smaller index.<br />
		<img src="imageEdgeRNested1.jpg"><br />
		In the parameters-section, we choose the "nested" -option and the effects:<br />
		<img src="imageEdgeRNested2.jpg">
	</p>
	<p>
		In the resulting edger_glm.tsv -file there will be several columns corresponding to the different comparisons.
		In
		the example situation: <br />
		as.factor(disease) = comparison between the cancer patients and the controls <br />
		as.factor(disease)1:as.factor(treatment)2 = comparison between the untreated and treated control patients <br />
		as.factor(disease)2:as.factor(treatment)2 = comparison between the untreated and treated cancer patients <br />
	</p>



	<h3>Output</h3>
	<ul>
		<li>edger_glm.tsv: Table containing the statistical testing results, including fold change and p-values. </li>
		<ul>
			<li>logFC = log2 fold change between the groups. E.g. value 2 means that the expression has increased 4-fold
			</li>
			<li>logCPM = the average log2-counts-per-million</li>
			<li>LR = likelihood ratio statistics </li>
			<li>PValue = the two-sided p-value</li>
			<li>FDR = adjusted p-value</li>
		</ul>
		</li>
		<li>dispersion-edger-glm.pdf: Biological coefficient of variation plot.</li>
		<li>design-matrix.tsv: the design matrix created based on the phenodata file and parameter selections (optional
			output).</li>
	</ul>
	<br>
	<h3>References</h3>
	<p>This tool uses the edgeR package for statistical analysis. Please cite the following articles:</p>

	<p>MD Robinson, DJ McCarthy and GK Smyth. <a href="http://www.ncbi.nlm.nih.gov/pubmed/19910308">
			edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.
			Bioinformatics, 26 (1):139-40, Jan 2010.</a></p>

	<p>DJ McCarthy, Y Chen and GK Smyth. <a href="http://www.ncbi.nlm.nih.gov/pubmed/22287627">
			Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation.
			Nucleic Acids Res, 40 (10):4288-97, May 2012.</a></p>

	<p> More information about the functions can be found also in the
		<a href="http://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">
			EdgeR Users Guide</a>
	</p>
</body>

</html>