<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>HISAT2 for paired-end reads and own genome</h2>

    <h3>Description</h3>
        <p>
            This tool uses HISAT2 to align paired-end reads to a provided reference genome .You need to supply the
            paired-end reads in FASTQ format. The files can be compressed with gzip. You also need to provide the reference 
            genome as input either in FASTA format or as a tar package with a HISAT2 index. If you would like to align reads 
            against publicly available reference genomes supplied by Chipster, please use the tool
            <i>HISAT2 for paired-end reads</i>.
        </p>
        <p>
            If you use a reference genome in FASTA format as input, the tool will output a .tar file with the HISAT2 index.
            If you would like to run the tool again for other samples of paired-end reads using the same reference genome, 
            you should use the .tar file as the reference input instead of the reference in FASTA format as this saves the 
            time needed to generate the index.
        </p>
        <p>
        If you have multiple samples with paired-end reads, you can use the "Run for Each Sample" option to run multiple 
        alignment jobs at the same time. In order to do this, you need to define first which FASTQ files belong to which sample. You can do this using the 
        "Define Samples" functionality. Please see the tutorial video <a href="https://www.youtube.com/watch?v=TIDS6aBHqqo">Define Samples to Enable 
        “Run for Each Sample” Option</a> for more information.
        </p>

     <h3>Parameters</h3>
        <ul>
            <li>RNA-strandness (unstranded, FR, RF) [unstranded]</li>
            <li>How many hits is a read allowed to have (1-1000000) [5]</li>
            <li>Base quality encoding used (phred+33, phred+64) [phred+33]</li>
            <li>Minimum intron length [20]</li>
            <li>Maximum intron length [500000]</li>
            <li>Allow soft clipping (yes, no) [yes] </li>
            <li>Are long anchors required (yes, no) [no]</li>
        </ul>

     <h3>Details</h3>
        <p>
            HISAT2 (hierarchical indexing for spliced alignment of transcripts) is a highly efficient system for
            aligning reads from RNA sequencing experiments.
            HISAT uses an indexing scheme based on the Burrows-Wheeler transform and the Ferragina-Manzini (FM) index,
            employing two types of indexes for alignment:
            a whole-genome FM index to anchor each alignment and numerous local FM indexes for very rapid extensions of
            these alignments.
            HISAT's hierarchical index for the human genome contains 48,000 local FM indexes, each representing a
            genomic region of ~64,000 bp.
        </p>
        <p>
            HISAT2 searches by default for up to 5 distinct, primary alignments for each read, but you can change this
            number.
            Primary alignments mean alignments whose alignment score is equal or higher than any other alignments. It is
            possible that multiple distinct alignments have the same score.
            The alignment score for a paired-end alignment equals the sum of the alignment scores of the individual
            mates.
            Note that HISAT2 does not "find" alignments in any specific order, so for reads that have more than 5
            distinct, valid alignments,
            HISAT2 does not guarantee that the 5 alignments reported are the best possible in terms of alignment score.
            By default soft clipping is allowed, meaning that the ends of the read don't need to align if this increases
            the alignment score.
        </p>
        <p>
            If you are planning to do transcriptome assembly afterwards, you should set the long anchor parameter to
            yes.
            With this option, HISAT2 requires longer anchor lengths for de novo discovery of splice sites. This leads to
            fewer alignments with short anchors, which helps transcript assemblers improve significantly in computation
            and memory usage.
        </p>
        <p>
            If you use a FASTA format reference genome as input to the tool, the tool will produce a .tar file with the 
            HISAT2 index. If you would like to run the alignment for other samples using the same reference genome, you 
            can use this .tar file as the reference genome input instead of the reference genome in FASTA format. 
        </p>
        <p>
            After running HISAT2, Chipster indexes the BAM file using the <a
                href="http://samtools.sourceforge.net/">SAMtools</a> package.</p>
        <p>
        Note that if you have more than two FASTQ files <i>per ONE sample</i> (Illumina NextSeq produces 8
        FASTQ files per sample), you need to reduce them to two first: 
        Merge the R1 FASTQ files using the tool <a href="merge-fastq.html">"Utilities / Merge FASTQ"</a> and rename the resulting merged.fq.gz to samplename_R1.fq.gz. 
        Repeat the same for the R2 FASTQ files and rename the result samplename_R2.fq.gz
        </p>
        
        <h3>Output</h3>
        <p>
            This tool returns the following files:
            <ul>
                <li>*.bam: BAM file containing the alignments</li>
                <li>*.bam.bai: Index for the BAM file</li>
                <li>hisat.log: Summary of the alignment results</li>
                <li>*.hisat2.tar: HISAT2 index as a tar package (only if FASTA genome provided)</li>
            </ul>
        </p>

        <h3>Reference</h3>

        <p>This tool is based on the <a href="https://ccb.jhu.edu/software/hisat2/manual.shtml">HISAT2</a> package.
            Please cite the following article:
            Kim D, Langmead B and Salzberg SL. <a
                href="http://www.nature.com/nmeth/journal/vaop/ncurrent/full/nmeth.3317.html"> HISAT: a fast spliced
                aligner with low memory requirements</a> Nature Methods 2015.
        </p>

</body>
