<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>HISAT2 for single end reads with own genome</h2>

    <h3>Description</h3>

    <p>
        This tool uses HISAT2 to align single-end reads to a provided reference genome. You need to supply the single-end 
        reads in FASTQ format. The files can be compressed with gzip. You also need to provide the reference genome as 
        input either in FASTA format or as a tar package with a HISAT2 index. If you would like to align reads against 
        publicly available reference genomes supplied by Chipster, please use the tool <i>HISAT2 for single-end reads</i>.
    </p>
    <p>
        If you use a reference genome in FASTA format as input, the tool will output a .tar file with the HISAT2 index.
        If you would like to run the tool again for other samples of single-end reads using the same reference genome, 
        you should use the .tar file as the reference input instead of the reference in FASTA format as this saves the 
        time needed to generate the index.
    </p>
    <p>
        If you have multiple samples of single-end reads, you can use the "Run for Each Sample" option to run multiple 
        single-end alignment jobs at the same time. To define the files with single-end reads for each sample, please 
        use the "Define Samples" option. This option allows to define the multiple different samples of single-end reads
        and enables the "Run for Each Sample" option such that the alignment tool can be run for multiple samples at the 
        same time. This option is similar to the option for paired-end alignment as explained in the tutorial video
         <a href="https://www.youtube.com/watch?v=TIDS6aBHqqo">Define Samples to Enable “Run for Each Sample” Option</a>, 
        but you need to select the "Single end" option in "Define Samples".
    </p>

    <h3>Parameters</h3>
    <ul>
        <li>RNA-strandness (unstranded, F, R) [unstranded]</li>
        <li>How many hits is a read allowed to have (1-1000000) [5]</li>
        <li>Base quality encoding used (phred+33, phred+64) [phred+33]</li>
        <li>Minimum intron length [20]</li>
        <li>Maximum intron length [500000]</li>
        <li>Allow soft clipping (yes, no) [yes] </li>
        <li>Are long anchors required (yes, no) [no]</li>
    </ul>

    <h3>Details</h3>
    <p>
        HISAT2 (hierarchical indexing for spliced alignment of transcripts) is a highly efficient system for aligning
        reads from RNA sequencing experiments.
        HISAT2 uses an indexing scheme based on the Burrows-Wheeler transform and the Ferragina-Manzini (FM) index,
        employing two types of indexes for alignment:
        a whole-genome FM index to anchor each alignment and numerous local FM indexes for very rapid extensions of
        these alignments.
        HISAT's hierarchical index for the human genome contains 48,000 local FM indexes, each representing a genomic
        region of ~64,000 bp.
    </p>
    <p>
        HISAT2 searches by default for up to 5 distinct, primary alignments for each read, but you can change this
        number.
        Primary alignments mean alignments whose alignment score is equal or higher than any other alignments. It is
        possible that multiple distinct alignments have the same score.
        Note that HISAT2 does not "find" alignments in any specific order, so for reads that have more than 5 distinct,
        valid alignments,
        HISAT2 does not guarantee that the 5 alignments reported are the best possible in terms of alignment score.
        By default soft clipping is allowed, meaning that the ends of the read don't need to align if this increases the
        alignment score.
    </p>
    <p>
        If you are planning to do transcriptome assembly afterwards, you should set the long anchor parameter to yes.
        With this option, HISAT2 requires longer anchor lengths for de novo discovery of splice sites. This leads to
        fewer alignments with short anchors, which helps transcript assemblers improve significantly in computation and
        memory usage.
    </p>
    <p>
        If you use a FASTA format genome, the tool will produce a .tar file with the HISAT2 indexes. If you run the tool
        again with the same genome, you should use the .tar file as the genome input, as this saves the time needed
        to generate the indexes.
    </p>
    <p>After running HISAT2, Chipster indexes the BAM file using the <a
            href="http://samtools.sourceforge.net/">SAMtools</a> package. </p>

    <h3>Output</h3>
    <p>
        This tool returns the following files:
        <ul>
            <li>*.bam: BAM file containing the alignments</li>
            <li>*.bam.bai: Index for the BAM file</li>
            <li>hisat.log: Summary of the alignment results</li>
            <li>*.hisat2.tar: HISAT2 index as a tar package (only if FASTA genome provided)</li>
        </ul>
    </p>

    <h3>Reference</h3>

    <p>This tool is based on the <a href="https://ccb.jhu.edu/software/hisat2/manual.shtml">HISAT2</a> package. Please
        cite the following article:
        Kim D, Langmead B and Salzberg SL. <a
            href="http://www.nature.com/nmeth/journal/vaop/ncurrent/full/nmeth.3317.html"> HISAT: a fast spliced aligner
            with low memory requirements</a> Nature Methods 2015.
    </p>

</body>