<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Create HISAT2 index</h2>

    <h3>Description</h3>

    <p>This tool creates index files for the HISAT2 aligner. 
    </p>

    <h3>Parameters</h3>
    <p>
        None
    </p>

    <h3>Input</h3>
    <p>
        <ul>
            <li>input.fa: reference genome FASTA file</li>
            <li>input.gtf: reference genome GTF file</li>
        </ul>

        If the input.gtf is given, the tools tries to create an index with splice site and exon annotations, which requires huge amount of memory for larger genomes. Memory consumption is much lower if the GTF file is not given.
    </p>

    <h3>Output</h3>
    <p>
        This tool produces a Tar package that can be given to the tools "HISAT2 for single end reads and own genome" and "HISAT2 for paired end reads and own genome".
    </p>

    <h3>Reference</h3>

    <p>This tool is based on the <a href="https://daehwankimlab.github.io/hisat2/">HISAT2</a> package. Please
        cite the following article:
        Kim D, Langmead B and Salzberg SL. <a
            href="http://www.nature.com/nmeth/journal/vaop/ncurrent/full/nmeth.3317.html"> HISAT: a fast spliced aligner
            with low memory requirements</a> Nature Methods 2015.
    </p>

</body>
