<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title>Chipster manual</title>
</head>

<body>

	<h2>Chipster manual</h2>

	<p>Chipster is an easy-to-use analysis software for high-throughput data. It offers a comprehensive collection of
		analysis tools for
	<ul>
		<li>RNA-seq</li>
		<li>Single cell RNA-seq</li>
		<li>Small RNA-seq (miRNA and VirusDetect)</li>
		<li>Microbial community analysis (16S, 18S and ITS)</li>
		<li>ChIP-seq and DNase-seq</li>
		<li>Exome and whole genome seq (variants and copy number aberrations)</li>
		<li>Microarrays</li>
	</ul>

	<p>Here you can take a quick tour on Chipster by clicking the elements in the picture:</p>
	<iframe src="https://e-learn.csc.fi/mod/hvp/embed.php?id=748" width="1026" height="586" frameborder="0"
		allowfullscreen="allowfullscreen"></iframe>
	<script src="https://e-learn.csc.fi/mod/hvp/library/js/h5p-resizer.js" charset="UTF-8"></script>


	<p></p>
	<b>General</b>
	<ul>
		<li> <a href="https://www.youtube.com/watch?v=sAPjlTYqh-Q">Introductory video</a> for getting started with
			Chipster.</li>
		<li> <a href="courses.html">Course materials</a>: Lecture videos, slides and exercises on selected topics.</li>
		<li> <a href="https://www.youtube.com/channel/UCnL-Lx5gGlW01OkskZL7JEQ/playlists">Chipster YouTube channel</a>:
			Playlists covering different types of NGS data analysis (RNA-seq, single cell RNA-seq, 16S, etc).</li>
		<li> <a href="https://github.com/chipster/chipster/wiki/TechnicalManual">Technical manual for admins and
				developers</a>: How to install your own Chipster server and integrate new tools </li>
		<li> <a href="copyright.html">Copyright</a> </li>
		<!--<li> Java client instructions:
			<ul>
				<li> Analysis sessions: How to continue your analysis later and share it. <a href="sessions.html">Manual</a>, <a href="https://www.youtube.com/watch?v=KOeSRSBy-LU">video</a>. </li>
				<li> Phenodata editor: Describing your experimental setup. <a href="visualisation-phenodata.html">Manual</a>, <a href="https://www.youtube.com/watch?v=eptYuRs2qws">video</a>. </li>
				<li> Interactive visualizations. <a href="https://www.youtube.com/watch?v=PzZQVVD812w">Video</a>. </li>
				<li> <a href="import-ngs.html">Data import</a></li>
			</ul>
		</li>-->
	</ul>

	<b>NGS data analysis</b>
	<ul>
		<li>Tutorials
			<ul>
				<li> <a href="blog-single-cell-new-features.html">New scRNAseq tools and features 2022</a></li>
				<li> <a href="rna-seq-summary.html">RNA-seq data analysis for differential expression</a></li>
				<li> <a href="library-type-summary.html">Dealing with stranded (directional) RNA-seq data</a> </li>
				<li> <a href="aligners-comparison.html">Read counts and alignments -terminology and how to read to
						results</a> </li>
				<li> <a href="statistical-terms-explained.html">Statistical testing -terminology and how to read to
						results</a> </li>
				<li> RNA-seq data analysis step by step (<a
						href="https://www.youtube.com/playlist?list=PLjiXAZO27elBj3KYi7ACscgOxlNkNOxPc">videos</a>)</li>
				<li> Community analysis of 16S data step by step (<a
						href="https://www.youtube.com/playlist?list=PLjiXAZO27elA5SthFXdP7s2FuIgpx9kWF">videos</a>)</li>
				<li> <a href="cn-tutorial.pdf">CNA-seq tutorial</a> (pdf)</li>
			</ul>
		</li>
		<li> <a href="tools-ngs.html">Analysis tools for NGS data</a>
			<ul>
				<li> <a href="tools-ngs.html#quality">Quality control</a></li>
				<li> <a href="tools-ngs.html#preprocessing">Preprocessing</a></li>
				<li> <a href="tools-ngs.html#utilities">Utilities</a></li>
				<li> <a href="tools-ngs.html#regions">Matching sets of genomic regions</a></li>
				<li> <a href="tools-ngs.html#alignment">Alignment</a></li>
				<li> <a href="tools-ngs.html#variants">Variants</a></li>
				<li> <a href="tools-ngs.html#RNA">RNA-seq</a></li>
				<li> <a href="tools-ngs.html#miRNA">Small RNA-seq</a></li>
				<li> <a href="tools-ngs.html#singlecell">Single-cell RNA-seq</a></li>
				<li> <a href="tools-ngs.html#ChIP">ChIP- and DNase-seq</a></li>
				<li> <a href="tools-ngs.html#micropreproc">Microbial amplicon data preprocessing</a></li>
				<li> <a href="tools-ngs.html#microanalysis">Microbial amplicon data analyses</a></li>
				<li> <a href="tools-ngs.html#CNA">CNA-seq</a></li>
			</ul>
		</li>
	</ul>

	<b>Microarray data analysis</b>
	<ul>
		<li> <a href="supported-chips.html">Supported microarray types</a> </li>
		<li> <a href="https://chipster.csc.fi/case-studies/">Example analysis sessions</a> </li>
		<li> <a href="import.html">Data import</a> (old Java client) </li>
		<li> <a href="tutorials.html">Tutorials</a> (old Java client)</li>
		<li> Getting started with Chipster: <a href="basic-functionality.html">5 minute tour</a> (old Java client)</li>
		<li> <a href="tools.html">Analysis tools for microarray data</a> </li>
		<ul>
			<li> <a href="tools.html#normalization">Normalization</a> </li>
			<li> <a href="tools.html#quality">Quality control</a> </li>
			<li> <a href="tools.html#preprocessing">Preprocessing</a> </li>
			<li> <a href="tools.html#statistics">Statistics</a> </li>
			<li> <a href="tools.html#clustering">Clustering</a> </li>
			<li> <a href="tools.html#annotation">Annotation</a> </li>
			<li> <a href="tools.html#pathways">Pathways</a> </li>
			<li> <a href="tools.html#promoter">Promoter analysis</a> </li>
			<li> <a href="tools.html#visualization">Visualization</a> </li>
			<li> <a href="tools.html#utilities">Utilities</a> </li>
			<li> <a href="tools.html#acgh">aCGH</a> </li>
		</ul>
	</ul>


</body>

</html>