<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=us-ascii">
</head>
<body>
<h2>Differential coverage analysis with MEDIPS</h2>

<h3>Description</h3>

<p>Given BAM files, this tool performs differential coverage analysis of sequencing data from DNA immunoprecipitation experiments. It was originally designed for methylation data (MeDIP-seq), 
but it works also for other quantitative sequencing data such as ChIP-seq, MBD-seq, CMS-seq, etc. 
In addition to comparing samples such as treatment vs. control, it can be used also to analyze a single condition.
You can provide an optional BAM file for input DNA as well.</p>

<h3>Parameters</h3>
<ul>
<li>Chromosome names in my BAM file look like (1, chr1) [1]</li>
<li>Genome (human hg19, human GRCh38, mouse mm10, rat rn5) [human GRCh38]</li>
<li>Use only promoters (yes, no) [no] </li>
<li>Fragment length (400, 800, 1600, 2400) [800] </li>
<li>Resolution for coverage calculation (25, 50, 100, 200) [50] </li>
<li>Smoothing extension (200, 400, 800, 1200) [400] </li>
<li>Promoter length, upstream from the TSS (1000, 2000, 5000) [1000] </li>
<li>Promoter length, downstream from the TSS (250, 500, 750, 1000) [500] </li>
<li>Save a BED file (yes, no) [yes] </li>
</ul>

<h3>Details</h3>

<p>During the analysis, the read coverage over the genome is calculated in bins of 50 bp (you can change this value using the parameter "Resolution for coverage calculation").
 Since reads don't cover the whole length of the sonicated DNA fragments, the results are smoothed by extending the reads to both directions by 400 bp (you can change this value using the parameter "Smoothing extension").
 </p>
 <p>
The Fragment length parameter specifies the area around each genomic bin that is used for calculating the local CpG density, which is required for normalization. 
It should be set close to the estimated size of the DNA fragments generated after the amplification.</p>

<p>You can choose to calculate the enrichment of methylated CpG islands for promoter regions only, and select the length of the promoter region used. 
The default is 1000 bps upstream and 500 bps downstream from the transcription start site.</p>

<p>The differences in coverage between two conditions are tested using edgeR.</p>

<h3>Output</h3>
<p>Results are given in the file methylation.tsv. You can also save the relative methylation scores as BED files, which you can visualize in the Chipster genome browser. 
When performing a comparative analysis of treatment vs. control, also p-values and fold changes are returned.</p>

Several quality control plots are generated: 
<ul>
<li>The saturation analysis plot shows whether the amount of data is enough to cover the genome and generate reliable methylation profiles.</li>
<li>The coverage analysis plot shows the number of CpGs covered by the short reads, and the depth of the coverage.</li>
<li>The calibration curve plot shows the dependency of local CpG densities and signals derived from the data.</li>
</ul>

<h3>References</h3>
<p>For more information about the MEDIPS package, please see the
<a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/MEDIPS/inst/doc/MEDIPS.pdf">
manual on Bioconductor site</a>.</p>
<p>If you publish results acquired using this tool, please cite Lienhard M, Grimm C, Morkel M, Herwig R and Chavez L (2014):
 <a href="http://bioinformatics.oxfordjournals.org/content/early/2013/11/13/bioinformatics.btt650.full.pdf">MEDIPS: genome-wide differential coverage analysis of sequencing data derived from DNA enrichment experiments</a>. Bioinformatics, 30, pp. 284-286.

</body>
</html>
