<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Differential OTU abundance analysis using DESeq2</h2>

<h3>Description</h3>
<p>
Tests for OTU differential abundances using <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a>. A Benjamini-Hochberg multiple inference correction is applied to the data, with OTUs filtered using a cut-off value of 0.01. While the analysis can be performed using a phenodata variable comprising >2 sample groups, output files are restricted to comparisons between two groups. Produces a result table and plots OTU differential abundances at a user-specified level of biological organization. Note that the input file should not contain transformed OTU abundance data (raw counts are required, see Details for information on the input file).
</p>

<h3>Parameters</h3>
<ul>
<li>Number of sample groups under the phenodata variable used for the analysis (two or above two; the default is two)</li>
<li>Lower-level taxonomic classification used for differential abundance plotting (Class, Order, Family, Genus; the default is Class)</li>
<li>Higher-level taxonomic classification used for differential abundance plotting (Phylum, Class, Order, Family; the default is Phylum)</li>
<li>Groups 1 and 2 for DESeq2 contrasts (if there are >2 sample groups under the phenodata variable used for the analysis)</li>
</ul>
<p>
The lower-level taxonomic classification should correspond to a level of biological organization that is lower than the higher-level taxonomic grouping. OTUs classified as NA at the lower taxonomic level are filtered out to enable clearer plots.
</p>

<h3>Details</h3>
<p> 
The tool requires the following input files:
<ul>
<li>A <a href="http://bioconductor.org/packages/release/bioc/html/phyloseq.html">phyloseq</a> object converted into the <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> format (i.e. the file "deseq2.Rda" generated by the "Transform OTU counts" tool). This file should contain non-transformed data (raw counts) only.</li>
</ul>
The tool employs the following R functions and packages:
<ul>
<li>DESeq() and results() (<a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a>)</li>
<li>ggplot() (<a href="https://cran.r-project.org/web/packages/ggplot2/index.html">ggplot2</a>)</li>
</ul>
Technical notes:
<ul>
<li>The DESeq() function employs a parametric fit type (see <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> for details).</li>
</ul>
</p>

<h3>Output</h3>
<p>
The tool produces the following output files:</p>
<ul>
<li>A <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> results table in TSV format (deseq2_otutable.tsv)</li>
<li>Summary plot showing log2 fold changes (deseq2_otuplot.pdf)</li>
</ul>
</p>


<h3>References</h3>
<p>
<a href="https://joey711.github.io/phyloseq/">Phyloseq resources and GitHub repository</a>
</p>
<p>
<a href="https://joey711.github.io/phyloseq-extensions/DESeq2.html">DESeq2 with phyloseq</a>
</p>
<p>
Gloor GB, Macklaim JM, Pawlowsky-Glahn V, Egozcue JJ.
<a href="https://www.frontiersin.org/articles/10.3389/fmicb.2017.02224/full">Microbiome datasets are compositional: and this is not optional. Frontiers in Microbiology, 2017. 8:2224.</a> 
</p>
<p>
Love MI, Huber W, Anders S.
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology, 2014. 15:550.</a> 
</p>

</body>

</html>
