<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Classify ITS sequences to taxonomic units using UNITE</h2>

<h3>Description</h3>

<p>
Given a fasta file of ITS sequences and a count file generated by Mothur, 
this tool assigns sequences to taxonomic units (~ species).
</p>

<h3>Parameters</h3>

<ul>
<li>Reference (own) [own, UNITE files]</li>
<li>Number of iterations (10-1000) [100]</li>
<li>Remove taxon Chloroplast (yes/no) [yes]</li>
<li>Remove taxon Mitochondria (yes/no) [yes]</li>
<li>Remove taxon Archae (yes/no) [yes]</li>
<li>Remove taxon Bacteria (yes/no) [yes]</li>
<li>Remove taxon unknown (yes/no) [yes]</li>
<li>Remove other lineages (string) [empty]</li>
</ul>

<h3>Details</h3>

This tool assigns ITS sequences to taxonomic units using the Wang method and the <a href="https://unite.ut.ee/">UNITE reference</a> sequences and taxonomy. Users can also provide their own reference sequence and taxonomy file.
<p>
The UNITE database is centered on the eukaryotic nuclear ribosomal ITS region. 
All eukaryotic ITS sequences from the International Nucleotide Sequence Database (INSD) are clustered to approximately the species level (distance between species in steps of 0.5%).
<ul>
<li>The reference options marked with _s are from the UNITE download2 set, so they comprise of all quality filtered unclustered UNITE and INSD sequences for Fungi (887 399 for UNITE v8.2).</li> 
<li>The options without _s are from the UNITE download1 set, which contains sequences included in the UNITE species hypotheses (565 915 for UNITE v8.2).</li> 
<li>The dynamic options are based on varying clustering thresholds for different species hypothesis. These choices were made manually by experts of those particular lineages of fungi.</li>
</ul>
</p>
<p>
The bootstrap confidence score is calculated with 100 iterations by default. 
This low number of iterations is used as a default to make the tool faster, please consider using higher number (1000 for example).
</p>
<p>
With the <b>Remove</b> parameters you can remove unwanted organisms/sequences from the data. 
Some common options are provided as yes/no parameters. You can list others in the "Remove other lineages" 
text field. Use dots (.) instead of semicolons (;) in names. Use dash (-) to separate taxons. For example: 
Bacteria.Firmicutes.-Bacteria.Bacteroidetes. 
</p>
<p>
This tool is based on the <a href="http://www.mothur.org/wiki/Classify.seqs">classify.seqs</a> 
and <a href="https://mothur.org/wiki/Remove.lineage">remove.lineage</a> commands of the Mothur package.
</p>  

<h3>Output</h3>

<p>The analysis output consists of the following:</p>
<ul>
<li>sequences-taxonomy-assignment.txt: Taxonomy assignment for each sequence </li> 
<li>classification-summary.tsv: Taxonomic outline indicating the number of sequences that were found at each level </li>
<li>picked.fasta: Fasta file without the sequences from removed lineages
<li>picked.count_table: Counts without the sequences from removed lineages 

</ul>
</p>


<h3>References</h3>
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/">Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.</a> 
</p>

</body>

