<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Screen sequences for several criteria</h2>

<h3>Description</h3>

<p>
Given a fasta file and a group/count file, this tool screens out sequences that do not fulfill user-defined criteria. 
Note that this tool is used in twice during preprosessing (before and after aligning sequences to the reference) and different parameters apply: 
You should use the length parameters before alignment, and the alignment start and end parameters after the alignment.
</p>

<h3>Parameters</h3>
<ul>
<li>Maximum number of ambiguous bases </li>
<li>Maximum homopolymer length </li>
<li>Minimum length </li>
<li>Maximum length </li>
<li>Alignment start position (remove sequences which start after this position)</li>
<li>Alignment end position (remove sequences which end before this position)</li>
<li>Optimize by (empty, minlength, start, end) [empty]</li>
<li>Optimization criteria (0-100) </li>
</ul>

<h3>Details</h3>
<p>
Screen out sequences based on their length or alignment position. 
You can either choose the cut-off yourself (first four parameters), or use the optimization based on these.
If you use the optimization, remember to determine the criteria: 
for example 85 means that Mothur will optimize the cutoff for the above chosen quality so that 85% of the sequences are kept.
<br>
<br>
After you have combined paired reads to contigs with VSEARCH, you need to run tool "Combine FASTQ or FASTA files and make a group file"
to get a group and fasta file. 
<p>
This tool is based on the <a href="http://www.mothur.org/wiki/Screen.seqs">screen.seqs</a> command of the Mothur package.
</p>


<h3>Output</h3>

<p>The analysis output consists of the following:</p>
<ul>
<li>screened.fasta.gz: Compressed sequence file with the screened reads.</li>
<li>screened.groups: If a group file was provided as input, a group file containing the names and sample assignment of sequences after screening will be provided.</li>
<li>summary.screened.tsv: Summary statistics for the sequences.</li>
<li>screened.count: If a count file was provided as input, a count file containing the names and numbers of unique sequences after screening will be provided.</li>
</ul>



<h3>References</h3>
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/">Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.</a> 
</p>

</body>

