<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Trim primers and barcodes and filter reads</h2>

<h3>Description</h3>

<p>
Given a FASTA or FASTQ file and oligos file, this tool trims primers and barcodes and filters reads for several criteria, including uniqueness. You can give 
a QUAL file as well. 
</p>

<h3>Parameters</h3>
<ul>
<li>Use reverse complement (yes, no) [no]</li>
<li>Minimum average quality of sequence (0-40) []</li>
<li>Minimum average quality of window (Integer) []</li>
<li>Window size (Integer) []</li>
<li>Window step size (Integer) []</li>
<li>Maximum ambiguous bases (0-10) []</li>
<li>Maximum homopolymer length (0-50) []</li>
<li>Minimum sequence length (0-1000) []</li>
<li>Maximum sequence length (0-1000) []</li>
<li>Maximum differences to primer sequences (0-10) []</li>
<li>Maximum differences to barcode sequences (0-10) []</li>
</ul>

<h3>Details</h3>
<p>
This tool trims from the sequences the primers and barcodes provided in the <a href="https://www.mothur.org/wiki/Oligos_File">oligos file</a>,
  and filters out reads which don't fulfill the user-defined criteria.
Also duplicate reads are removed and a count_table is produced showing how many times each representative sequence occurs in the different samples. 
</p>
<p>
The oligos file contains sample names for each barcode. Please note that <b>Mothur does not accept hyphens in sample names</b>, 
  so you need to remove them from the oligos file if you have any.
</p>  
<p>
This tool is based on the <a href="http://www.mothur.org/wiki/Trim.seqs">Trim.seqs</a>, <a href="http://www.mothur.org/wiki/Unique.seqs">Unique.seqs</a> and <a href="https://mothur.org/wiki/summary.seqs/">summary.seqs</a> commands of the Mothur package.
</p>


<h3>Output</h3>

<p>The analysis output consists of the following files:</p>
<ul>
<li>trim.unique.fasta: Trimmed and filtered unique sequences.</li>
<li>trim.unique.count_table: Table showing how many times each representative sequence occurs in the different samples.</li>
<li>summary.trim.unique.tsv: Summary statistics for the sequences</li>
</ul>


<h3>References</h3>
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/">Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.</a> 
</p>

</body>

