<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Combine FASTQ files into one FASTA file and make a Mothur count file</h2>

<h3>Description</h3>

<p>
Combines FASTQ files of each sample to one FASTA file and creates a Mothur count file. Input should be a tar package containing FASTQ files. Input files can be gzipped.

</p>

<h3>Parameters</h3>
None

<h3>Details</h3>
<p>
For the Mothur-based tools in Chipster your samples needs to be in one FASTA file. 
In order to assign sequences to a specific sample, you need also a <a href="https://mothur.org/wiki/count_file/">Mothur count file</a>.
If you have single end data (e.g. Ion Torrent) and your samples are in separate FASTQ files or if you have paired end reads which you have merged 
into contigs with VSEARCH, you can convert the FASTQ files to FASTA and combine them with this tool 
which generates also the count file.  

</p>

<p>
This tool is based on the Mothur tools <a href="http://www.mothur.org/wiki/make.count/">make.count</a>, <a href="http://www.mothur.org/wiki/merge.files/">merge.files</a> and <a href="http://www.mothur.org/wiki/summary.seqs/">summary.seqs</a>.
</p>

<h3>Output</h3>

<p>The output consists of the following files:</p>
<ul>
<li>sequences.fasta.gz: Compressed FASTA file containing sequences from all the samples</li> 
<li>sequences.count_table: Mothur count file indicating what sequence belongs to what sample and how many times</li>
<li>sequences-summary.tsv: Summary statistics for the sequences </li> 
</ul>


<h3>References</h3>
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/">Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.</a> 
</p>

</body>