<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Remove chimeric sequences</h2>

<h3>Description</h3>

<p>
Given a fasta-formatted alignment file, removes chimeric sequences.
</p>

<h3>Parameters</h3>

<ul>
<li>Reference (16S rRNA Silva gold bacteria, none) [16S rRNA Silva gold bacteria]</li>
<li>Method (vsearch, uchime) [vsearch]</li>
<li>Dereplicate (false, true) [false]</li>
</ul>

<h3>Details</h3>
<p>Removes chimeric sequences using the UCHIME or the (faster) VSEARCH method. 
Chimeras are sequences formed from two or more biological sequences joined together. Amplicons with chimeric sequences can form during PCR when closely related sequences are amplified. 
The majority of chimeras are believed to arise from incomplete extension: During subsequent cycles of PCR, a partially extended strand can bind to a template derived from a different but similar sequence. 
This then acts as a primer that is extended to form a chimeric sequence. 
</p>

<p>
You can use the <a href="https://mothur.org/w/images/9/98/Silva.bacteria.zip">16S rRNA Silva gold bacterial set</a> as a reference, or you can detect chimeras de novo using the more abundant sequences in your samples as a reference. 
De novo detection uses the more abundant sequences from the same sample to check the query sequence. 
When a sequence is flagged as chimeric in one sample, it can be removed from only that sample by setting dereplicate = true, or from all samples by setting dereplicate = false. 
If you use the Silva reference, you should be aware of their <a href="https://www.arb-silva.de/silva-license-information">dual-use license</a>: Silva is free for academic users, but commercial users need to purchase a license.
</p>
<p>
This tool is based on the <a href="http://www.mothur.org/wiki/Chimera.vsearch">Chimera.vsearch</a>, <a href="http://www.mothur.org/wiki/Chimera.uchime">Chimera.uchime</a>, and <a href="http://www.mothur.org/wiki/Remove.seqs">Remove.seqs</a> commands of the Mothur package.
</p>

<h3>Output</h3>

<p>The analysis output consists of the following:</p>
<ul>
<li>chimeras-removed.fasta: Alignment with chimeras removed</li> 
<li>chimeras.removed.count_table: Count table with chimeras removed</li>
<li>chimeras-removed-summary.tsv: Summary statistics for the alignment </li> 
</ul>
</p>


<h3>References</h3>
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/">Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.</a> 
</p>

</body>

