<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Assess error rates</h2>

<h3>Description</h3>

<p>
You can measure the error rate of your sequences if you have co-sequenced a mock community. 
This tool pulls out the sequences that were from the mock sample, calculates the overall error rate,
 and removes the mock sample from the fasta, count_table and sequence-taxonomy-assignment files.
Error rate is defined as: sum of mismatches to reference / Sum of bases in query



</p>

<h3>Parameters</h3>
<ul>
<li>Mock sample name [Mock]</li>
<li>Are the reference sequences aligned (yes,no) [no]</li>
</ul>

<h3>Details</h3>
<p>
As input this tool takes a fasta file, a count_table file, a sequence-taxonomy-assignment file, and a mock reference file.
<br>
If your reference sequences are aligned, you should set the aligned parameter to yes.
<br>
This tool is based on the Mothur tools <a href="https://mothur.org/wiki/get.groups/">get.goups</a>, <a href="https://mothur.org/wiki/seq.error/">seq.error</a> and <a href="https://mothur.org/wiki/remove.groups/">remove.groups</a>.
You can read more about assessing error rates in the
<a href="https://mothur.org/wiki/miseq_sop/#assessing-error-rates">Mothur MiSeq SOP</a></p>

<h3>Output</h3>

<p>The analysis output consists of the following files:</p>
<ul>
<li>error-rate-summary.txt: Summary statictics of error rates and how many sequences were selected. </li>
<li>mock-removed.fasta.gz: Sequences with the mock sequences removed.</li>
<li>mock-removed.count_table: Count table with the mock sequences removed.</li>
<li>filtered-taxonomy-assignment.txt: Taxonomy-assignment with the mock sequences removed. </li>
<li>error-summary.tsv: Summary statistics of the remaining sequences. </li>

</ul>

<h3>References</h3>
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/">Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.</a> 
</p>


</body>

