<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Classify 16S or 18S sequences to taxonomic units using Silva</h2>

<h3>Description</h3>

<p>
Given a fasta file of aligned sequences and a count file generated by Mothur, 
this tool assigns the sequences to taxonomic units (~ species).
</p>

<h3>Parameters</h3>

<ul>
<li>Reference (silva.nr_v138.1, own) [silva.nr_v138.1]</li>
<li>Number of iterations (10-1000) [100]</li>
<!--<li>Remove taxon Chloroplast (yes/no) [yes]</li>
<li>Remove taxon Mitochondria (yes/no) [yes]</li>
<li>Remove taxon Archae (yes/no) [yes]</li>
<li>Remove taxon Eukaryota (yes/no) [yes]</li>
<li>Remove taxon unknown (yes/no) [yes]</li>
<li>Remove other lineages (string) [empty]</li>
-->
</ul>

<h3>Details</h3>

This tools assigns aligned sequences to taxonomic units using the Wang method and a reference. 
<ul>
<li>As of the 138 release, SILVA databases, taxonomy and all the files provided for download are freely available for academic and 
    commercial use: <a href="https://www.arb-silva.de/silva-license-information">SILVA-license</a></li>
<li>If you use your own reference, please provide the fasta and taxonomy file as input.</li>
</ul>
<p>
The bootstrap confidence score is calculated with 100 iterations by default. 
This low number of iterations is used as a default to make the tool faster, please consider using higher number (1000 for example).
</p>
<p>
If you would like to remove certain taxa or filter by a taxonomic group, you can use the following Phyloseq tools in Chipster 
(note that prior to this you need to run the tools <a href="mothur-phyloseq-inputs.html">Generate input files for phyloseq</a> 
and <a href="metabarcoding-import.html">Convert mothur files into phyloseq object</a>):
<ul>
<li><a href="metabarcoding-filter-taxgroup.html">Filter by taxonomic group</a> tidies a phyloseq object so that OTUs only from the desired taxonomic group 
    (bacteria, archaea, eukaryotes or fungi) are retained. Features with ambiguous phylum-level annotation (e.g. NA, unknown, uncharacterized) are removed. </li>
<li><a href="metabarcoding-filter-individual.html">Remove selected taxa</a> removes chloroplast and mitochondrial sequences from a phyloseq object, 
    and up to five user-specified taxa at the desired level of biological organization.</li>
</ul> 
<!--With the <b>Remove</b> parameters you can remove unwanted organisms/sequences from the data. 
Some common options are provided as yes/no parameters. You can list others in the "Remove other lineages" 
text field. Use dots (.) instead of semicolons (;) in names. Use dash (-) to separate taxons. For example: 
Bacteria.Firmicutes.-Bacteria.Bacteroidetes.
--> 
</p>
<p>
This tool is based on the <a href="http://www.mothur.org/wiki/Classify.seqs">classify.seqs</a> 
<!--and <a href="https://mothur.org/wiki/Remove.lineage">remove.lineage</a>--> command of the Mothur package.

</p>  

<h3>Output</h3>

<p>The analysis output consists of the following files:</p>
<ul>
<li>sequences-taxonomy-assignment.txt: Taxonomy assignment for each sequence </li> 
<li>classification-summary.tsv: Taxonomic outline indicating the number of sequences that were found at each level </li>
<!--<li>picked.fasta: Fasta file without the sequences from removed lineages
<li>picked.count_table: Counts without the sequences from removed lineages
--> 
</ul>
</p>


<h3>References</h3>
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/">Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.</a> 
</p>

</body>

