<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Merge aligned and unaligned BAM</h2>

<h3>Description</h3>

<p>This Picard tool merges alignment data from a standardly aligned BAM file with data in an unmapped BAM file.
 Before merging, the tool also sorts the aligned BAM file by queryname (QNAME) so that merging is possible.
This tool produces a new BAM file that includes all aligned and unaligned reads, while also retaining the additional read attributes from the unmapped BAM 
(attributes that are otherwise lost in the process of alignment). 
The purpose of this tool is to use information from the unmapped BAM to fix aligner output.  
In short, this tool can be used to add the cell and molecular barcode information, as well as other tags  
that may have been lost when aligning to the aligned BAM file.
</p>

<p>After merging, Drop-seq tool TagReadWithGeneExon adds a BAM tag GE to reads which overlap an exon of a gene. 
This tag contains the gene name, as reported in the GTF genome annotation file. 
For more details, please refer to the <a href="http://mccarrolllab.com/wp-content/uploads/2016/03/Drop-seqAlignmentCookbookv1.2Jan2016.pdf/">Drop-seq manual.</a>
</p>

 
<p>
Please note that this tool does not account for secondary alignments.</p>

<p>
Please ensure that you select the same reference that was used in the alignment, and please check that the input files are assigned correctly.</p>

<p>
For more details, please refer to the <a href="https://broadinstitute.github.io/picard/command-line-overview.html/">home page of Picard tools.</a>
</p>


<h3>Parameters</h3>
<ul>
		<li>Reference [Homo_sapiens.GRCh38] (Homo_sapiens.GRCh38, Mus_musculus.GRCm38)</li>
				<li>Organism [other]</li>
	</ul>
Chipster provides Ensembl-based GTF files for many organisms in the dropdown menu. If you would like to use your own GTF, please set the organism parameter to "other" and check that your GTF file has been correctly assigned in the parameter panel.
		

<p>
<h3>Output</h3>
</p>

<ul>
<li>merged.bam: Merged BAM file where reads overlapping an exon are tagged with a gene name.</li></ul>

</body>
