<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Seurat -Combine multiple samples </h2>

    <h3>Description</h3>

    <p>
        This tool can be used to combine/integrate/align multiple Seurat objects for later joined analysis.

    </p>

    <h3>Parameters</h3>
    <ul>
        <li>Normalisation method used previously [Global scaling normalization]</li>
        <li>Anchor identification method [CCA, RPCA]</li>
        <li>Number of CCs to use in the neighbor search [20]</li>
        <li>Number of PCs to use in the anchor weighting [20]</li>

    </ul>


    <h3>Details</h3>

    <p>
        As inputs, give the Seurat objects generated with <b>Seurat Setup tool</b> and
        preprocessed either with the <b>Seurat -Filter, normalize, regress and detect variable genes</b> or
        the <b>SCTransform</b> version of the same tool. Make sure you indicate which normalisation method
        was used (the first parameter). <br>
        NOTE: make sure you have assigned the sample or group name in the Setup tool (use short names like "CTRL1",
        "CTRL2"
        "TREAT1", "TREAT2").
        This name will be used later on to separate the samples.
    </p>

    <p>
        The first parameter asks about the normalisation step performed either
        in the tool: <b>Seurat -Filter, normalize, regress and detect variable genes</b>, in which case the
        <b>Global scaling normalization</b> (NormalizeData) method was used, or in tool: <b>Seurat -SCTransform: Filter,
            normalize, regress and detect
            variable genes</b>,
        in which case the user should change the parameter to <b>SCTransform</b>.
        In case SCTransform option is selected,
        an extra step (PrepSCTIntegration) is run,
        FindIntegrationAnchors() and IntegrateData() functions acknowledge the normalisation method,
        and the ScaleData step is skipped.

    </p>

    <p>
        The <b>"anchor identification method"</b> parameter determines which anchor identification method to use.
        By default, canonical correlation analysis CCA is used, but user can also decide to use the
        faster and more conservative reciprocal PCA approach.
        When determining anchors between any two datasets using RPCA, each dataset is projected into the
        others PCA space and the anchors are constrained by the same mutual neighborhood requirement.

        By identifying shared sources of variation between datasets, CCA is well-suited for identifying anchors
        when cell types are conserved, but there are very substantial differences in gene expression
        across experiments, i.e. when experimental
        conditions or disease states introduce very strong expression shifts,
        or when integrating datasets across modalities and species.
        However, CCA-based integration may also lead to overcorrection,
        especially when a large proportion of cells are non-overlapping across datasets.

        RPCA-based integration runs significantly faster, and also represents a more conservative
        approach where cells in different biological states are less likely to "align" after integration.
        RPCA is thus recommended during integrative analysis where:
    <ul>
        <li>A substantial fraction of cells in one dataset have no matching type in the other </li>
        <li>Datasets originate from the same platform (i.e. multiple lanes of 10x Genomics) </li>
        <li>There are a large number of datasets or cells to integrate </li>
    </ul>
    See the link to Seurat RPCA vignette below for more information!
    </p>

    <p>
        This tool performs data integration of multiple samples (in .Robj).
        First, Canonical correlation analysis (CCA) is performed to identify common sources of variation between
        the datasets.
        The tool then identifies anchors, gives scores (=weights) to these anchors
        and then uses these anchors to integrate the datasets.
    </p>

    <p>
        The third parameter tells which dimensions from the CCA/PCA to use for specifying the neighbor seach space.
        The neighbors are used to determine the anchors for the alignment.
    </p>

    <p>
        The fourth parameter is the number of PCs to use in the anchor weighting procedure.
        The anchors and their weights are used to compute the correction vectors,
        which are used for correction, thus allowing the datasets to be integrated.
    </p>

    <h3>Further information</h3>
    <p>
        For more details, please check the <a href="https://satijalab.org/seurat/immune_alignment.html">Seurat tutorials
            for multiple sample analysis</a>.
    </p>

    <p>
        For more details on use of reciprocal PCA, please check the <a
            href="https://satijalab.org/seurat/articles/integration_rpca.html">Seurat vignette
            for "Fast integration using reciprocal PCA (RPCA)"</a>.
    </p>
    <p>
        For more detailed description of the integration process, read "Comprehensive integration
        of single cell data <a href="https://www.biorxiv.org/content/10.1101/460147v1">the paper</a>
        by Rahul Satija et al.
    </p>

    <h3>Output</h3>

    <ul>

        <li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. Not viewable in
            Chipster. </li>
    </ul>



</body>