<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Scater - QC </h2>

    <h3>Description</h3>

    <p>
        Please provide the Seurat object as an input. This tool then generates a PDF with several plots that can be used to estimate
        the quality of the data and compare samples. The plots may vary depending
        on which values are available in the Seurat input. Cell-cycle scores (after the "Filter, normalize, regress and detect variable genes" -tool), if available, may be used when plotting.
       
        This tool transforms the Seurat object into Scater format and computes several QC metrics.
    </p>

    <h3>Parameters</h3>
     None

    <h3>Details</h3>

    <p>
        The Seurat object is converted into a Scater object (SingleCellExperiment, or SCE).
        After this, several QC plots are printed in the output PDF.
    </p>


    <h4>Plot 1: Most expressed features</h4>
    <p>
        This plot shows the features (genes) with the (50) highest average expression across all cells,
        along with their expression in each individual cell.
        You can check for the top 50 most expressed genes, and potentially detect
        overabundant genes.
        For example, you can see if there are mitochondrial genes (MT-) that you may want to
        remove from your analysis.
        The x-axis shows the percentage of total counts, and the y-axis lists the top genes.
        Each tick mark represents a cell.
        The color of the tick mark represents the total number of genes in that cell.
        The dot in each line shows the average epression.
    </p>

    <h4>Plot 2: Cumulative expression </h4>
    <p>
        After identifying the 1000 most highly expressed genes across all cells, this plot then shows the proportion of a specific gene out of the total gene counts within a given cell.
        The x-axis shows the counts for each gene in a given cell, and the y-axis is the cumulative proportion. Each cell is presented as a line.
        This plot can be used to identify differences in expression distributions between samples.
    </p>


    <h4>Plot 3: Comparing statistics</h4>
    <p>
        Several QC values plotted against each other.
        You can read more in the <a
            href="https://www.rdocumentation.org/packages/scater/versions/1.0.4/topics/calculateQCMetrics">Scater
            documentation for calculateQCmMtrics</a>
        which defines each QC value.
        <br>For example:
        <br>pct_counts_top_50_features: What percentage of the total counts is accounted for by the 50 highest-count
        genes?
        <br>pct_counts_feature_controls: Percentage of the total counts for cells found in control
        genes:
        mt.genes (mitochondrial genes, starting with "MT") are used as controls here.
    </p>


    <h4>Plots 4 and 5: Principal component (PC) comparisons</h4>
    <p>
        Here, we can see the first 4 PCs plotted against one another.
        The percentage of the data's variability explained by each PC is also reported.
        The dots representing the cells are colored based on mitochondrial percentage and total number of genes.
    </p>

    <h4>Plots 6-9: Explanatory factors</h4>
    <p>
        We can investigate the relative importance of different explanatory factors with these plots.
        In the first two plots, a density plot is created for each variable,
        showing the distribution of R-squared across all genes.
        Scater calculates R-squared for a factor/variable by fitting a line for each gene
        against the factor/variable.
        R-squared is the proportion of variance for the dependent variable explained by an independent variable.

        We can also identify PCs that strongly correlate to certain QC factor or meta-data values, which can be observed in the second set of plots.
        Some selected factors are plotted, as well as the top 10 factors.
        If cell-cycle scores are present in the provided Seurat object, the
        values for the phase, G2M score and S score will also be plotted.
    </p>


    <p>
        For more details, please check the <a
            href="https://bioconductor.org/packages/release/bioc/html/scater.html">Scater package in Bioconductor</a>
        and the Scater article: McCarthy DJ, Campbell KR, Lun ATL, Willis QF (2017).
        “Scater: pre-processing, quality control, normalisation and visualisation of single-cell RNA-seq data in R.”
        Bioinformatics, 33, 1179-1186. <a href="10.1093/bioinformatics/btw777">DOI</a>.
    </p>

    <h3>Output</h3>

    <ul>
        <li>QCplots.pdf: QC plots </li>
    </ul>



</body>