<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Seurat -Combine multiple samples </h2>

    <h3>Description</h3>

    <p>
        This tool can be used to combine, integrate, or align multiple Seurat objects for downstream joint analysis.

    </p>

    <h3>Parameters</h3>
    <ul>
        <li>Normalisation method used previously [Global scaling normalization]</li>
        <li>Anchor identification method [CCA, RPCA]</li>
        <li>Number of CCs to use in the neighbor search [20]</li>
        <li>Number of PCs to use in the anchor weighing [20]</li>
        <li>Samples to use as references [No references selected]</li>

    </ul>


    <h3>Details</h3>

    <p>
        Please input the Seurat objects that were generated with <b>Seurat Setup tool</b> and
        preprocessed with <b>Seurat -Filter, normalize, regress and detect variable genes</b> or
        the <b>SCTransform</b> tool. Please ensure that you indicate which normalisation method
        was used. <br>
        Please note: You should confirm that you have assigned the sample or group name in the Setup tool (use short names like "CTRL1",
        "CTRL2"
        "TREAT1", or "TREAT2").
        These names will be used downstream to separate the samples.
    </p>

    <h4>Normalization method: Global scaling normalization vs. SCTransform</h4>
    <p>
        You will need to input the tool previously used for normalization in the first parameter. If you used this tool <b>Seurat -Filter, normalize, regress and detect variable genes</b>, then you will need to select
        <b>Global scaling normalization</b>, which is using NormalizeData. Alternatively, if you used the tool: <b>Seurat -SCTransform: Filter,
            normalize, regress and detect
            variable genes</b>, you will need to use <b>SCTransform</b>.
        If selecting SCTransform,
        an extra step PrepSCTIntegration will be run, then
        FindIntegrationAnchors() and IntegrateData() functions will acknowledge the normalization method,
        and the ScaleData step will be skipped.

    </p>

    <h4>Canonical Correlation Analysis (CCA) vs. Reciprocal PCA (RPCA) in anchor identification</h4>

    <p>
        For the <b>"anchor identification method"</b> parameter, you can use the default, canonical correlation analysis (CCA), or alternatively select reciprocal PCA (RPCA), which is signficantly faster and more conservative.
        When determining anchors between any two datasets using RPCA, each dataset is projected into the
        others' PCA space and the anchors are constrained by the same mutual neighborhood requirement.
    </p>
    <p>
        By identifying shared sources of variation between datasets, CCA is well-suited for identifying anchors
        when cell types are conserved, but there are substantial differences in gene expression
        across experiments, i.e., when experimental
        conditions or disease states introduce very strong expression shifts,
        or when integrating datasets across modalities and species.
        However, CCA-based integration may also lead to overcorrection,
        especially when a large proportion of cells are non-overlapping across datasets.
    </p>
    <p>
        RPCA-based integration involves cells in different biological states being less likely to 'align' after integration.
        RPCA is therefore recommended during the following types of integrative analysis:
    <ul>
        <li>A substantial fraction of cells in one dataset have no matching type in another </li>
        <li>Datasets originate from the same platform (i.e., multiple lanes of 10x Genomics) </li>
        <li>There are a large number of datasets or cells to integrate </li>
    </ul>
    See the link to Seurat RPCA vignette below for more information.
    </p>


    <h4>Large dataset integration and using part of samples as references</h4>
    <p>
        The <b>"Samples to use as references"</b> parameter allows you to list sample names which are used as integration
        references.
        This is especially useful when integrating <b>large datasets</b>, as the tool might run out of memory if the
        reference samples are not predetermined.
        By default, all samples are compared to each other (i.e., for 10 samples,
        there are 45 comparisons).
        You can list the sample name(s) to be used as reference(s) in the corresponding parameter field, and separate multiple names using comma(s).
        Please ensure that the sample name is typed correctly (exactly like you typed it in the Setup tool).
        Also, please ensure that you select good representation of your samples as references (e.g., in the event that you may have samples from male and
        female patients, please pick one reference from both).
        Please see the link to Seurat vignette below for more information.

    </p>

    <h4>Integration steps and parameters</h4>

    <p>
        This tool performs data integration of multiple samples (in .Robj).
        First, Canonical correlation analysis (CCA) is performed to identify common sources of variation between
        the datasets.
        The tool then identifies anchors, assigns scores (=weights) to these anchors
        and then uses them to integrate the datasets.
    </p>

    <p>
        The third parameter is used to select the dimensions from the CCA/PC which specify the neighbor search space.
        The neighbors are used to determine the anchors for alignment.
    </p>

    <p>
        The fourth parameter is the number of PCs to use in the anchor weighing procedure.
        The anchors and their weights are used to compute the correction vectors,
        which are used for correction, thus allowing the datasets to be integrated.
    </p>

    <h3>Further information</h3>
    <p>
        For more details, please refer to <a
            href="https://satijalab.org/seurat/articles/integration_introduction.html">Seurat tutorials
            for multiple sample analysis</a>.
    </p>

    <p>
        For more details on use of reciprocal PCA, please refer to <a
            href="https://satijalab.org/seurat/articles/integration_rpca.html">Seurat vignette
            for "Fast integration using reciprocal PCA (RPCA)"</a>.
    </p>

    <p>
        For more details on using reference samples for integration, please refer to <a
            href=" https://satijalab.org/seurat/articles/integration_large_datasets.html">Seurat vignette
            "Tips for integrating large datasets"</a>.
    </p>


    <p>
        For a more detailed description of the integration process, please refer to "Comprehensive integration
        of single cell data <a href="https://www.biorxiv.org/content/10.1101/460147v1">the paper</a>
        by Rahul Satija <i>et al.</i>
    </p>

    <h3>Output</h3>

    <ul>

        <li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import into R, but not viewable in
            Chipster. </li>
    </ul>



</body>