<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title></title>
</head>

<body>

	<h2>Seurat -Setup and QC </h2>

	<h3>Description</h3>

	<p>
		This tool creates a Seurat object from one of the following input files (<b>see more info below</b>):
	<ol>
		<li>tar package of a folder which contains 10X filtered feature-barcode matrix files in MEX format</li>
		<li>Matrix/table (DGE = digital expression matrix) containing the counts per each gene in each cell (.tsv, tab
			separated
			table)</li>
		<li>10X or CellBender filtered feature-barcode matrix in hdf5 file format (.h5) </li>

	</ol>
	After setting up the R-object (.Robj), some quality control plots are drawn.
	</p>

	<h3>Parameters</h3>
	<ul>
		<li>Project name for plotting [Project_name]</li>
		<li>Keep genes which are expressed in at least this many cells [3]</li>
		<li>Sample name [empty]</li>
		<li>Sample group [empty]</li>
	</ul>


	<h3>Details</h3>

	<p>
		You can give <b>one</b> of the following files as input. Make sure that your input file is assigned correctly
		under the parameters:
	<ol>
		<li>A tar packed folder of 10X feature-barcode matrix files <i>barcodes.tsv, genes.tsv/features.tsv and
				matrix.mtx
			</i> in MEX format. Note that you should have only these files in a single folder, the tool won't work if
			you have
			subfolders in your tar package. These three files are generated by the Cell Ranger -pipeline, and can be
			found under a folder
			corresponding to the filtered_feature_bc_matrix example
			<a
				href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices">here</a>.
			Make sure that you use the filtered data which contains only the estimated cell-associated barcodes,
			as opposed to the unfiltered data containing barcodes corresponding to empty droplets in addition to
			cell-associated barcodes.
			<br>You need to tar the folder into a package. There are graphical programs for this, such as 7-zip, or
			you can use
			command prompt (Windows) / Terminal (Mac) with command: tar -czvf name-for-your-package.tar.gz
			/path/to/folder-you-want-to-tar. You can also use the Chipster tool <i>Utilities - Make a tar
				package</i>.
		</li>
		<li>A table / DGE matrix which you have for example received from colleague, downloaded from some database like
			GEO,
			or generated with the DropSeq data using the tool <i>Single cell RNA-seq - Create
				digital gene expression matrix</i> in Chipster.<b> Note, that the input file needs to be .tsv (= tab
				separated table)
				format!</b> You can modify the file from .csv (=comma separated format) into .tsv for example in Excel
			(open file, choose "Save as" an "Text (Tab delimited) (*.txt)" and rename the file ending as .tsv.)
			Please note, that more often than not, this first step of getting the "raw" data into a format that you can
			analyse
			requires some tailoring (such as switching the comma separators etc) -you can often do this in Excel or
			similar tool.
		</li>
		<li>A 10X or CellBender filtered feature-barcode matrix in hdf5 file format (.h5). This is a general
			"Hierarchical Data Format version 5", designed to store
			and
			organize large amounts of data. 10X Cell Ranger software can output the feature-barcode matrix in this
			format in addition to the MEX format.
			The <i> Remove background contamination with CellBender</i> tool also produces the feature-barcode matrix in
			this format.</li>
	</ol>

	This tool allows you to filter genes based on the number of cells they are detected in. It also produces the
	following quality control plots and counts the number of cells present in the data
	(note that it might be easier to view the large QCplots.pdf file with an external viewer):
	<ul>
		<li>Violin plots showing the number of genes (nFeature), UMIs (nCount_RNA) and the percentage of
			mitochondrial genes (percent.mt) in the cells.These plots can be used to estimate the upper limits for
			number of genes or molecules per cell
			and the percentage of mitochondrial transcripts, which you can use in the <a
				href="https://chipster.2.rahtiapp.fi/manual/single-cell-seurat-filter-v5.html">filtering tool</a>.
		</li>
		<li>Scatter plots comparing the percent.mt and nFeature to nCount_RNA.</li>
	</ul>

	<p>
		If you have multiple samples in your data, it is important that you use the "Sample name" parameter to name your
		sample (for example: covid1, covid2, normal1, normal2),
		and the "Sample group" parameter to determine the sample groups (for example COVID, NORMAL).
		These names are used later in the tools meant for multiple sample analysis, such as the CCA and integrated
		analysis tools.
		It is best not to use empty or special characters in the names.
	</p>

	<p>
		For more details, please check the <a href="http://satijalab.org/seurat/get_started.html">Seurat
			tutorials</a>.
	</p>

	<h3>Output</h3>

	<ul>

		<li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. You cannot open
			this file in Chipster
			but you can view the contents of the object using the <i>Extract information from Seurat object </i> tool.
		</li>
		<li>QCplots.pdf: Quality control plots </li>
	</ul>



</body>