<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Seurat -SCTransform: Filter, normalize, regress and detect variable genes </h2>

    <h3>Description</h3>

    <p>
        This tool uses the SCTransform method for normalization, scaling, and finding variable genes.
        You can also choose to regress out differences caused by the cell cycle stage.
        Before normalization, the tool filters out empty and damaged cells, as well as multiplets.
        <br>
        <b>Please note</b>, this tool and the <a
            href="https://chipster.rahtiapp.fi/manual/single-cell-seurat-filter-regress-v3.html">Seurat -Filter,
            normalize, regress and detect variable genes</a>
        tool accomplish the same goal using different methods. Feel free to select which of these two tools suits your needs best.
    </p>

    <h3>Parameters</h3>
    <p>
    <ul>
        <li>Filter out cells which have fewer than this many genes expressed [200]</li>
        <li>Filter out cells which have more than this many genes expressed [2500]</li>
        <li>Filter out cells which have higher mitochondrial transcript percentage [5]</li>
        <li>Number of variable genes to return [3000] </li>
        <li>Regress out cell cycle differences [no] </li>
    </ul>
    </p>


    <h3>Details</h3>

    <p>
        The tool filters out poor quality cells and then performs the three steps included in the SCTransform tool: normalization, scaling and finding variable
        genes.
    </p>
    <p>
        Please input the Seurat R-object (Robj) generated using the Seurat setup
        -tool. The Seurat R-object output can be used as an input for the subsequent tool, Seurat -PCA.
    </p>
    <p>
        Compared to the standard Seurat normalization + scaling + variable gene findings,
        the SCTransform workflow performs more effective normalization by efficiently removing technical effects from the data.
        This allows more PCs to be selected in the PCA step, as the higher PCs are more likely to represent
        subtle, but biologically relevant, sources of heterogeneity instead of variation in sequencing depth.
    </p>


    <ol>
        <li><b>Filtering</b> is performed to remove empty and damaged cells, as well as multiplets.
            You can use the QC-plots.pdf to estimate the parameters for this step.</li>
        <br>
        <li><b>Expression values are normalized</b> using the SCTransform normalisation method, which uses
            Pearson residuals generated by a regularized negative binomial regression,
            where cellular sequencing depth is utilized as a covariate in a generalized linear model (GLM).
            The parameters for the model are estimated by pooling information across genes expressed at
            similar levels.
            This should remove any technical variation while also preserving biological heterogeneity. Moreover, it will also help to avoid overfitting the model to the data.
        </li>
        <br>
        <li><b>Highly variable genes across the cells are detected</b> using the Pearson residuals that were
            computed during the normalization step. The highly variable genes will be used in downstream analysis, like PCA.
        </li>
        <br>
        <li><b>Uninteresting sources of variation in the expression values are regressed out</b> during normalization to improve
            dimensionality reduction and clustering downstream.
            This tool regresses out <!--<i>the number of detected molecules per cell</i> as well as--> the <i>percentage of
                mitochondrial transcript content</i>.

            <br>You can also choose to regress out <i>cell cycle differences</i>.
            By choosing <i>all differences</i>, the tool removes all signal associated with the cell cycle.
            In some cases, this method can negatively impact downstream analysis,
            particularly in differentiating processes, where stem cells are quiescent and differentiated cells are
            proliferating (or vice versa).
            Alternatively, you can <i>regress out the differences between the G2M and S phase scores</i>.
            This means that signals separating non-cycling cells and cycling cells will be maintained while differences
            in the cell cycle phase amongst proliferating cells (which are often uninteresting) will be regressed out of
            the data. For more information, please refer to the <a
				href="https://satijalab.org/seurat/cell_cycle_vignette.html#assign-cell-cycle-scores">Seurat vignette on cell cycle scoring and regression</a>.
            <br>
            In the current Seurat version, a list of cell cycle markers (from <a
                href="http://science.sciencemag.org/content/352/6282/189"> Tirosh <i>et al.,</i> 2015</a>) is loaded with
            Seurat:
            <br>
            <br>
            s.genes
            </br>
            <br>
            "MCM5" "PCNA" "TYMS" "FEN1" "MCM2"
            "MCM4" "RRM1" "UNG" "GINS2" "MCM6"
            "CDCA7" "DTL" "PRIM1" "UHRF1" "MLF1IP"
            "HELLS" "RFC2" "RPA2" "NASP" "RAD51AP1"
            "GMNN" "WDR76" "SLBP" "CCNE2" "UBR7"
            "POLD3" "MSH2" "ATAD2" "RAD51" "RRM2"
            "CDC45" "CDC6" "EXO1" "TIPIN" "DSCC1"
            "BLM" "CASP8AP2" "USP1" "CLSPN" "POLA1"
            "CHAF1B" "BRIP1" "E2F8"
            <br>
            <br>
            g2m.genes
            </br>
            <br>
            "HMGB2" "CDK1" "NUSAP1" "UBE2C" "BIRC5"
            "TPX2" "TOP2A" "NDC80" "CKS2" "NUF2"
            "CKS1B" "MKI67" "TMPO" "CENPF" "TACC3"
            "FAM64A" "SMC4" "CCNB2" "CKAP2L" "CKAP2"
            "AURKB" "BUB1" "KIF11" "ANP32E" "TUBB4B"
            "GTSE1" "KIF20B" "HJURP" "CDCA3" "HN1"
            "CDC20" "TTK" "CDC25C" "KIF2C" "RANGAP1"
            "NCAPD2" "DLGAP5" "CDCA2" "CDCA8" "ECT2"
            "KIF23" "HMMR" "AURKA" "PSRC1" "ANLN"
            "LBR" "CKAP5" "CENPE" "CTCF" "NEK2"
            "G2E3" "GAS2L3" "CBX5" "CENPA"
        </br>
        </li>
    </ol>


    <p>
        The SCTransform function is described here: <a
            href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1874-1">Hafemeister, C., Satija,
            R. Normalization and variance stabilization of single-cell RNA-seq data using regularized negative binomial
            regression. Genome Biol 20, 296 (2019)</a>, DOI: <a
            href="https://doi.org/10.1186/s13059-019-1874-1">https://doi.org/10.1186/s13059-019-1874-1</a>.
        <br>
        The use of the SCTransform function is demonstrated on the <a
            href="https://satijalab.org/seurat/articles/sctransform_vignette.html">Seurat SCTransform vignette page</a>.
    </p>

    <h3>Output</h3>

    <ul>

        <li>seurat_obj_sctransform.Robj: The Seurat R-object to pass to the next Seurat tool, or to import into R, but not viewable in
            Chipster. </li>
        <li>Dispersion.pdf: The variation vs. average expression plots (in the second plot, the 10 most highly variable
            genes are labeled).
            If you have selected to regress out cell cycle differences, PCA plots of cell cycle genes (before and after the
            regression)
            will be added to the end of the PDF. Also, it lists the number of highly variable genes. </li>
    </ul>



</body>