<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title></title>
</head>

<body>

	<h2>Seurat -Clustering and detection of cluster marker genes</h2>

	<h3>Description</h3>

	<p>
		This tool clusters cells, visualizes the result in a tSNE plot, and finds marker genes for the clusters.

	</p>

	<h3>Parameters</h3>
	<ul>
		<li>Number of principal components to use [10]</li>
		<li>Resolution for granularity [0.6]</li>
		<li>Perplexity, expected number of neighbors for tSNE plot [30]</li>
		<li>Point size in tSNE plot [30]</li>
		<li>Min fraction of cells where a cluster marker gene is expressed [0.25]</li>
		<li>Differential expression threshold for a cluster marker gene [0.25]</li>
		<li>Which test to use for finding marker genes [wilcox]</li>
		<li>Give a list of average expression in each cluster [no]</li>
	</ul>

	<h3>Details</h3>

	<p>
		Cells are clustered using principal components (PCs) rather than genes.
		Therefore you need to give as input the Seurat R-object from the Seurat PCA -tool.
		That tool also produces PC heatmaps and the elbow plot, which help you to decide the number of PCs to use.
		<br><br>
		If you have low number of cells in your data, try lowering the perplexity parameter (the expected number of
		neighbours).
		<br><br>
		Graph-based clustering is performed using the Seurat function FindClusters, which first constructs a KNN graph
		using the Euclidean distance in PCA space,
		and then refines the edge weights between any two cells based on the shared overlap in their local neighborhoods
		(Jaccards distance).
		It then cuts the graph in clusters using the Louvain algorithm which optimizes the standard modularity function
		(please see the links below for more information).
		<br><br>
		The <b>resolution</b> parameter sets the 'granularity',
		with increased values leading to more clusters.
		It has been found that setting this parameter between 0.6-1.2 typically returns good results for single cell
		datasets
		of around 3K cells. Optimal resolution often increases for larger datasets
		-use a value above (below) 1.0 if you want to obtain a larger (smaller) number of communities.
		<br>
	</p>

	<p>
		In order to visualize the clusters, non-linear dimensional reduction is performed using tSNE on the same PCs as
		used for the graph-based clustering, and the tSNE plot is then colored by the clustering results.
		Cells belonging to the same cluster should co-localize on the tSNE plot, because tSNE aims to place cells, which
		have a similar
		local neighborhood in high-dimensional space, together in low-dimensional space. The perplexity parameter is a
		guess
		about the number of close neighbors each cell has, so it allows you to balance attention between local and
		global aspects of the data (<a href="https://distill.pub/2016/misread-tsne/">read more</a>).
		If you have low number of cells, try lowering the perplexity value.
		The tSNE result is visualized in the <b>tSNEplot.pdf</b>.
	</p>

	<p>
		For visualisation, the tool also computed UMAP dimension reduction. UMAP is another non-linear, graph-based
		dimension reduction method, similar to tSNE. Unlike tSNE, UMAP also considers the global aspects of the data.
		Later on, you can choose whether you want to plot tSNE or UMAP plots.
	</p>
	<p>
		Next, Seurat function FindAllMarkers is used to identify (positive) marker genes for the clusters.
		These genes are differentially expressed between a cluster and all the other cells. You can filter out genes
		prior to statistical testing
		by requiring that a gene has to be expressed in at least a certain fraction of cells in either of the two groups
		(min.pct=0.25).
		You can also require that the change in expression has to be at least certain percentage between the groups
		(thresh.test=0.25).
		Both of these parameters can be set to 0, but with a dramatic increase in time since this will test a large
		number of genes that are unlikely to be highly discriminatory.
		The marker genes for each cluster are written in the <b>markers.tsv</b> file.
		<br>
		<br>Seurat currently implements the following tests:
	<ul>
		<li>"wilcox": Wilcoxon rank sum test (default)</li>
		<li>"bimod": Likelihood-ratio test for single cell gene expression, (McDavid et al., Bioinformatics, 2013)
		</li>
		<li>"roc": Standard AUC classifier</li>
		<li>"t": Student's t-test</li>
		<li>"tobit": Tobit-test for differential gene expression (Trapnell et al., Nature Biotech, 2014)</li>
		<li>"poisson": Likelihood ratio test assuming an underlying poisson distribution. Use only for UMI-based
			datasets</li>
		<li>"negbinom": Likelihood ratio test assuming an underlying negative binomial distribution. Use only for
			UMI-based datasets</li>
		<li>"MAST": GLM-framework that treates cellular detection rate as a covariate (Finak et al, Genome Biology,
			2015)</li>
		<!--  <li>"DESeq2" : DE based on a model using the negative binomial distribution (Love et al, Genome Biology, 2014)</li>-->
	</ul>
	<b>The "Poisson" and "negbiom" options should ONLY be used on UMI datasets</b>, as they assume an underlying
	poisson and negative-binomial distribution, respectively.
	Please note that the DESeq2 method has not been included, because it was not designed for situations where there
	are thousands of samples (cells) and it is therefore very slow.
	</p>

	<p>
		The <b>markers.tsv</b> result file contains marker genes and associated statistics <b>for all the clusters</b>:
	<ul>
		<li><b>p-val</b> = p-values for the differentially expressed genes (larger the p-value -> higher the
			likelihood that the gene is in the list just be chance) </li>
		<li><b>avg_logFC</b> = average log fold change (how much higher (lower) the expression of this gene is in
			the particular
			cluster, compared to all the other cells </li>
		<li><b>pct.1</b> = what percentage of the cells in the particular cluster show some expression for this gene
		</li>
		<li><b>pct.2</b> = what percentage of the cells <b>not</b> in the particular cluster (=all the other cells)
			show some expression for this gene</li>
		<li><b>p-val_adj</b> = adjusted/corrected p-value. This value is multiple testing corrected:
			when we test over thousands of genes, we would statistically start getting some significantly
			differentially
			expressed genes just by chance. There are different methods to correct for this,
			here a <i>Bonferroni</i> correction is used. When filtering the table and reporting your results,
			use this value.</li>
		<li><b>cluster</b> = the cluster for which this gene is a marker. Use this column to filter
			the table to get the markers for a cluster of interest (see below). </li>
	</ul>

	You can retrieve markers for a specific cluster using the tool <a
		href="/manual/ngs-filter-results-column.html"></a>Utilities / Filter table by column value</a> .
	For example, to get the markers for cluster 2, fill in the parameters accordingly: <br>
	Column to filter by = cluster <br>
	Does the first column have a title = no <br>
	Cutoff = 2 <br>
	Filtering criteria = equal-to <br>
	</p>

	<p>
		The tool can also return a table with expression for an 'average' single cell in each cluster.
		Read more about the <a
			href="https://rdrr.io/github/satijalab/seurat/man/AverageExpression.html">AverageExpression function</a>.
	</p>

	<p>
		For more details, please check:<br>
		The <a href="http://satijalab.org/seurat/get_started.html">Seurat tutorials</a>
		<br><br>
		The Seurat clustering approach was heavily inspired by the manuscripts
		<a href="http://bioinformatics.oxfordjournals.org/content/early/2015/02/10/bioinformatics.btv088.abstract">SNN-Cliq,
			Xu and Su, Bioinformatics, 2015</a> and
		<a href="http://www.ncbi.nlm.nih.gov/pubmed/26095251">PhenoGraph, Levine et al., Cell, 2015</a> which applied
		graph-based clustering approaches to scRNA-seq data and CyTOF data, respectively.<br>
		<br><br>
	</p>

	<h3>Output</h3>

	<ul>
		<li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. Not viewable in
			Chipster. </li>
		<li>tSNEplot.pdf: Cluster visualization in tSNE plot, heatmap showing the expression of ten top marker genes (in
			terms of fold change) for each cluster. </li>
		<li>markers.tsv: Top marker genes</li>
		<li>aver_expr_in_clusters.tsv: A table for "average expression" for each gene in each cluster</li>
	</ul>



</body>