<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Seurat -Scater QC </h2>

    <h3>Description</h3>

    <p>
        This tool gives as an output an .pdf file with several plots that can be used to estimate
        the quality of the data. As an input, give a Seurat object. The plots vary a bit depending on
        what values are available in the Seurat input: if the cell-cycle scores are available
        (after the "Filter, normalize, regress and detect variable genes" -tool)
        those are used in plotting.
    </p>

    <h3>Parameters</h3>
    <ul>
        <li>-</li>

    </ul>


    <h3>Details</h3>

    <p>
        The Seurat object given as an input is converted into a Scater object (SCE).
        After this, several QC plots are printed in the output pdf file.
    </p>


    <h4>Plot 1: Most expressed features</h4>
    <p>
        Plot shows the features with the (50) highest average expression across all cells,
        along with their expression in each individual cell.
        In this plot, you can check which genes are the top 50 most expressed, and detect possibly
        overabundant genes.
        For example, you can see if there are mitochondrial genes (MT-) that you might want to
        remove from the analysis.
        Y-axis lists the top genes, and x-axis the percentage of total counts.
        Each small bar/tick represents a cell.
        The color of the tick represents the total number of features in that cell.
        The dot in each line shows the average epression.
    </p>

    <h4>Plot 2: Cumulative expression </h4>
    <p>
        Plot shows the relative proportion of the library size that is accounted for by the 1000 most highly
        expressed features for each cell. Y-axis is the cumulative proportion and
        x-axis number of the top expressers included. Each cell is presented as a line.
    </p>


    <h4>Plot 3: Comparing statistics</h4>
    <p>
        Several QC values plotted against each other.
    </p>


    <h4>Plots 4 and 5: Principal component comparisons</h4>
    <p>
        Here we see first 4 principal components plotted against each other.
        The percentage of the variability of the data explained by each PC is also reported.
        The dots representing the cells are colored based on mitochondrial % and total umber of features.
    </p>

    <h4>Plots 6-9: Explanatory factors</h4>
    <p>
        We can investigate the relative importance of different explanatory factors with these plots.
        In the first two plots, a density plot is created for each variable,
        showing the distribution of R-squared across all genes.
        Scater calculates R-squared for a factor/variable by fitting a line for each gene
        against the factor/variable.
        R-squared is the propotion of variance for dependant variable explained by an independent variable.

        We can also identify PCs that correlate strongly to certain QC factor or meta-data values.
        The second two plots represent these.
        In both cases, some selected factors are plotted, and then the top 10 factors.
        If there are cell-cycle scores present in the Seurat object given as an input, the
        values for the phase, G2M score and S score are also plotted.
    </p>


    <p>
        For more details, please check the <a
            href="https://bioconductor.org/packages/release/bioc/html/scater.html">Scater package in Bioconductor</a>
        and the Scater article: McCarthy DJ, Campbell KR, Lun ATL, Willis QF (2017).
        “Scater: pre-processing, quality control, normalisation and visualisation of single-cell RNA-seq data in R.”
        Bioinformatics, 33, 1179-1186. <a href="10.1093/bioinformatics/btw777">DOI</a>.
    </p>

    <h3>Output</h3>

    <ul>
        <li>QCplots.pdf: Quality control plots </li>
    </ul>



</body>