<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title></title>
</head>

<body>

	<h2>Seurat -Combine two samples </h2>

	<h3>Description</h3>

	<p>
		This tool can be used toc ombine two Seurat objects for later joined analysis.

	</p>

	<h3>Parameters</h3>
	<ul>
		<li>Normalisation method used previously [Global scaling normalization]</li>
		<li>Number of CCs to use in the neighbor search [20]</li>
		<li>Number of PCs to use in the anchor weighting [20]</li>
	</ul>


	<h3>Details</h3>

	<p>
		As inputs, give two Seurat object generated with Seurat Setup tool. <br>
		NOTE: make sure you have assigned the sample or group name in the Setup tool (use short names like "CTRL",
		"TREAT").
		This name will be used later on to separate the samples.
	</p>

	<p>
		The first parameter asks about the normalisation step performed either
		in the tool: <b>Seurat -Filter, normalize, regress and detect variable genes</b>, in which case the
		<b>Global scaling normalization</b> (NormalizeData) method was used, or in tool: <b>Seurat -SCTransform: Filter,
			normalize, regress and detect
			variable genes</b>,
		in which case the user should change the parameter to <b>SCTransform</b>.
		In case SCTransform option is selected,
		an extra step (PrepSCTIntegration) is run,
		FindIntegrationAnchors() and IntegrateData() functions acknowledge the normalisation method,
		and the ScaleData step is skipped.

	</p>

	<p>
		This tool performs data integration of two samples (in .Robj).
		First, Canonical correlation analysis (CCA) is performed to identify common sources of variation between
		the datasets.
		The tool then identifies anchors, gives scores (=weights) to these anchors
		and then uses these anchors to integrate the datasets.
	</p>

	<p>
		The first parameter tells which dimensions from the CCA to use for specifying the neighbor seach space.
		The neighbors are used to determine the anchors for the alignment.
	</p>

	<p>
		The second parameter is the number of PCs to use in the anchor weighting procedure.
		The anchors and their weights are used to compute the correction vectors,
		which are used for correction, thus allowing the datasets to be integrated.
	</p>


	<p>
		For more details, please check the <a href="https://satijalab.org/seurat/immune_alignment.html">Seurat tutorials
			for multiple sample analysis</a>.
	</p>
	<p>
		For more detailed description of the integration process, read "Comprehensive integration
		of single cell data <a href="https://www.biorxiv.org/content/10.1101/460147v1">the paper</a>
		by Rahul Satija et al.
	</p>

	<h3>Output</h3>

	<ul>

		<li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. Not viewable in
			Chipster. </li>
	</ul>



</body>