<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title></title>
</head>

<body>

	<h2>Seurat -Setup and QC </h2>

	<h3>Description</h3>

	<p>
		This tool generates a Seurat object from one of the following input files:
	<ol>
		<li>Tar package of a folder which contains the 10X Genomics output files</li>
		<li>Digital gene expression (DGE) matrix containing the counts per gene in each cell (.tsv)</li>
		<li>HDF5 file (.h5) </li>

	</ol>
	Next, some quality control (QC) plots are generated.
	</p>

	<h3>Parameters</h3>
	<ul>
		<li>Project name for plotting [Project_name]</li>
		<li>Keep genes which are expressed in at least this many cells [3]</li>
		<li>Sample or group name [empty]</li>
	</ul>


	<h3>Details</h3>

	<p>
		You can provide <b>one</b> of the following files as input. Please make sure that your input file is assigned to the correct file type:
	<ol>
		<li><b>A tar packed folder of 10X Genomics output files</b> <i>barcodes.tsv, genes.tsv/features.tsv</i> and <i>matrix.mtx</i>. Please note that you should only have these files in a single folder; the tool will not work if you have
			subfolders in your tar package.
			<br>These three files are generated using the Cell Ranger -pipeline and can be found in the folder
			corresponding to the filtered_feature_bc_matrix example
			<a
				href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices">here</a>.
			Please ensure that you use the filtered data containing only the cell barcodes,
			as opposed to the unfiltered data containing millions of barcodes (this would otherwise jam the tool).
			<br>You will need to tar the folder into a package. There are graphical programs for this, such as 7-zip, or
			you can use
			Command Prompt (Windows) / Terminal (Mac) with the command: tar -czvf name-for-your-package.tar.gz
			/path/to/folder-you-want-to-tar. You can also use the Chipster tool <i>Utilities - Make a tar
				package</i>.
		</li>
		<li><b>A DGE matrix</b> that was obtained from a database like
			GEO
			or generated using DropSeq data from the tool <i>Single cell RNA-seq - Create
				digital gene expression matrix</i> in Chipster. Please note, the input file needs to be a .tsv. You can modify the file from .csv (=comma separated format) into .tsv in Excel
			(open file, choose "Save as" a "Text (Tab delimited) (*.txt)" and rename the file so that its filename extension is .tsv.)
			Please note, that more often than not, the first step of transforming raw data into an analyzable format
			requires some tuning (e.g., switching the comma separators). You can often do these types of modifications in Excel or
			another similar tool.
		</li>
		<li><b>A 10X CellRanger hdf5 file (.h5).</b> This is a general "Hierarchical Data Format version 5", designed to store
			and
			organize large amounts of data. 10X CellRanger software can output the data in this format. </li>
	</ol>

	This tool allows you to filter genes based on the number of cells in which they are detected. It also counts the number of cells present in the data and produces the
	QC plots listed below. Please note, it may be easier to view the QCplots.pdf using an external viewer.
	<ul>
		<li><b>Violin plots</b> showing the number of genes (nFeature), UMIs (nCount_RNA), and the percentage of
			mitochondrial genes (percent.mt) in cells. These plots can be used when selecting the thresholds for filtering, which can be done using the tool: <a
				href="https://chipster.csc.fi/manual/single-cell-seurat-filter-regress.html">Seurat v4 -Filter cells, normalize, regress and detect variable genes</a>.
		</li>
		<li><b>Scatter plots</b> comparing both the percent.mt and nFeature to nCount_RNA.</li>
	</ul>

	<p>
		If you have multiple samples, use the "Sample or group name" parameter to name your sample(s).
		These names are later used with tools meant for multiple sample analysis, such as the CCA and integrated
		analysis tools.
		Choose short names, like "CTRL", "TREATED".
	</p>

	<p>
		For more details, please check the <a href="http://satijalab.org/seurat/get_started.html">Seurat
			tutorials</a>.
	</p>

	<h3>Output</h3>

	<ul>

		<li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import into R, but is not viewable in
			Chipster. </li>
		<li>QCplots.pdf: QC plots </li>
	</ul>



</body>