<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title></title>
</head>

<body>

	<h2>Seurat -Setup and QC </h2>

	<h3>Description</h3>

	<p>
		This tool creates a Seurat object from one of the following input files (<b>see more info below</b>):
	<ol>
		<li>tar package of a folder which contains the 10X Genomics output files</li>
		<li>A matrix/table (DGE = digital expression matrix) containing the counts per each gene in each cell (.tsv, tab
			separated
			table)</li>
		<li>A HDF5 file format (.h5) </li>

	</ol>
	After setting up the R-object (.Robj), some quality control plots are drawn.
	</p>

	<h3>Parameters</h3>
	<ul>
		<li>Project name for plotting [Project_name]</li>
		<li>Keep genes which are expressed in at least this many cells [3]</li>
		<li>Sample or group name [empty]</li>
	</ul>


	<h3>Details</h3>

	<p>
		You can give <b>one</b> of the following files as input. Make sure that your input file is assigned correctly
		under the parameters:
	<ol>
		<li>tar packed folder of 10X Genomics output files <i>barcodes.tsv, genes.tsv/features.tsv and matrix.mtx
			</i>. Note that you should have only these files in a single folder, the tool won't work if you have
			subfolders in your tar package.
			<br>These three files are generated by the Cell Ranger -pipeline, and can be found under a folder
			corresponding to the filtered_feature_bc_matrix example
			<a
				href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices">here</a>.
			Make sure that you use the filtered data which contains only the cell barcodes present in your data,
			as opposed to the unfiltered data containing millions of barcodes (this will very likely jam the tools).
			<br>You need to tar the folder into a package. There are graphical programs for this, such as 7-zip, or
			you can use
			command prompt (Windows) / Terminal (Mac) with command: tar -czvf name-for-your-package.tar.gz
			/path/to/folder-you-want-to-tar. You can also use the Chipster tool <i>Utilities - Make a tar
				package</i>.
		</li>
		<li>A table / DGE matrix which you have for example received from colleague, downloaded from some database like
			GEO,
			or generated with the DropSeq data using the tool <i>Single cell RNA-seq - Create
				digital gene expression matrix</i> in Chipster.<b> Note, that the input file needs to be .tsv (= tab
				separated table)
				format!</b> You can modify the file from .csv (=comma separated format) into .tsv for example in Excel
			(open file, choose "Save as" an "Text (Tab delimited) (*.txt)" and rename the file ending as .tsv.)
			Please note, that more often than not, this first step of getting the "raw" data into a format that you can
			analyse
			requires some tailoring (such as switching the comma separators etc) -you can often do this in Excel or
			similar tool.
		</li>
		<li>10X CellRanger hdf5 file (.h5). This is a general "Hierarchical Data Format version 5", designed to store
			and
			organize large amounts of data. 10X CellRanger software can output the data in this format. </li>
	</ol>

	This tool allows you to filter genes based on the number of cells they are detected in. It also produces the
	following quality control plots and counts the number of cells present in the data
	(note that it might be easier to view the large QCplots.pdf file with an external viewer):
	<ul>
		<li>Violin plots showing the number of genes (nFeature), UMIs (nCount_RNA) and the percentage of
			mitochondrial genes (percent.mt) in the cells.These plots can be used to estimate the upper limits for
			number of genes or molecules per cell
			and the percentage of mitochondrial transcripts, which you can use in the <a
				href="https://chipster.csc.fi/manual/single-cell-seurat-filter-regress-v3.html">filtering tool</a>.
		</li>
		<li>Scatter plots comparing the percent.mt and nFeature to nCount_RNA.</li>
	</ul>

	<p>
		If you have multiple samples, use the "Sample or group name" parameter to name your sample.
		These names are used later in the tools meant for multiple sample analysis, such as the CCA and integrated
		analysis tools.
		Choose short names, like "CTRL", "TREATED".
	</p>

	<p>
		For more details, please check the <a href="http://satijalab.org/seurat/get_started.html">Seurat
			tutorials</a>.
	</p>

	<h3>Output</h3>

	<ul>

		<li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. Not viewable in
			Chipster. </li>
		<li>QCplots.pdf: Quality control plots </li>
	</ul>



</body>