<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title></title>
</head>

<body>

	<h2>Seurat -PCA, clustering, and visualization </h2>

	<h3>Description</h3>

	<p>
		This tool performs principal component analysis (PCA) on the highly variable genes across the single cells,
		selected using
		the tool <b>SCTransform: Filter cells, normalize, regress and detect high-variance features in spatial data</b>.
	</p>

	<h3>Parameters</h3>
	<ul>
		<li>Number of PCs to compute [20]</li>
		<li>Print loadings in a file [no]</li>
		<li>Number of genes to list in the loadings file [5]</li>
	</ul>


	<h3>Details</h3>

	<p>
		As input, give the Seurat R-object (Robj) from the tool Seurat -SCTransform: Filter cells, normalize, regress and detect high-variance features in spatial data.
	</p>

	<p>
		If you have low number of cells in your data, you might get an error message. Try lowering the number of
		principal components to compute.
	</p>

	<p>
		To overcome the extensive technical noise in any single gene for scRNA-seq data,
		Seurat clusters cells based on their PCA scores,
		with each PC essentially representing a 'metagene' that combines information across a correlated gene set.
		Determining how many PCs to include downstream is therefore an important step, but it can be challenging.
		The result file <b>PCAplots.pdf</b> contains several plots which can help to explore the primary sources of
		heterogeneity in the dataset and to decide
		which principal components to include in the downstream analysis.
	</p>

	<p>
		This tool produces the following cluster plots.
		<ul>
			<li>Dimplot showing the results of the dimensional reduction on a 2D scatter plot where each point is a cell and it's positioned based on the cell embeddings determined by the reduction technique.
			</li>
			<li>SpatialDimPlot showing the resulting clusters overlaid on the spatial image.</li>
		</ul>
	</p>

	<p>
		You can also request the tool to output a txt file which lists the top N (default = 5)
		genes associated with high/low loadings for the first 5 PCs in the <b>PCAloadings.txt</b> file.
	</p>

	<p>
		For more details, please check the <a href="http://satijalab.org/seurat/get_started.html">Seurat tutorials</a>.
	</p>

	<h3>Output</h3>

	<ul>

		<li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. Not viewable in
			Chipster. </li>
		<!--  <li>PCAgenes.txt: Top genes associated with first principal components</li>-->
		<li>PCAplots.pdf: Graphics presenting the principal components. Last plots are meant to aid in the selection of
			the PCs for the downstream analysis. </li>
		<li>PCAloadings.txt: If requested, the loadings (=genes responsible) of principal components </li>
	</ul>



</body>