<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Seurat v5 -Combine multiple samples, normalization, PCA, clustering and visualization </h2>

    <h3>Description</h3>

    <p>
        This tool can be used to perform a combined analysis by merging or integrating multiple data sets.
        The combined analysis includes similar steps as the single sample analysis which includes 
        normalization, PCA, clustering and visualization. Use this tool instead of the single sample analysis
        tool when you want to analyze multiple samples at the same time.
    </p>

    <h3>Parameters</h3>
    <ul>
        <li>Number of variable genes to return in SCTransform [3000]</li>
        <li>Combining method [merge]</li>
        <li>Number of PCs to compute [50]</li>
        <li>Print loadings in a file [no]</li>
        <li>Number of genes to list in the loadings file [5]</li>
        <li>Dimensions of reduction to use as input [30]</li>
        <li>Resolution for granularity for clustering [0.8]</li>
    </ul>

    <h3>Details</h3>

    <p>
        As inputs, give two or more Seurat objects after the filtering step. This tool will perform similar 
        analysis for multiple samples as explained in the tool <i>Seurat v5 -Normalization, PCA, clustering, 
        and visualization</i>.
    </p>


    <h4>Merge vs integration as the combining method</h4>

    <p>
        The "combining method" parameter determines which method is used to combine.
        By default, merge is used, but the user can also choose to use integration.
        Merge simply combines the objects into one.
    </p>

    <p>
        The integration method uses canonical correlation analysis (CCA) to identify common sources of variation across datasets.
        After this, the tool identifies anchors, generates scores (=weights) for these anchors
        and then uses the anchors to integrate the datasets.
        The integration method should be used to remove strong batch effects present in the data. More information on data integration
        can be found from single-cell tool <i>Seurat v5 -Integrate multiple samples</i>.
        

    <h3>Further information</h3>

    <p>
        For more details, please check the <a href="https://satijalab.org/seurat/articles/spatial_vignette.html">Seurat tutorial</a> and
         <a href="https://satijalab.org/seurat/articles/integration_introduction.html"> Introduction to scRNA-seq integration</a>.
    </p>


    <h3>Output</h3>

    <ul>

        <li>seurat_obj_multiple.Robj: The Seurat R-object to pass to the next Seurat tool, or to import into R. 
			You cannot open this file in Chipster but you can view the contents of the object using the 
			<i>Seurat v5 -Extract information from Seurat object </i> tool. </li>
        <li>merged_plot.pdf: If the data is combined by merging, this tool will output a UMAP and SpatialDimPlot
                where the data has been merged. </li>
        <li>integrated_plot.pdf: If the data is combined using integration, this tool will output a UMAP and SpatialDimPlot
            where the data has been integrated. </li>
        <li>PCAloadings.txt: If selected, the loadings (=genes responsible) of PCs.</li>
    </ul>



</body>
