<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title></title>
</head>

<body>

	<h2>Seurat v5 -Identify spatially variable genes without pre-annotation</h2>

	<h3>Description</h3>

	<p>
		This tool searches for genes that exhibit spatial differential gene expression patterning without cluster annotation information in the absence of pre-annotated anatomical 
		regions. If you would like to use cluster information, you should use the tool <i> Seurat v5 -Identify spatially variable genes based on clusters</i>. 
    </p>
	<p>
		As an input, you can provide the Seurat object after the "Seurat v5 -Clustering" step. If you performed multiple sample analysis i.e. if your samples have been merged 
		or integrated into one single seurat object, select 'yes' as the value for parameter 'Multiple sample analysis'. However, this tool will still identify spatially variable
		genes for each sample individually.
	</p>

	<h3>Parameters</h3>
	<ul>
			<li>Method to use [moransi]</li>
			<li>Number of variable genes to use [1000]</li>
			<li>Number of genes to plot [6]</li>
			<li>Determine color scale based on all genes [no]</li>
			<li>Multiple sample analysis [no]</li>
	</ul>

	<h3>Details</h3>

	<p>
        This tool identifies and selects genes exhibiting spatial patterning by using the Moran's I ("moransi") or Markvariogram ("markvariogram") method. The Markvariogram method
		models spatial transcriptomics data as a mark point process and computes a 'variogram', which identifies genes whose expression level is dependent on their spatial location. 
		Specifically, this process calculates gamma(r) values measuring the dependence between two spots that are a certain “r” distance apart. The Moran's I method measures spatial
		autocorrelation that explains how much nearby locations resemble each other in terms of gene expression. It computes the Moran's I autocorrelation coefficient using the gene
		expression values, the spatial weights matrix that defines the spatial relationships among locations, and the mean of the gene expression values.
	</p>
		
		By default, spatially variable genes are only computed from the highest variable genes, where variation is calculated independently of spatial location. You can choose to
		speed up the computation by selecting a lower number for the parameter "Number of variable genes to use". However, lowering this number too much may result in loss of 
		performace.
    </p>
	<p>
		This tool finally visualizes and lists the spatially variable genes determined by this tool. By default, the color scale is determined for each 
		gene individually and may differ between genes. If you wish to compare expression between different genes, it may be useful to set the parameter value for "Determine color
		scale based on all genes" to "yes" so that the color scale is the same for all genes. If you performed multiple sample analysis i.e. if your samples have been merged or
		integrated into one single seurat object, select 'yes' as the value for parameter 'Multiple sample analysis'. However, this tool will still identify and visualize spatially
		variable genes for each sample individually.
	</p>

	<p>
		For more details on the spatial transcriptomics analysis pipeline, please check the <a href="https://satijalab.org/seurat/articles/spatial_vignette.html#dimensionality-reduction-clustering-and-visualization-1">Seurat tutorial</a>.
	</p>

	<h3>Output</h3>

	<ul>
		<li>Markerplot2.pdf</li>
		<li>spatially_variable_genes.tsv</li>

	</ul>



</body>
