<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
    <title></title>
</head>

<body>

    <h2>Seurat v4 -Filter spots, normalize with SCTransform and detect high-variance genes </h2>

    <h3>Description</h3>

    <p>
        This tool uses the SCTransform method for normalization, scaling and finding variable genes.
        Before normalization, the tool filters out bad quality spots.
    </p>

    <h3>Parameters</h3>
    <p>
    <ul>
        <li>Filter out spots which have higher mitochondrial transcript percentage [25]</li>
        <li>Filter out spots which have higher hemoglobin transcript percentage [20]</li>
        <li>Number of variable genes to return [3000] </li>
    </ul>
    </p>


    <h3>Details</h3>

    <p>
        The tool filters out spots with damaged cells, and then performs the three steps included in the SCTransform tool: normalization, scaling and finding variable
        genes.
    </p>
    <p>
        As input, you can give the Seurat object from the tool "Seurat v4 -Setup and QC". 
        In multiple sample analysis, the normalised Seurat objects can be combined during the next step, when using the tool "Seurat v4 -Combine multiple samples".
    </p>
    <p>
        Compared to the basic Seurat normalization + scaling + variable gene finding,
        the SCTransform workflow performs more effective normalization, removing technical effects from the data.
        This allows more PCs to be selected in the PCA step, as the higher PCs are more likely to represent
        subtle, but biologically relevant, sources of heterogeneity instead of variation in sequencing depth.
    </p>


    <ol>
        <li><b>Filtering</b> is performed in order to remove damaged spots.
            You can use the QC-plots.pdf to estimate the parameters for this step.</li>
        <br>
        <li><b>Expression values are normalized</b> using the SCTransform normalisation method, which uses
            Pearson residuals from regularized negative binomial regression,
            where cellular sequencing depth is utilized as a covariate in a generalized linear model (GLM).
            The parameters for the model are estimated by pooling information across genes that are expressed at
            similar levels.
            This should remove the technical variation but preserve the biological heterogeneity,
            while also avoiding overfitting the model to the data.
        </li>
        <br>
        <li><b>Genes which are highly variable across the cells are detected</b> using the Pearson residuals
            computed in the normalization step. The highly variable genes will be used in downstream analysis, like PCA.
        </li>
    </ol>


    <p>
        The SCTransform function is described in <a
            href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1874-1">Hafemeister, C., Satija,
            R. Normalization and variance stabilization of single-cell RNA-seq data using regularized negative binomial
            regression. Genome Biol 20, 296 (2019)</a>, DOI: https://doi.org/10.1186/s13059-019-1874-1
        <br>
        The use of the SCTransform function is demonstrated in the <a
            href="https://satijalab.org/seurat/articles/sctransform_vignette.html">Seurat SCTransform vignette page</a>.
    </p>

    <h3>Output</h3>

    <ul>

        <li>seurat_obj_sctransform.Robj: The Seurat R-object to pass to the next Seurat tool, or to import into R, but not viewable in
            Chipster. </li>
    </ul>



</body>
