<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title></title>
</head>

<body>

	<h2>Seurat -PCA, clustering, and visualization </h2>

	<h3>Description</h3>

	<p>
		This tool performs principal component analysis (PCA) on the highly variable genes across the single cells,
		selected using
		the tool <b>Filtering, regression and detection of variable genes</b>.
	</p>

	<h3>Parameters</h3>
	<ul>
		<li>Number of principal components to plot as heatmaps [12]</li>
		<li>Number of PCs to compute [20]</li>
		<li>Print loadings in a file [no]</li>
		<li>Number of genes to list in the loadings file [5]</li>
	</ul>


	<h3>Details</h3>

	<p>
		As input, give the Seurat R-object (Robj) from the tool Seurat -Filtering, regression and detection of variable
		genes.
	</p>

	<p>
		If you have low number of cells in your data, you might get an error message. Try lowering the number of
		principal components to compute.
	</p>

	<p>
		To overcome the extensive technical noise in any single gene for scRNA-seq data,
		Seurat clusters cells based on their PCA scores,
		with each PC essentially representing a 'metagene' that combines information across a correlated gene set.
		Determining how many PCs to include downstream is therefore an important step, but it can be challenging.
		The result file <b>PCAplots.pdf</b> contains several plots which can help to explore the primary sources of
		heterogeneity in the dataset and to decide
		which principal components to include in the downstream analysis.
	</p>

	<p>
		In the first plot, the top genes associated with the first two principal components are shown.
		Second plot shows the PCA plot: principal components 1 and 2 for the cells in the dataset, and the third plot
		is a heatmap focusing on principal component 1.
		In heatmap, both cells and genes are sorted by their principal component scores: both extremes are shown.
		Rows represent the extreme genes, and columns are the extreme cells.
		Purple: positive correlation, yellow: negative correlation.
	</p>

	<p>
		Next plots can be used to decide which principal components to use. By default, 12 heatmaps are plotted,
		focusing on the 12 first principal components, in which both cells and genes are sorted by their principal
		component scores.
		You can change the number of heatmaps to plot. Explore especially the PCs which you choose for downstream
		analysis:
		heatmaps can display the 'extremes' across both genes and cells,
		and can be useful to exclude PCs that may be driven primarily by ribosomal/mitochondrial or cell cycle genes.
	</p>

	<p>
		The last plot shows the standard deviations of the principal components.
		Here, the <b>cutoff</b> is where there is an elbow in the graph.
	</p>

	<p>
		You can also request the tool to output a txt file which lists the top N (default = 5)
		genes associated with high/low loadings for the first 5 PCs in the <b>PCAloadings.txt</b> file.
	</p>

	<p>
		For more details, please check the <a href="http://satijalab.org/seurat/get_started.html">Seurat tutorials</a>.
	</p>

	<h3>Output</h3>

	<ul>

		<li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. Not viewable in
			Chipster. </li>
		<!--  <li>PCAgenes.txt: Top genes associated with first principal components</li>-->
		<li>PCAplots.pdf: Graphics presenting the principal components. Last plots are meant to aid in the selection of
			the PCs for the downstream analysis. </li>
		<li>PCAloadings.txt: If requested, the loadings (=genes responsible) of principal components </li>
	</ul>



</body>