<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
	<title></title>
</head>

<body>

	<h2>Seurat v4-Setup and QC </h2>

	<h3>Description</h3>

	<p>
		This tool creates a Seurat object from a tar packaged folder containing the 10x Genomics output files.
	    After setting up the R-object (.Robj), some quality control plots are drawn.
	</p>

	<h3>Parameters</h3>
	<ul>
		<li>Sample name [slice1]</li>
	</ul>

	<h3>Details</h3>

	<p>
		This tool creates a Seurat object from a tar packed folder containing the following 10X Genomics output files:
    <ol>
        <li>filtered_feature_bc_matrix.h5</li>
        <li>tissue_positions_list.csv</li>
        <li>scalefactors_json.json</li>
        <li>tissue_hires_image.png</li>
        <li>tissue_lowres_image.png</li>
    </ol>

		<br>These five files are generated by the Space Ranger -pipeline.
		<a href="https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/output/matrices">here</a>.
		Make sure that you use the filtered data which contains only the cell barcodes present in your data,
		as opposed to the unfiltered data containing millions of barcodes (this will very likely jam the tools). Moreover, 
		make sure the tissue_positions_list.csv doesn't contain the column names.
		<br>You need to tar the folder into a package. There are graphical programs for this, such as 7-zip, or
		you can use
		command prompt (Windows) / Terminal (Mac) with command: tar -czvf name-for-your-package.tar.gz
		/path/to/folder-you-want-to-tar. You can also use the Chipster tool <i>Utilities - Make a tar
		package</i>.

	<p>
		If you have multiple samples, use the "Sample name" parameter to name each sample.
		These names are then used later in the analysis to, for example, distinguish different samples in multiple sample analysis.
	</p>
	
		
	</ul>
	This tool also produces the following quality control plots:
	(note that it might be easier to view the large QCplots.pdf file with an external viewer):
	<ul>
		<li>Violin plots showing the number of UMIs (nCount_Spatial), the number of features (nFeature_Spatial), the percentage of mitochondrial reads (percent_mito) and the percentage of hemoglobin reads (percent_hb) in the spots. These plots can be used to estimate the upper limits for
			the percentage of mitochondrial and hemoglobin transcripts, which you can use in the "Seurat v4 -SCTransform: Filter cells, normalize, regress and detect high-variance features in spatial data" tool</a>.
		</li>
		<li>Spatial feature plot showing the number of UMIs, the number of features and the percentages of mitochondrial and hemoglobin reads in the spots on the tissue section.</li>
	</ul>

	<p>
		For more details, please check the <a href="https://satijalab.org/seurat/articles/spatial_vignette.html#data-preprocessing-2">Seurat
			tutorial</a>.
	</p>

	<h3>Output</h3>

	<ul>

		<li>seurat_spatial_setup.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. Not viewable in
			Chipster. </li>
		<li>QC_plots.pdf: Quality control plots </li>
	</ul>



</body>