<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>STAR for paired end reads and large input data</h2>

<h3>Description</h3>

<p>
    This tool uses STAR to align large paired-end reads to a publicly available reference genome. You need
    to supply the paired-end reads in FASTQ format. The files can be compressed with gzip. Use this
    tool if you have more than 200 million reads per sample.
</p>
<p> 
    Note that this tool uses publicly available reference genomes. If you would like us to add new reference 
    genomes to Chipster, please contact us. 
</p>
<p>
    If you have multiple samples of paired-end reads, you can use the "Run for Each Sample" option to run multiple 
    alignment jobs at the same time. To define the files with paired-end reads for each sample, please use the
    "Define Samples" option. This option allows to define paired-end reads for multiple samples and enables the
    "Run for Each Sample" option such that the alignment tool can be run for multiple samples at the same time. 
    Please see the tutorial video <a href="https://www.youtube.com/watch?v=TIDS6aBHqqo">Define Samples to Enable 
    “Run for Each Sample” Option</a> for more information.
</p>
<p> 
    Alignment results are given in a BAM file, which is automatically indexed. 
</p>

<h3>Parameters</h3>
<ul>
<li>Genome [Homo_sapiens.GRCh38]</li>
<li>Create index file [No index file]</li>
<li>Maximum alignments per read [10]</li>
<li>Maximum mismatches per read pair [10]</li>
<li>Create log files [Final log only]</li>
</ul>

<h3>Details</h3>

<p>This tool uses the STAR (Spliced Transcripts Alignment to a Reference) aligner, which can accurately detect annotated and novel splice junctions in RNA-seq data. 
The tool uses a 2-pass mapping process where STAR performs the 1st pass mapping, automatically extracts splice junctions, inserts them into the genome index, and re-maps all reads in the 2nd mapping pass.
This doesn't increase the number of detected novel junctions, but it allows more spliced reads mapping to novel junctions.
</p>

<p>Chipster offers an Ensembl GTF file to detect annotated splice junctions, but you can also give your own. For example the <a href = "http://www.gencodegenes.org/">GENCODE</a> GTF is recommended by the STAR developers. 
The minimum required overhang is 1 base for annotated junctions and 8 bases for novel junctions. The tool uses STAR's parameter <i>--outFilterType BySJout</i> to filter out alignments which contain spurious novel junctions.
It also uses the STAR parameter <i>--outSAMstrandField intronMotif</i> to add XS strand tags to spliced reads (the Cufflinks assembler needs these) and to remove unannotated non-canonical junctions for which the strand cannot be determined.
</p>
<p>
Maximum alignments per read -parameter sets the maximum number of loci the read is allowed to map to. Alignments (all of them) will be output
only if the read maps to no more loci than this value. Otherwise no alignments will be output, and the read will be counted as "mapped to too many loci" in the Log_final.txt file.
</p>
<p>
Maximum number of mismatches per pair -parameter filters out alignments which contain more mismatches than this number. Use value 999 to switch off this filter.
</p>
<p>
Note that if you have more than two FASTQ files <i>per sample</i> (for example, Illumina NextSeq produces 8
FASTQ files per sample), you will need to provide two list files containing the ordered file names in order to assign
the FASTQ files to each direction. Please produce the list files using the tool <a
href="make_namelist.html">"Utilities / Make a list of file names"</a>.
</p>

<h3>Output</h3>

<ul>
<li>alignment.bam: alignments sorted by chromosomal coordinates.</li>
<li>alignment.bam.bai: index file for the alignments</li>
<li>Log_final.txt: summary file listing the percentage of uniquely mapped reads etc.</li>
<li>Log_progress.txt: process summary allowing you to see for example if the 2nd pass mapping was done.</li>
</ul>

<h3>References</h3>

<p>This tool uses the STAR aligner. Please cite the article:</p>

<p>Alexander Dobin et al: STAR: ultrafast universal RNA-seq aligner (2013) Bioinformatics 29: 15-21.

<p>Please see the <a href = "https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf">STAR manual</A> for more details.</p>
</body>