<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">

<head>
        <title>Chipster manual</title>
</head>

<body>

        <h2>Analysis tools for next generation sequencing data</h2>


        <p>Chipster's NGS analysis tools are grouped in the categories listed below. Chipster also contains a <a
                        href="genomeBrowser.html">built-in genome browser</a>,
                allowing you to view reads and results in their genomic context.</p>

        <ol>
                <li> <a href="#quality">Quality control</a></li>
                <li> <a href="#preprocessing">Preprocessing</a></li>
                <li> <a href="#utilities">Utilities</a></li>
                <li> <a href="#regions">Matching genomic regions</a></li>
                <li> <a href="#alignment">Alignment</a></li>
                <li> <a href="#variants">Variants</a></li>
                <li> <a href="#RNA">RNA-seq</a></li>
                <li> <a href="#miRNA">miRNA-seq</a></li>
                <li> <a href="#ChIP">ChIP- and DNase-seq</a></li>
                <li> <a href="#CNA">CNA-seq</a></li>
                <li> <a href="#micropreproc">Microbial amplicon data preprocessing</a></li>
                <li> <a href="#microanalysis">Microbial amplicon data analyses</a></li>
                <li> <a href="#singlecell">Single-cell RNA-seq</a></li>
        </ol>
        <hr>
        <ul>
                <b><a name="quality">Quality control</a></b>
                <li> <a href="fastqc.html">Read quality with FastQC</a></li>
                <li> <a href="fastx-statistics.html">Quality statistics with FASTX</a></li>
                <li> <a href="prinseq-statistics.html">Read quality statistics with PRINSEQ</a></li>
                <li> <a href="rseqc.html">RNA-seq quality metrics with RseQC</a></li>
                <li> <a href="rseqc_infer_rnaseq_experiment.html">RNA-seq strandedness inference and inner distance
                                estimation
                                using RseQcC</a></li>
                <li> <a href="picard-collectmultiplemetrics.html">Collect multiple metrics from BAM</a></li>
                <!-- <li> <a href="deseq2-pca-heatmap.html">PCA and heatmap of samples with DESeq2</a></li> -->
                <li> <a href="heatmap-for-rnaseq.html">Heatmap for RNA-seq results</a></li>
                <li> <a href="check-fastq.html">Check FASTQC file for errors</a></li>
                <li> <a href="multiqc.html">Combine reports using MultiQC</a></li>
                <br>
                <br>
                <b><a name="preprocessing">Preprocessing</a></b>
                <li> <a href="trimmomatic.html">Trim reads with Trimmomatic</a></li>
                <li> <a href="prinseq-trimmer.html">Trim reads for several criteria with PRINSEQ</a></li>
                <li> <a href="fastx-trimmer.html">Trim reads with FASTX</a></li>
                <li> <a href="tagcleaner-trim.html">Trim primers/adaptors with TagCleaner</a></li>
                <li> <a href="tagcleaner-predict.html">Predict primers/adaptors with TagCleaner</a></li>
                <li> <a href="tagcleaner-statistics.html">Statistics for primers/adaptors with TagCleaner</a></li>
                <li> <a href="prinseq-filter.html">Filter reads for several criteria with PRINSEQ</a></li>
                <li> <a href="fastx-quality-filter.html">Filter reads for quality with FASTX</a></li>
                <li> <a href="fastx-clipper.html">Filter reads for adapters, length and Ns with FASTX</a></li>
                <br>
                <br>
                <b><a name="utilities">Utilities</a></b>
                <li> <a href="ngs-create-experiment.html">Define NGS experiment</a></li>
                <li> <a href="ngs-filter-annotations.html">Filter table by column term</a></li>
                <li> <a href="ngs-filter-results-column.html">Filter table by column value</a></li>
                <li> <a href="filter-bed-value.html">Filter BED by column value</a></li>
                <li> <a href="extract-samples-from-dataset.html">Extract samples from table</a></li>
                <li> <a href="select_columns.html">Table converter</a></li>
                <li> <a href="sort_and_select_columns.html">Sort table by column</a></li>
                <li> <a href="join_tsv_files.html">Join tables</a></li>
                <li> <a href="general_file_processing.html">Modify text</a></li>
                <li> <a href="row_count.html">Count rows</a></li>
                <li> <a href="process-preprocessed-counts.html">Preprocess count table</a></li>
                <li> <a href="normalise-counts.html">Normalize count table</a></li>
                <li> <a href="annotate-ensembl-ids.html">Annotate Ensembl identifiers</a></li>
                <li> <a href="deseq2-normalized-counts-for-gene.html">Plot normalized counts for a gene</a></li>
                <li> <a href="samtools-index.html">Index BAM</a></li>
                <li> <a href="samtools-sort-index-BAM.html">Sort and index BAM</a></li>
                <li> <a href="samtools-convert-SAMtoBAM-sort-index.html">Convert SAM to BAM, sort and index</a></li>
                <li> <a href="samtools-subset.html">Make a subset of BAM</a></li>
                <li> <a href="samtools-merge.html">Merge BAM</a></li>
                <li> <a href="samtools-idxstats.html">Count alignments per chromosome in BAM</a></li>
                <li> <a href="samtools-alignment-statistics.html">Count alignment statistics for BAM</a></li>
                <li> <a href="samtools-count.html">Count alignments in BAM</a></li>
                <li> <a href="samtools-rmdup.html">Remove duplicate reads from BAM</a></li>
                <li> <a href="samtools-consensus.html">Create consensus sequence from BAM</a></li>
                <li> <a href="unique-alignments-from-bam.html"> Retrieve unique alignments from BAM</a></li>
                <li> <a href="bedtools-bamtobed.html">Convert BAM to BED</a></li>
                <li> <a href="fastx-fastq-to-fasta.html">Convert FASTQ to FASTA</a></li>
                <li> <a href="samtools-index-fasta.html">Index FASTA</a></li>
                <li> <a href="subset-fastq.html">Make a subset of FASTQ</a></li>
                <li> <a href="fastx-renamer.html">Rename reads in FASTQ</a></li>
                <li> <a href="fastx-collapser.html">Remove duplicate reads from FASTQ</a></li>
                <li> <a href="bedtools-sort.html">Sort BED</a></li>
                <li> <a href="sort-gtf.html">Sort GTF</a></li>
                <li> <a href="sort-vcf.html">Sort VCF</a></li>
                <li> <a href="sort-tsv.html">Sort TSV</a></li>
                <li> <a href="bedtools-fastafrombed.html">FASTA from BED</a></li>
                <li> <a href="extract-features-from-BED.html">Extract features from BED</a></li>
                <li> <a href="DownloadFile.html">Download file from url to server</a></li>
                <li> <a href="srafetch.html">Retrieve FASTQ files from SRA database</a></li>
                <li> <a href="srabamfetch.html">Retrieve BAM files from SRA database</a></li>
                <li> <a href="ensemblfetch.html">Retrieve sequences for a given organism in Ensembl</a></li>
                <li> <a href="gunzip.html">Extract .gzip file</a></li>
                <li> <a href="extract_gzip_tar.html">Extract .gzip.tar file</a></li>
                <li><a href="gzip.html">Compress a file with gzip</a></li>
                <li> <a href="make_tar.html">Make a tar package</a></li>
                <li><a href="list_tar_contents.html">List contents of a tar file</a></li>
                <li> <a href="a5-miseq.html">A5 assembly pipeline for microbial genomes</a></li>
                <li> <a href="make_namelist.html">Make a list of file names</a></li>
                <li> <a href="virusdetect.html">VirusDetect</a></li>
                <li> <a href="virusdetect-with-index-building.html">VirusDetect with own host genome</a></li>

                <br>
                <br>
                <b><a name="regions">Matching genomic regions</a></b>
                <li> <a href="FindOverlappingTool.html">Find overlapping regions</a></li>
                <li> <a href="RemoveOverlappingTool.html">Remove overlapping regions</a></li>
                <li> <a href="CombineRegionsTool.html">Combine region files</a></li>
                <li> <a href="bedtools-subtractbed.html">Subtract BED</a></li>
                <li> <a href="bedtools-intersectbed.html">Intersect BED</a></li>
                <li> <a href="bedtools-closestbed.html">Closest BED</a></li>
                <li> <a href="bedtools-windowbed.html">Window BED</a></li>
                <li> <a href="bedtools-mergebed.html">Merge BED</a></li>
                <li> <a href="bedtools-coveragebed.html">Coverage BED</a></li>
                <li> <a href="bedtools-genomecoveragebed.html">Genome coverage BED</a></li>
                <li> <a href="bedtools-overlap.html">Overlap BED</a></li>
                <li> <a href="bedtools-slopbed.html">Increase regions in BED</a></li>
                <li> <a href="bedtools-pairtobed.html">Overlap BEDPE with BED</a></li>
                <li> <a href="bedtools-pairtopair.html">Compare two BEDPE files</a></li>
                <br>
                <br>
                <b><a name="alignment">Alignment</a></b>

                <li> <a href="bowtie2.html">Bowtie2 for single end reads</a></li>
                <li> <a href="bowtie2-paired-end.html">Bowtie2 for paired end reads</a></li>
                <li> <a href="bowtie2-with-index-building.html">Bowtie2 for single end reads and own genome</a></li>
                <li> <a href="bowtie2-paired-end-with-index-building.html">Bowtie2 for paired end reads and own
                                genome</a></li>
                <li> <a href="hisat2.html">HISAT2 for single end reads</a></li>
                <li> <a href="hisat2-paired-end.html">HISAT2 for paired end reads</a></li>
                <li> <a href="hisat2-single-end-with-index-building.html">HISAT2 for single end reads and own genome</a>
                </li>
                <li> <a href="hisat2-paired-end-with-index-building.html">HISAT2 for paired end reads and own genome</a>
                </li>
                <li> <a href="star-single-end.html">STAR for single end reads</a></li>
                <li> <a href="star-paired-end.html">STAR for paired end reads</a></li>
                <li> <a href="bwa-mem.html">BWA MEM for single or paired end reads</a></li>
                <li> <a href="bwa-mem-with-index-building.html">BWA MEM for single or paired-end reads and own
                                genome</a></li>
                <li> <a href="bowtie.html">Bowtie for single-end reads</a></li>
                <li> <a href="bowtie-paired-end.html">Bowtie for paired-end reads</a></li>
                <li> <a href="bowtie-with-index-building.html">Bowtie for single-end reads and own genome</a></li>
                <li> <a href="bowtie-paired-end-with-index-building.html">Bowtie for paired-end reads and own genome</a>
                </li>
                <li> <a href="tophat2.html">TopHat2 for paired end reads</a></li>
                <li> <a href="tophat2-single-end.html">TopHat2 for single end reads</a></li>
                <li> <a href="tophat2.html">TopHat2 for paired end reads and own genome</a></li>
                <li> <a href="tophat2-single-end.html">TopHat2 for single end reads and own genome</a></li>
                <li> <a href="bwa.html">BWA-backtrack for single-end reads</a></li>
                <li> <a href="bwa-paired-end.html">BWA-backtrack for paired-end reads</a></li>
                <li> <a href="bwa-with-index-building.html">BWA-backtrack for single-end reads and own genome</a></li>
                <li> <a href="bwa-paired-end-with-index-building.html">BWA-backtrack for paired-end reads and own
                                genome</a>
                </li>
                <br>
                <br>
                <b><a name="variants">Variants</a></b>
                <li><a href="gatk4-mutect2-call-snv-and-indels.html">GATK4 -Call somatic SNVs and INDELs with
                                Mutect2</a></li>
                <li><a href="gatk4-createsomaticpanelofnormals.html">GATK4 -Create Somatic Panel of Normals</a></li>
                <li><a href="gatk4-getpileupsummaries.html">GATK4 -Tabulate pileup metrics for inferring
                                contamination</a></li>
                <li><a href="gatk4-estimate-contamination.html">GATK4 -Estimate contamination</a></li>
                <li><a href="gatk4-filtermutectcalls.html">GATK4 -Filter Mutect2 calls</a></li>
                <li><a href="gatk4-collectsequencingartifactmetrics.html">GATK4 -Generate sequencing artifact metrics
                                for
                                filtering by orientation bias</a></li>
                <li><a href="gatk4-filterbyorientation.html">GATK4 -Filter Mutect2 calls by orientation bias</a></li>
                <li><a href="vcftools.html">Filter variants</a></li>
                <li><a href="annotate-variant.html">Annotate variants</a></li>
                <li><a href="vep.html">Ensembl variant effector predictor</a></li>
                <li><a href="bcftools-intersect-vcf.html">Intersect VCF files</a></li>
                <li><a href="vcftools-reports.html">Calculate statistics on VCF file</a></li>
                <li><a href="samtools-snp-indel-single.html"> Call SNPs and short INDELs</a></li>
                <br>
                <br>
                <b><a name="RNA">RNA-seq</a></b>
                <li><a href="htseq-count.html">Count aligned reads per genes with HTSeq</a></li>
                <li><a href="htseq-count-own-gtf.html">Count aligned reads per genes with HTSeq using own GTF</a></li>
                <!--  <li> Map aligned reads to genes with coverageBed using own BED </li>-->
                <li><a href="dexseq-exoncount.html">Count aligned reads per exons for DEXSeq</a></li>
                <li><a href="dexseq-exoncount-own-gtf.html">Count aligned reads per exons for DEXSeq using own GTF</a>
                </li>
                <li><a href="express.html">Count reads per transcripts using eXpress</a></li>
                <li><a href="deseq2-transform.html">Transform read counts</a></li>
                <li><a href="deseq2.html">Differential expression using DESeq2</a></li>
                <li><a href="ngs-dea-edger-RNA.html"> Differential expression using edgeR</a></li>
                <li><a href="edgeR-multivariate.html"> Differential expression using edgeR for multivariate
                                experiments</a></li>
                <li><a href="DEXSeq.html">Differential exon expression using DEXSeq</a></li>
                <li><a href="cufflinks2.html">Assemble transcripts using Cufflinks</a></li>
                <li><a href="cuffmerge2.html">Merge transcript assemblies with Cuffmerge</a></li>
                <li><a href="cuffcompare2.html">Compare assembly to reference using Cuffcompare</a></li>
                <li><a href="cuffdiff2.html">Differential expression using Cuffdiff</a></li>
                <li><a href="cuffdiff2_with_replicates.html">Differential expression using Cuffdiff with replicates</a>
                </li>
                <li><a href="pathways-hypergeometric-cpdb.html">Hypergeometric test for ConsensusPathDB</a></li>
                <br>
                <br>
                <b><a name="miRNA">Small RNA-seq</a></b>
                <li><a href="convert-bam-to-edger.html">Convert genomic BAM file to count table</a></li>
                <li><a href="convert-miRBase-bam-to-edger.html">Convert miRBased BAM file to count table</a></li>
                <li><a href="deseq2.html">Differential expression using DESeq2</a></li>
                <li><a href="ngs-dea-edger-RNA.html">Differential expression using edgeR</a></li>
                <li><a href="edgeR-multivariate.html"> Differential expression using edgeR for multivariate
                                experiments</a></li>
                <li><a href="ngs-annotate-miRNA-targets.html">Find miRNA targets</a></li>
                <li><a href="ngs-pathways-mirna-hyperg-go.html">GO enrichment for miRNA targets</a></li>
                <li><a href="miRNA_RNA-seq_integration.html">Correlate miRNA-seq and RNA-seq data</a></li>
                <li><a href="ngs-correlation-analysis-mirna.html">Correlate miRNA-seq and gene expression array data</a>
                </li>
                <li><a href="ngs-up-down-analysis-mirna.html">Up-down analysis of miRNA target genes with array data</a>
                </li>
                <li><a href="ngs-extract-target-gene-expression.html">Extract gene expression array data for miRNA
                                targets</a>
                </li>
                <li> <a href="virusdetect.html">VirusDetect</a></li>
                <li> <a href="virusdetect-with-index-building.html">VirusDetect with own host genome</a></li>
                <br>
                <br>
                <b><a name="ChIP">ChIP- and DNase-seq</a></b>
                <li> <a href="macs2.html">Find peaks using MACS2</a></li>
                <li> <a href="ngs-find-peaks-macs-one.html">Find peaks using MACS, treatment only</a></li>
                <li> <a href="ngs-find-peaks-macs-two.html">Find peaks using MACS, treatment vs. control</a></li>
                <li> <a href="ngs-find-peaks-fseq.html">Find broad peaks using F-seq</a></li>
                <li> <a href="ngs-find-nearest-genes.html">Find the nearest genes for regions</a></li>
                <li> <a href="ngs-find-unique-genes.html">Find unique and annotated genes</a></li>
                <li> <a href="pathways-ngs-hyperg-go.html">GO enrichment for list of genes</a></li>
                <li> <a href="ngs-find-motifs-jaspar.html">Find motifs with GADEM and match to JASPAR</a></li>
                <li> <a href="dimontdata.html">Dimont sequence extractor</a></li>
                <li> <a href="dimont-extract-data-custom.html">Dimont sequence extractor using own genome</a></li>
                <li> <a href="dimont.html">Find motifs with Dimont</a></li>
                <li> <a href="dimont-predictor.html">DimontPredictor</a></li>
                <br>
                <br>
                <b><a name="CNA">CNA-seq</a></b>
                <li><a href="cna-define-experiment.html ">Define CNA-seq experiment</a></li>
                <li><a href="cna-filter-bins.html ">Filter copy number bins</a></li>
                <li><a href="cna-correct.html">Correct for GC content</a></li>
                <li><a href="acgh-normalize.html">Normalize copy number data</a></li>
                <li><a href="cna-plot-profile.html">Plot copy number profiles</a></li>
                <li><a href="acgh-segment.html">Segment copy number data</a></li>
                <li><a href="acgh-call-aberrations.html">Call aberrations from segmented copy number data</a></li>
                <li><a href="acgh-identify-regions.html">Identify common regions from called copy number data</a></li>
                <li><a href="acgh-add-cytobands.html">Add cytogenetic bands</a></li>
                <li><a href="acgh-count-overlapping-cnvs.html">Count overlapping CNVs</a></li>
                <li><a href="acgh-plot-frequencies.html">Plot copy number aberration frequencies</a></li>
                <li><a href="cna-count-aberrations.html">Count the number of aberrations per sample</a></li>
                <li><a href="acgh-cluster.html">Cluster called copy number data</a></li>
                <li><a href="cna-correlate.html">Plot correlations of called copy number data</a></li>
                <li><a href="acgh-group-test.html">Group tests for called copy number data</a></li>
                <li><a href="acgh-survival-test.html">Survival test for called copy number data</a></li>
                <li><a href="acgh-plot-survival.html">Plot survival curves for called copy number data</a></li>
                <li><a href="cna-fuse-regions-by-column.html">Fuse regions by column value</a></li>
                <li><a href="cna-fuse-regions-manually.html">Fuse regions manually</a></li>
                <li><a href="acgh-convert-from-probes-to-genes.html">Detect genes from called copy number data</a></li>
                <li><a href="acgh-pathways-go.html">GO enrichment for called gene copy numbers</a></li>
                <li><a href="acgh-match-probes.html">Match copy number and expression features</a></li>
                <li><a href="acgh-plot-combined-profile.html">Plot profiles of matched copy number and expression</a>
                </li>
                <li><a href="acgh-expression-test.html">Test for copy number induced expression changes</a></li>
                <li><a href="acgh-plot-combined-expression.html">Plot copy number induced gene expression</a></li>
                <br>
                <br>
                <b><a name="micropreproc">Microbial amplicon data preprocessing</a></b>
                <li><a href="mothur-trimseqs-uniqueseqs.html">Trim primers and barcodes and filter reads</a></li>
                <li><a href="mothur-makegroup.html">Combine FASTQ or FASTA files and make a group file</a></li>
                <li><a href="mothur-makecontigs.html">Combine paired reads to contigs</a></li>
                <li><a href="mothur-screenseqs.html">Screen sequences for several criteria</a></li>
                <li><a href="mothur-uniqueseqs.html">Extract unique sequences</a></li>
                <li><a href="mothur-alignseqs.html">Align sequences to reference</a></li>
                <li><a href="mothur-filterseqs.html">Filter sequence alignment</a></li>
                <li><a href="mothur-precluster.html">Precluster aligned sequences</a></li>
                <li><a href="mothur-chimera-uchime.html">Remove chimeric sequences</a></li>
                <li><a href="mothur-classifyseqs.html">Classify 16S or 18S sequences to taxonomic units using Silva</a></li>
                <li><a href="mothur-classifyseqs-its.html">Classify ITS sequences to taxonomic units using UNITE</a></li>
                <!--<li><a href="mothur-classify-counttable.html">Produce count table and phenodata</li> -->
                <!-- <li><a href="mothur-analyses.html">Statistical analysis for marker gene studies</a></li> -->
                <li><a href="mothur-summaryseqs.html">Summarize sequences</a></li>
                <li><a href="mothur-sffinfo.html">Extract sequences from SFF file</a></li>
                <li><a href="mothur-fastqinfo.html">Split FASTQ file to FASTA and QUAL files</a></li>
                <li><a href="merge-fasta-qual.html">Merge FASTA or QUAL files</a></li>
                <li><a href="fixed_text_conversions.html">Convert contig names to Mothur format</a></li>
                <br>
                <br>
                <b><a name="microanalysis">Microbial amplicon data analyses</a></b>
                <li><a href="mothur-phyloseq-inputs.html">Generate input files for phyloseq</a></li>
                <li><a href="metabarcoding-import.html">Convert mothur files into phyloseq object</a></li>
                <li><a href="metabarcoding-rarecurve-alpha.html">Sequence numbers, rarefaction curve and alpha diversity estimates</a></li>
                <li><a href="metabarcoding-rarefy-even.html">Rarefy to even depth</a></li>
                <li><a href="metabarcoding-taxon-overview.html">Overview of taxon composition</a></li>
                <li><a href="metabarcoding-filter-taxgroup.html">Filter by taxonomic group</a></li>
                <li><a href="metabarcoding-filter-individual.html">Remove selected taxa</a></li>
                <li><a href="metabarcoding-additional-prevalence.html">Additional prevalence summaries</a></li>
                <li><a href="metabarcoding-single-double.html">Remove OTUs with 0-2 occurrences</a></li>
                <li><a href="metabarcoding-prevalence-filter.html">Proportional prevalence filtering</a></li>
                <li><a href="metabarcoding-transform-otu.html">Transform OTU counts</a></li>
                <li><a href="metabarcoding-stacked-barplot.html">OTU relative abundance bar plots</a></li>
                <li><a href="metabarcoding-distance-ordination.html">Distance matrices and ordinations</a></li>
                <li><a href="metabarcoding-global-permanova.html">Global PERMANOVA for OTU abundance data</a></li>
                <li><a href="metabarcoding-permdisp.html">PERMDISP for OTU abundance data</a></li>
                <li><a href="metabarcoding-pairwise-permanova.html">Post-hoc pairwise PERMANOVA for OTU abundance data</a></li>
                <li><a href="metabarcoding-pairwise-tukey.html">Post-hoc pairwise Tukey's HSD for OTU abundance data</a></li>
                <li><a href="metabarcoding-deseq2.html">Differential OTU abundance analysis using DESeq2</a></li>
                <br>
                <br>
                <b><a name="singlecell">Single-cell RNA-seq</a></b>

                <li><a href="single-cell-seurat-setup.html">Seurat -Setup and QC</a></li>
                <li><a href="single-cell-seurat-filter-regress.html">Seurat -Filtering, regression and detection of
                                variable genes</a></li>
                <li><a href="single-cell-seurat-sctransform.html">Seurat -SCTransform: Filtering, regression and
                                detection of variable genes</a></li>
                <li><a href="single-cell-seurat-pca.html">Seurat -PCA</a></li>
                <li><a href="single-cell-seurat-clustering.html">Seurat -Clustering and detection of cluster marker
                                genes</a></li>
                <li><a href="single-cell-seurat-find-markers-v3.html">Seurat -Find differentially expressed genes
                                between clusters</a></li>
                <li><a href="single-cell-seurat-gene-plots.html">Seurat -Visualize genes</a></li>
                <li><a href="single-cell-seurat-featureplot-v3">Seurat -Visualise features in UMAP plot </a></li>
                <li><a href="single-cell-seurat-CCA.html">Seurat -Combine two samples</a></li>
                <li><a href="single-cell-seurat-integrated-analysis.html">Seurat -Integrated analysis of two samples</a>
                </li>
                <li><a href="single-cell-seurat-diffexp-samples.html">Seurat -Find conserved cluster markers and DE
                                genes in two samples</a></li>
                <li><a href="single-cell-seurat-split-dot-plot.html">Seurat -Visualize genes with cell type specific
                                responses in two samples</a></li>
                <li><a href="single-cell-seurat-extract-cluster.html">Seurat BETA -Extract cells in a cluster</a></li>
                <li><a href="single-cell-seurat-subset.html">Seurat BETA -Subset Seurat objects based on gene
                                expression</a> </li>

                <!-- DropSeq tools:-->
                <li><a href="single-cell-preprocessing.html">Preprocessing of DropSeq FASTQ files</a></li>
                <!-- <li><a href="picard-fastqtosam.html">FASTQ to BAM</a></li> -->
                <!-- <li><a href="single-cell-TagBam.html">Tag BAM</a></li> -->
                <!-- <li><a href="single-cell-FilterBam.html">Filter and trim BAM</a></li> -->
                <!-- <li><a href="picard-samtofastq.html">BAM to FASTQ</a></li> -->
                <!-- <li><a href="picard-sortsam.html">Sort BAM</a></li> -->
                <li><a href="picard-merge-bam-alignment.html">Merge aligned and unaligned BAM</a></li>
                <!--<li><a href="single-cell-tag-read-with-gene-exon.html">Tag reads with gene names</a></li> -->
                <li><a href="single-cell-bam-tag-histogram.html">Estimate number of usable cells</a></li>
                <!-- <li><a href="single-cell-detect-bead-synthesis-erros.html">Detect bead synthesis errors</a></li> -->
                <li><a href="single-cell-digital-expression.html">Create digital gene expression matrix</a></li>
                <!--<li><a href="single-cell-bam-tag-histogram.html">BAM tag histogram</a></li>-->








                <!--        <li> <a href=" "> </a> </li>        -->
        </ul>

</body>

</html>