<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Alignment / TopHat2 for single end reads and own genome</h2>

<h3>Description</h3>

<p>
    This tool uses TopHat2 to align single-end reads to a provided reference genome to identify exon-exon splice 
    junctions. You need to supply the single-end reads in FASTQ format. The files can be compressed with gzip.
    You also need to provide the reference genome as input in FASTA format. 
</p>
<p>
    If you have multiple samples of single-end reads, you can use the "Run for Each Sample" option to run multiple 
    single-end alignment jobs at the same time. To define the files with single-end reads for each sample, please 
    use the "Define Samples" option. This option allows to define the multiple different samples of single-end reads
    and enables the "Run for Each Sample" option such that the alignment tool can be run for multiple samples at the 
    same time. This option is similar to the option for paired-end alignment as explained in the tutorial video
     <a href="https://www.youtube.com/watch?v=TIDS6aBHqqo">Define Samples to Enable “Run for Each Sample” Option</a>, 
    but you need to select the "Single end" option in "Define Samples".
</p>

<p>
    Also, supplying a GTF file containing known gene and exon locations is recommended, because it improves the alignment 
    process. 
</p> 


<h3>Parameters</h3> 

<ul>
<li>Library type (fr-unstranded, fr-firststrand, fr-secondstrand) [fr-unstranded]</li>
<li>When GTF file is used, ignore novel junctions (yes, no) [yes]</li>
<li>Base quality encoding used (phred+33, phred+64) [phred+33]</li>
<li>How many hits is a read allowed to have (1-1000000) [20]</li>
<li>Number of mismatches allowed in final alignment [2]</li>
<li>Minimum anchor length (3-1000) [8]</li>
<li>Maximum number of mismatches allowed in the anchor (0, 1, 2) [0]</li>
<li>Minimum intron length [70]</li>
<li>Maximum intron length [500,000]</li>
</ul>

<h3>Details</h3>

<p>TopHat2 maps Illumina RNA-Seq reads to a genome in order to identify exon-exon splice junctions. The alignment process consists of several steps. 
If annotation is available as a GTF file, TopHat will extract the transcript sequences and use Bowtie2 to align reads to this virtual transcriptome first. 
Only the reads that do not fully map to the transcriptome will then be mapped on the genome. 
The reads that still remain unmapped are split into shorter segments, which are then aligned to the genome. Segment mappings are used to find potential splice sites. 
Sequences flanking a splice site are concatenated, and unmapped segments are mapped to them. Segment alignments are then stitched together to form whole read alignments. 
</p>
<p>
The "anchor length" means that TopHat2 will report junctions spanned by reads with at least this many bases on each side of the junction. 
Note that individual spliced alignments may span a junction with fewer than this many bases on one side. However, every junction involved 
in spliced alignments is supported by at least one read with this many bases on each side. By default no mismatches are allowed in the anchor, but you can change this. 
</p>
<p>
TopHat2 will ignore donor-acceptor pairs which are closer than the minimum intron length or further than the maximum intron length apart.  
</p>

<p>After running TopHat2, Chipster indexes the BAM file using the <a href="http://samtools.sourceforge.net/">SAMtools</a> package. </p>

<p>
If your RNA-seq data was produced with a stranded/directional protocol, it is important that you select the correct strandedness option in the parameter "Library type":
<ul>
<li>fr-unstranded (Standard Illumina).</li>
<li>fr-firststrand (dUTP, NSR, NNSR): The right-most end of the fragment (in transcript coordinates) is sequenced. Equivalently, it is assumed that only the strand generated during first strand synthesis is sequenced. For example Illumina TruSeq Stranded Total RNA Sample Prep Kit is a dUTP method.</li>
<li>fr-secondstrand	(Ligation, Standard SOLiD): The left-most end of the fragment (in transcript coordinates) is sequenced. Equivalently, it is assumed that only the strand generated during second strand synthesis is sequenced.</li>
</ul>
You can read more about stranded data <a href="library-type-summary.html">here</a>.
</p>

<h3>Output</h3>
<p>
This tool returns the following files:
<ul>
<li>tophat.bam: BAM file containingthe alignments.</li>
<li>tophat.bam.bai: Index file for the BAM file.</li>
<li>tophat-summary.txt: Summary of the alignment results.</li>
<li>junctions.bed: Each junction consists of two connected BED blocks, where each block is as long as the maximal overhang of any read spanning the junction. 
The score is the number of alignments spanning the junction.</li>
</ul>
In cases where Tophat2 for some reason fails, file tophat2.log is returned instead.
<p>
 

<h3>Reference</h3>

<p>This tool is based on the <a href="http://tophat.cbcb.umd.edu/">TopHat</a> package. Please cite the following article:</p>
Kim D, Petrtea G, Trapnell C, et al. <a href="http://genomebiology.com/2013/14/4/R36/abstract">TopHat2: accurate alignments of transcriptomes in the presence of insertions, deletions and gene fusions.</a>  Genome Biology (2013) 14: R36. 
</p>

</body>