<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>VirusDetect</h2>

<h3>Description</h3>

<p>This tool runs the VirusDetect pipeline (please find a flowchart below), that performs virus identification using small RNA (sRNA) sequencing data. 
Given a FASTQ file, it performs <i>de novo</i> assembly and reference-guided assembly by aligning sRNA reads to the reference database of known viruses. The assembled contigs are 
compared to the reference virus sequences for virus identification first using BLASTN and then BLASTX. Virus assignments are selected based on the three cutoff parameters described below.
</p>
<p>More detailed description of the VirusDetect pipeline is available at the <a href="http://bioinfo.bti.cornell.edu/cgi-bin/virusdetect/index.cgi">home page of VirusDetect</a>.</p>

<h3>Input data</h3>
<p>Input data (reads) should be given as a FASTQ formatted sequence file. If several FASTQ files are provided,
a separate VirusDetect analysis will be done for each file.
</p>

<h3>Parameters</h3>
<ul>
<p>
<li><b>Reference virus database</b>: VirusDetect is mainly used for detecting plant viruses, but you can use it for other viruses too. Use this parameter to select a virus reference database 
matching your sample type.</li>
</p>
<p>
<li><b>Host Organism</b>: If possible, the reads should be cleaned from sequences originating from the host genome. This can be done by mapping the reads to the host genome 
and selecting only those reads that do not match. Also the resulting contigs are aligned to the host genome and the matching ones are removed.
<ul>
<li>Select the host genome from the drop-down menu listing the organisms for which Chipster has a pre-calculated genome index. If your host genome is not available you should set this parameter to none.</li>
<li>If the host genome is not available in Chipster but you have it as a fasta formatted sequence file, you should use the tool <a href="virusdetect-with-index-building.html">VirusDetect with own genome</a>.</li>
</ul>
</li>
</p>
<p>
<li><b>Minimum fraction of a contig covered by virus reference</b>: At least this proportion of the contig has to match to the reference in order to consider the match significant for the virus assignment. 
For example the default value 0.75 means that only contigs that match to a virus reference for more than 75% of their length are considered as significant matches. </li>
</p>
<p>
<li><b>Minimum fraction of virus reference covered by contigs</b>: Virus assignment is reported only if more than the given proportion of the virus reference is
 covered with significant matches to contigs. For example the default value 0.1 means that at least 10 % of the virus reference sequence must match to the contigs generated by VirusDetect.</li>
</p>
<p>
<li><b>Minimum read depth</b>: The average number of times each nucleotide of the reference sequence must be covered by reads in the sample.</li>
</p>
<p>
<li><b>Use input names in output file names</b>: The names of the VirusDetect result files will start with the name of the input file.</li>
</p>
<p>
<li><b>Return results in one archive file</b>: All the output files are stored in a single tar file. This feature is useful if you run VirusDetect on several samples at the same time. 
The result files can be extracted from the tar package using the tool <a href="extract_gzip_tar.html">Extract .tar.gz file</a>.</li>
</ul>

<h3>Output</h3>

VirusDetect produces a large number of result files. Output related options are used to select, what data is returned. By default VirusDetect returns the following files:
<ul>
<li><b>blastn_matching_references.html</b> Table listing reference viruses that have corresponding virus contigs identified by BLASTN. A pdf formatted report file is returned for each match.</li>
<li><b>blastx_matching_references.html</b> Table listing reference viruses that have corresponding virus contigs identified by BLASTX. A pdf formatted report file is returned for each match. </li>
<li><b>blastn_matches.tsv</b> Table of BLASTN matches to the reference virus database.</li>
<li><b>blastx_matches.tsv</b> Table of BLASTX matches to the reference virus database.</li>
<li><b>contigs_with_blastn_matches.fa</b> Sequences of contigs that match to virus references by BLASTN.</li>
<li><b>contigs_with_blastx_matches.fa</b> Sequences of contigs that match to virus references by BLASTX.</li>
<li><b>undetermined.html</b> Table listing the length, siRNA size distribution and 21-22nt percentage of undetermined contigs. Potential virus contigs (21-22 nt > 50%) are indicated in green.</li>
<li><b>undetermined_blast.html</b> Table listing contigs having hits in the virus reference database but not assigned to any reference viruses because they did not meet the coverage or depth criteria.</li>
<li><b>undetermined_contigs.fa</b> Sequences of contigs that do not match to virus references.</li>
<li><b>virusdetect_contigs.fa</b> Sequences of non-redundant contigs derived through reference-guided and <i>de novo</i> assemblies.</li>
</ul>

If the parameter <b>Return matching reference sequences</b> is turned on, also the following files are returned 
<ul>
<li><b>blastn_matching_references.fa</b> and <b>.fai</b>. Virus reference sequences that produced hits for BLASTN search with the potential virus contigs.</li>
<li><b>blastx_matching_references.fa</b> and <b>.fai</b>. Virus reference sequences that produced hits for BLASTX search with the potential virus contigs.</li>
</ul>
If the parameter <b>Return BAM formatted alignments</b> is turned on, also the following files are returned 
<ul>
<li><b>blastn_matches.bam</b> and <b>.bai</b>. BAM file containing the BLASTN alignment of each contig to its corresponding virus reference sequence.</li> 
<li><b>blastx_matches.bam</b> and <b>.bai</b>. BAM file containing the BLASTX alignment of each contig to its corresponding virus reference sequence.</li> 
</ul>


<br></br>
<img src="VirusDetect_vuokaavio.jpg"></img>
<p><b>Figure 1.</b>Flowchart of the VirusDetect pipeline. Green boxes indicate the output files. 
Steps where the parameters are used are indicated with blue letters.</p>

</body>

